<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Integrated Exposure-Omics Analysis Powered by Tidy Principles • tidyexposomics</title>
<!-- mathjax math --><script src="https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/tex-mml-chtml.js" integrity="sha256-MASABpB4tYktI2Oitl4t+78w/lyA+D7b/s9GEP0JOGI=" crossorigin="anonymous"></script><script>
  window.MathJax = {
    chtml: {
      fontURL: "https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/output/chtml/fonts/woff-v2"
    }
  };
</script><script src="lightswitch.js"></script><script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/Atkinson_Hyperlegible-0.4.10/font.css" rel="stylesheet">
<link href="deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><link href="extra.css" rel="stylesheet">
<meta property="og:title" content="Integrated Exposure-Omics Analysis Powered by Tidy Principles">
<meta name="description" content="The tidyexposomics package is designed to facilitate the integration of exposure and omics data to identify exposure-omics associations. We structure our commands to fit into the tidyverse framework, where commands are designed to be simplified and intuitive. Here we provide functionality to perform quality control, sample and exposure association analysis, differential abundance analysis, multi-omics integration, and functional enrichment analysis.">
<meta property="og:description" content="The tidyexposomics package is designed to facilitate the integration of exposure and omics data to identify exposure-omics associations. We structure our commands to fit into the tidyverse framework, where commands are designed to be simplified and intuitive. Here we provide functionality to perform quality control, sample and exposure association analysis, differential abundance analysis, multi-omics integration, and functional enrichment analysis.">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">tidyexposomics</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="articles/tidyexposomics.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/BioNomad/tidyexposomics/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch">
<li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul>
</li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="tidyexposomics--integrated-exposure-omics-analysis-powered-by-tidy-principles">tidyexposomics <a href="#"><img src="articles/logo.png" align="right" width="25%"></a><br><span style="font-size: 0.5em">Integrated exposure-omics analysis powered by tidy principles</span>
<a class="anchor" aria-label="anchor" href="#tidyexposomics--integrated-exposure-omics-analysis-powered-by-tidy-principles"></a>
</h1></div>
<p><br><br></p>
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>The <code>tidyexposomics</code> package is designed to facilitate the integration of exposure and omics data to identify exposure-omics associations. Functions follow the tidyverse framework, where commands are designed to be simplified and intuitive. The tidyexposomics package provides functionality to perform quality control, sample and exposure association analysis, differential abundance analysis, multi-omics integration, and functional enrichment analysis.</p>
<p align="center" width="100%">
<img width="80%" src="articles/overview.png"></p>
</div>
<div class="section level2">
<h2 id="command-structure">Command Structure<a class="anchor" aria-label="anchor" href="#command-structure"></a>
</h2>
<p>To make the package more user-friendly, we have named our functions to be more intuitive. For example, we use the following naming conventions:</p>
<p align="center" width="100%">
<img width="80%" src="articles/command_str.png"></p>
<p>Results can be added to the <code>MultiAssayExperiment</code> object or returned directly with <code>action = 'get'</code>. We suggest adding results, given that pipeline steps are tracked and can be output to the R console, plotted as a workflow diagram, or exported to an excel worksheet.</p>
</div>
<div class="section level2">
<h2 id="quick-start">Quick Start<a class="anchor" aria-label="anchor" href="#quick-start"></a>
</h2>
<p>The following code is an example of a basic tidyexposomics workflow. It includes loading example data, performing basic quality control, running exposure-wide association studies (ExWAS), differential abundance analysis, correlating differentially expressed genes (DEGs) with exposures, and functional enrichment analysis. However, there is so much more to the <code>tidyexposomics</code> package! So check out the <a href="articles/tidyexposomics.html">Get Started</a> page for a more detailed walkthrough of the package’s functionality.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>The <code>tidyexposomics</code> package depends on R (&gt;= 4.4.0) and can be installed using the following code:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Install and Load Packages</span></span>
<span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"BioNomad/tidyexposomics"</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">tidyexposomics</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="load-example-data">Load Example Data<a class="anchor" aria-label="anchor" href="#load-example-data"></a>
</h2>
<p>We provide example data based off the <a href="https://www.sciencedirect.com/science/article/pii/S016041202200349X?via%3Dihub" class="external-link">ISGlobal Exposome data challenge 2021</a>. Here, we will examine how exposures and omics features relate to asthma status.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load the example data</span></span>
<span><span class="fu"><a href="reference/load_example_data.html">load_example_data</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create ExpOmicSet</span></span>
<span><span class="va">expom</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/create_expomicset.html">create_expomicset</a></span><span class="op">(</span></span>
<span>  codebook <span class="op">=</span> <span class="va">annotated_cb</span>,</span>
<span>  exposure <span class="op">=</span> <span class="va">meta</span>,</span>
<span>  omics <span class="op">=</span> <span class="va">omics_list</span>,</span>
<span>  row_data <span class="op">=</span> <span class="va">fdata</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Grab exposure variables</span></span>
<span><span class="va">exp_vars</span> <span class="op">&lt;-</span> <span class="va">annotated_cb</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">category</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>    <span class="st">"exposure to oxygen molecular entity"</span>,</span>
<span>    <span class="st">"aerosol"</span>,</span>
<span>    <span class="st">"environmental zone"</span>,</span>
<span>    <span class="st">"main group molecular entity"</span>,</span>
<span>    <span class="st">"transition element molecular entity"</span>,</span>
<span>    <span class="st">"exposure to environmental process"</span>,</span>
<span>    <span class="st">"polyatomic entity"</span> </span>
<span>  <span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">variable</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="quality-control">Quality Control<a class="anchor" aria-label="anchor" href="#quality-control"></a>
</h2>
<p>We provide several quality control functions including those that handle filtering missing data, imputation, variable normality checks, and variable transformation.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Filter samples and exposures</span></span>
<span><span class="va">expom</span> <span class="op">&lt;-</span> <span class="va">expom</span><span class="op">[</span>, <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">expom</span><span class="op">$</span><span class="va">hs_asthma</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">expom</span> <span class="op">&lt;-</span> <span class="va">expom</span><span class="op">[</span>, <span class="va">expom</span><span class="op">$</span><span class="va">FAS_cat_None</span> <span class="op">==</span> <span class="st">"Low"</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Filter &amp; impute exposures</span></span>
<span><span class="va">expom</span> <span class="op">&lt;-</span> <span class="va">expom</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="reference/filter_missing.html">filter_missing</a></span><span class="op">(</span>na_thresh <span class="op">=</span> <span class="fl">5</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="reference/run_impute_missing.html">run_impute_missing</a></span><span class="op">(</span>exposure_impute_method <span class="op">=</span> <span class="st">"missforest"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Filter omics</span></span>
<span><span class="va">expom</span> <span class="op">&lt;-</span> <span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="reference/filter_omics.html">filter_omics</a></span><span class="op">(</span></span>
<span>    method <span class="op">=</span> <span class="st">"variance"</span>,</span>
<span>    assays <span class="op">=</span> <span class="st">"Methylation"</span>,</span>
<span>    assay_name <span class="op">=</span> <span class="fl">1</span>,</span>
<span>    min_var <span class="op">=</span> <span class="fl">0.05</span></span>
<span>  <span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="reference/filter_omics.html">filter_omics</a></span><span class="op">(</span></span>
<span>    method <span class="op">=</span> <span class="st">"variance"</span>,</span>
<span>    assays <span class="op">=</span> <span class="st">"Metabolomics"</span>,</span>
<span>    assay_name <span class="op">=</span> <span class="fl">1</span>,</span>
<span>    min_var <span class="op">=</span> <span class="fl">0.1</span></span>
<span>  <span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="reference/filter_omics.html">filter_omics</a></span><span class="op">(</span></span>
<span>    method <span class="op">=</span> <span class="st">"expression"</span>,</span>
<span>    assays <span class="op">=</span> <span class="st">"Gene Expression"</span>,</span>
<span>    assay_name <span class="op">=</span> <span class="fl">1</span>,</span>
<span>    min_value <span class="op">=</span> <span class="fl">1</span>,</span>
<span>    min_prop <span class="op">=</span> <span class="fl">0.3</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="co"># Check variable normality &amp; transform variables</span></span>
<span><span class="va">expom</span> <span class="op">&lt;-</span> <span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="co"># Check variable normality</span></span>
<span>  <span class="fu"><a href="reference/run_normality_check.html">run_normality_check</a></span><span class="op">(</span>action <span class="op">=</span> <span class="st">"add"</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  </span>
<span>  <span class="co"># Transform variables </span></span>
<span>  <span class="fu"><a href="reference/transform_exposure.html">transform_exposure</a></span><span class="op">(</span>transform_method <span class="op">=</span> <span class="st">"boxcox_best"</span>,</span>
<span>                     exposure_cols <span class="op">=</span> <span class="va">exp_vars</span><span class="op">)</span> </span></code></pre></div>
</div>
<div class="section level2">
<h2 id="exwas">ExWAS<a class="anchor" aria-label="anchor" href="#exwas"></a>
</h2>
<p>Here we model the association between exposures and asthma status and adjust our model for child age, biological sex, and cohort.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Perform ExWAS Analysis</span></span>
<span><span class="va">expom</span> <span class="op">&lt;-</span> <span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="reference/run_association.html">run_association</a></span><span class="op">(</span></span>
<span>    source <span class="op">=</span> <span class="st">"exposures"</span>,</span>
<span>    outcome <span class="op">=</span> <span class="st">"hs_asthma"</span>,</span>
<span>    feature_set <span class="op">=</span> <span class="va">exp_vars</span>,</span>
<span>    covariates <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"hs_child_age_None"</span>,</span>
<span>                   <span class="st">"e3_sex_None"</span>,</span>
<span>                   <span class="st">"h_cohort"</span><span class="op">)</span>,</span>
<span>    action <span class="op">=</span> <span class="st">"add"</span>,</span>
<span>    family <span class="op">=</span> <span class="st">"binomial"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Visualize associations</span></span>
<span><span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="reference/plot_association.html">plot_association</a></span><span class="op">(</span></span>
<span>    subtitle <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"Covariates:"</span>,</span>
<span>                     <span class="st">"Age,"</span>,</span>
<span>                     <span class="st">"Biological Sex, "</span>,</span>
<span>                     <span class="st">"Cohort"</span><span class="op">)</span>,</span>
<span>    source <span class="op">=</span> <span class="st">"exposures"</span>,</span>
<span>    terms <span class="op">=</span> <span class="va">exp_vars</span>,</span>
<span>    filter_thresh <span class="op">=</span> <span class="fl">0.15</span>,</span>
<span>    filter_col <span class="op">=</span> <span class="st">"p.value"</span>,</span>
<span>    r2_col <span class="op">=</span> <span class="st">"adj_r2"</span><span class="op">)</span></span></code></pre></div>
<p><img src="articles/exwas_assoc.png"></p>
</div>
<div class="section level2">
<h2 id="differential-abundance">Differential Abundance<a class="anchor" aria-label="anchor" href="#differential-abundance"></a>
</h2>
<p>Differentially abundance analysis is supported in <code>tidyexposomics</code>. Here we use <code>limma_trend</code> to identify features associated with asthma status.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Run differential abundance analysis</span></span>
<span><span class="va">expom</span> <span class="op">&lt;-</span> <span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="reference/run_differential_abundance.html">run_differential_abundance</a></span><span class="op">(</span></span>
<span>    formula <span class="op">=</span> <span class="op">~</span> <span class="va">hs_asthma</span> <span class="op">+</span> <span class="va">hs_child_age_None</span> <span class="op">+</span> <span class="va">e3_sex_None</span> <span class="op">+</span> <span class="va">h_cohort</span>,</span>
<span>    method <span class="op">=</span> <span class="st">"limma_trend"</span>,</span>
<span>    scaling_method <span class="op">=</span> <span class="st">"none"</span>,</span>
<span>    action <span class="op">=</span> <span class="st">"add"</span><span class="op">)</span></span>
<span>    </span>
<span><span class="co"># Plot Differential Abundance Results</span></span>
<span><span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="reference/plot_volcano.html">plot_volcano</a></span><span class="op">(</span></span>
<span>    top_n_label <span class="op">=</span> <span class="fl">3</span>,</span>
<span>    feature_col <span class="op">=</span> <span class="st">"feature_clean"</span>,</span>
<span>    logFC_thresh <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span>,</span>
<span>    pval_thresh <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>    pval_col <span class="op">=</span> <span class="st">"P.Value"</span>,</span>
<span>    logFC_col <span class="op">=</span> <span class="st">"logFC"</span>,</span>
<span>    nrow <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p><img src="articles/volcano.png"></p>
</div>
<div class="section level2">
<h2 id="multi-omics-integration">Multi-Omics Integration<a class="anchor" aria-label="anchor" href="#multi-omics-integration"></a>
</h2>
<p>Multi-omics integration is supported to derive insights across omics layers. Here we use the <code>DIABLO</code> method and set the outcome variable of interest to asthma status.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Perform Multi-Omics Integration</span></span>
<span><span class="va">expom</span> <span class="op">&lt;-</span> <span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="reference/run_multiomics_integration.html">run_multiomics_integration</a></span><span class="op">(</span>method <span class="op">=</span> <span class="st">"DIABLO"</span>,</span>
<span>                             n_factors <span class="op">=</span> <span class="fl">5</span>,</span>
<span>                             outcome <span class="op">=</span> <span class="st">"hs_asthma"</span>,</span>
<span>                             action <span class="op">=</span> <span class="st">"add"</span><span class="op">)</span></span>
<span>                             </span>
<span><span class="co"># Identify factors that correlate with the outcome</span></span>
<span><span class="va">expom</span> <span class="op">&lt;-</span> <span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="reference/run_association.html">run_association</a></span><span class="op">(</span></span>
<span>    source <span class="op">=</span> <span class="st">"factors"</span>,</span>
<span>    outcome <span class="op">=</span> <span class="st">"hs_asthma"</span>,</span>
<span>    feature_set <span class="op">=</span> <span class="va">exp_vars</span>,</span>
<span>    covariates <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>      <span class="st">"hs_child_age_None"</span>,</span>
<span>      <span class="st">"e3_sex_None"</span>,</span>
<span>      <span class="st">"h_cohort"</span><span class="op">)</span>,</span>
<span>    action <span class="op">=</span> <span class="st">"add"</span>,</span>
<span>    family <span class="op">=</span> <span class="st">"binomial"</span><span class="op">)</span></span>
<span>    </span>
<span><span class="co"># Extract top features that contribute to a factor</span></span>
<span><span class="va">expom</span> <span class="op">&lt;-</span> <span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="reference/extract_top_factor_features.html">extract_top_factor_features</a></span><span class="op">(</span>method <span class="op">=</span> <span class="st">"percentile"</span>,</span>
<span>                              pval_col <span class="op">=</span> <span class="st">"p_adjust"</span>,</span>
<span>                              pval_thresh <span class="op">=</span> <span class="fl">0.05</span>, </span>
<span>                              percentile <span class="op">=</span> <span class="fl">0.95</span>,</span>
<span>                              action <span class="op">=</span> <span class="st">"add"</span><span class="op">)</span> </span>
<span>                              </span>
<span><span class="co"># Determine which features drive multiple factors</span></span>
<span><span class="va">expom</span> <span class="op">&lt;-</span> <span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="reference/run_factor_overlap.html">run_factor_overlap</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="exposure-omics-association">Exposure-Omics Association<a class="anchor" aria-label="anchor" href="#exposure-omics-association"></a>
</h2>
<p>Now that we have our multi-omics features associated with asthma status, we can correlate these with our exposures. This helps identify how exposure classes may affect asthma biology.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Grab top common factor features and ensure </span></span>
<span><span class="co"># feature is renamed to variable for the variable_map</span></span>
<span><span class="va">top_factor_features</span> <span class="op">&lt;-</span> <span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="reference/extract_results.html">extract_results</a></span><span class="op">(</span>result <span class="op">=</span> <span class="st">"multiomics_integration"</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://purrr.tidyverse.org/reference/pluck.html" class="external-link">pluck</a></span><span class="op">(</span><span class="st">"common_top_factor_features"</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span>variable<span class="op">=</span><span class="va">feature</span>,</span>
<span>                      <span class="va">exp_name</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Correlate top factor features with exposures</span></span>
<span><span class="va">expom</span>  <span class="op">&lt;-</span> <span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="co"># Perform correlation analysis between factor features </span></span>
<span>  <span class="co"># and exposures</span></span>
<span>  <span class="fu"><a href="reference/run_correlation.html">run_correlation</a></span><span class="op">(</span>feature_type <span class="op">=</span> <span class="st">"omics"</span>,</span>
<span>                  variable_map <span class="op">=</span> <span class="va">top_factor_features</span>,</span>
<span>                  exposure_cols <span class="op">=</span> <span class="va">exp_vars</span>,</span>
<span>                  action <span class="op">=</span> <span class="st">"add"</span>,</span>
<span>                  correlation_cutoff <span class="op">=</span> <span class="fl">0.2</span>,</span>
<span>                  pval_cutoff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>                  cor_pval_column <span class="op">=</span> <span class="st">"p.value"</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="co"># Perform correlation analysis between factor features</span></span>
<span>  <span class="fu"><a href="reference/run_correlation.html">run_correlation</a></span><span class="op">(</span>feature_type <span class="op">=</span> <span class="st">"omics"</span>,</span>
<span>                  variable_map <span class="op">=</span> <span class="va">top_factor_features</span>,</span>
<span>                  feature_cors <span class="op">=</span> <span class="cn">T</span>,</span>
<span>                  action <span class="op">=</span> <span class="st">"add"</span>,</span>
<span>                  correlation_cutoff <span class="op">=</span> <span class="fl">0.2</span>,</span>
<span>                  pval_cutoff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>                  cor_pval_column <span class="op">=</span> <span class="st">"p.value"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="enrichment-analysis">Enrichment Analysis<a class="anchor" aria-label="anchor" href="#enrichment-analysis"></a>
</h2>
<p>After identifying features associated with asthma and exposures, we can perform functional enrichment analysis to understand what biological processes are affected.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Run enrichment analysis on factor features correlated with exposures</span></span>
<span><span class="va">expom</span>  <span class="op">&lt;-</span> <span class="va">expom</span>  <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="reference/run_enrichment.html">run_enrichment</a></span><span class="op">(</span></span>
<span>    feature_type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"omics_cor"</span><span class="op">)</span>,</span>
<span>    feature_col <span class="op">=</span> <span class="st">"feature_clean"</span>,</span>
<span></span>
<span>    db <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GO"</span><span class="op">)</span>, </span>
<span>    species <span class="op">=</span> <span class="st">"goa_human"</span>, </span>
<span>    fenr_col <span class="op">=</span> <span class="st">"gene_symbol"</span>,</span>
<span>    padj_method <span class="op">=</span> <span class="st">"none"</span>,</span>
<span>    pval_thresh <span class="op">=</span> <span class="fl">.1</span>,</span>
<span>    min_set <span class="op">=</span> <span class="fl">1</span>,</span>
<span>    max_set <span class="op">=</span> <span class="fl">800</span>,</span>
<span></span>
<span>    clustering_approach <span class="op">=</span> <span class="st">"diana"</span>,</span>
<span>    action <span class="op">=</span> <span class="st">"add"</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot enrichment term network plot</span></span>
<span><span class="va">expom</span>    <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="reference/plot_enrichment.html">plot_enrichment</a></span><span class="op">(</span></span>
<span>    feature_type <span class="op">=</span> <span class="st">"omics_cor"</span>,</span>
<span>    plot_type <span class="op">=</span> <span class="st">"network"</span>,</span>
<span>    label_top_n <span class="op">=</span> <span class="fl">2</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<p><img src="articles/enr_network.png"></p>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/BioNomad/tidyexposomics/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/BioNomad/tidyexposomics/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li>
<a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing tidyexposomics</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Jason Laird <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0009-0000-5994-2236" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a>  </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Jason Laird.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
