<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Introduction to tidyexposomics • tidyexposomics</title>
<!-- mathjax math --><script src="https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/tex-mml-chtml.js" integrity="sha256-MASABpB4tYktI2Oitl4t+78w/lyA+D7b/s9GEP0JOGI=" crossorigin="anonymous"></script><script>
  window.MathJax = {
    chtml: {
      fontURL: "https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/output/chtml/fonts/woff-v2"
    }
  };
</script><script src="../lightswitch.js"></script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/Atkinson_Hyperlegible-0.4.10/font.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Introduction to tidyexposomics">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">tidyexposomics</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/tidyexposomics.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/BioNomad/tidyexposomics/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch">
<li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul>
</li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Introduction to tidyexposomics</h1>
                        <h4 data-toc-skip class="author">Jason
Laird</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/BioNomad/tidyexposomics/blob/HEAD/vignettes/tidyexposomics.Rmd" class="external-link"><code>vignettes/tidyexposomics.Rmd</code></a></small>
      <div class="d-none name"><code>tidyexposomics.Rmd</code></div>
    </div>

    
    
<p>Exposure to environmental factors is a major determinant of health
and disease. The exposome is a term that represents the totality of
environmental exposures (or in practical terms, the integrated
compilation of all physical, chemical, biological, and (psycho)social
influences that impact biology) from conception to death, that drive
disease and overall health (Miller &amp; Banbury Exposomics Consortium,
2025; Wild, 2005). Characterizing the relationship between a
multiplicity of exposures and disease outcomes can be a daunting task.
Epidemiological studies have moved towards exposure-wide association
studies (ExWAS) where sets of exposures are associated with an outcome
(Chung et al., 2024). The advent of high-throughput technologies has
enabled profiling of several layers of biological information. These
layers can also be integrated to better understand the relationship
between exposures and disease outcomes.</p>
<p>The <code>tidyexposomics</code> package is designed to facilitate the
integration of exposure and omics data to identify exposure-omics
associations and their relevance to health outcomes. We structure our
commands to fit into the <code>tidyverse</code> framework, with a
simple, intuitive interface. Here we provide functionality to perform
quality control, sample and exposure association analysis, differential
abundance analysis, multi-omics integration, and functional enrichment
analysis.</p>
<div class="figure" style="text-align: center">
<img src="overview.png" alt="**tidyexposomics pipeline overview.**Exposure and omics data are stored in a MultiAssayExperiment. Data then undergoes quality control, sample and exposure associations, differential abundance analysis, multi-omics integration, and functional enrichment analysis. The pipeline is designed to be flexible and modular, allowing users to customize their analysis." width="120%"><p class="caption">
<strong>tidyexposomics pipeline overview.</strong>Exposure and omics
data are stored in a MultiAssayExperiment. Data then undergoes quality
control, sample and exposure associations, differential abundance
analysis, multi-omics integration, and functional enrichment analysis.
The pipeline is designed to be flexible and modular, allowing users to
customize their analysis.
</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install the current development version</span></span>
<span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"BioNomad/tidyexposomics"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># load the package</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://BioNomad.github.io/tidyexposomics">tidyexposomics</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="command-structure">Command Structure<a class="anchor" aria-label="anchor" href="#command-structure"></a>
</h2>
<hr>
<p>To make the package more user-friendly, we have named our functions
to be more intuitive. For example, we use the following naming
conventions:</p>
<p><img src="command_str.png" width="474" style="display: block; margin: auto;"></p>
<p>We provide functionality to either add results to the existing object
storing the omics/exposure data or to return results directly using
<code>action = "get"</code>. We suggest adding results, given that
pipeline steps are tracked and can be output to the R console, plotted
as a workflow diagram, or exported to an Excel worksheet.</p>
</div>
<div class="section level2">
<h2 id="exposure-metadata-and-ontology-annotation">Exposure Metadata and Ontology Annotation<a class="anchor" aria-label="anchor" href="#exposure-metadata-and-ontology-annotation"></a>
</h2>
<hr>
<div class="section level3">
<h3 id="codebook-setup">Codebook Setup<a class="anchor" aria-label="anchor" href="#codebook-setup"></a>
</h3>
<p>Before starting an exposomics data analysis we recommend having a
codebook, with information on your exposure variables. Some
suggestions:</p>
<ul>
<li><p><strong>Variable Name</strong>: The name of the variable in the
data set.</p></li>
<li><p><strong>Variable Description</strong>: A concise description of
what the variable measures, including units (e.g., “urinary bisphenol A
(ng/mL)”).</p></li>
<li><p><strong>Variable Type</strong>: The type of variable, such as
continuous, categorical, or binary.</p></li>
<li><p><strong>Variable Period</strong>: The period of time over which
the variable was measured, such as “lifetime”, “year”, “month”, or
“day”.</p></li>
<li><p><strong>Variable Location</strong>: The location where the
variable was measured, such as “home”, “work”, “school”, or “geospatial
code”.</p></li>
<li><p><strong>Variable Ontology</strong>: The ontology term associated
with the variable.</p></li>
</ul>
</div>
<div class="section level3">
<h3 id="ontology-choices">Ontology Choices<a class="anchor" aria-label="anchor" href="#ontology-choices"></a>
</h3>
<p>Variables captured in the codebook should be annotated with ontology
terms to provide a standardized vocabulary for the variables. We
recommend using the following ontologies for exposure and outcome
variables:</p>
<ul>
<li><p><a href="https://www.ebi.ac.uk/ols4/ontologies/ecto" class="external-link">Environment
Exposure Ontology</a> to annotate your exposure variables.</p></li>
<li><p><a href="https://www.ebi.ac.uk/ols4/ontologies/hp" class="external-link">Human
Phenotype Ontology</a> to annotate your outcome variables and phenotypic
data.</p></li>
<li><p><a href="https://www.ebi.ac.uk/ols4/ontologies/chebi" class="external-link">Chemical
Entities of Biological Interest</a> to annotate your chemical exposure
variables.</p></li>
</ul>
<p><strong>Why annotate with ontologies?</strong></p>
<ul>
<li><p><strong>Interpretability</strong>: Ontology labels clarify
ambiguous or inconsistently named variables.</p></li>
<li><p><strong>Harmonization</strong>: You can compare and combine
variables across datasets when they map to the same term.</p></li>
<li><p><strong>Grouping</strong>: Ontologies allow you to collapse
fine-grained exposures into broader categories.</p></li>
<li><p><strong>Integration</strong>: Many public tools, knowledge
graphs, and repositories are ontology-aware. This can make your results
more interoperable and reusable.</p></li>
</ul>
</div>
<div class="section level3">
<h3 id="ontology-annotation-app">Ontology Annotation App<a class="anchor" aria-label="anchor" href="#ontology-annotation-app"></a>
</h3>
<p>To help annotate exposure variables, we provide a lightweight shiny
app:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Launch the shiny app to annotate exposure variables</span></span>
<span><span class="fu"><a href="../reference/ont_annot_app.html">ont_annot_app</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><strong>To use the app:</strong></p>
<p>Click <strong>Browse</strong> to select your exposure metadata
file.</p>
<ul>
<li><p>Then you can click the variable you’d like to link to an
annotation term and search in the <strong>Choose Ontology Term</strong>
dropdown.</p></li>
<li><p>After you select a term, you will see a short description of the
term.</p></li>
<li><p>After you are done, click <strong>Apply Annotate</strong> to save
the annotation.</p></li>
<li><p>Now you can group exposures into larger categories by selecting
each line and then choose your ontology and root depth level (where a
lower number means a more general term).</p></li>
<li><p>Then you can click <strong>Apply Categorization</strong> to apply
the selected categorization to the selected rows.</p></li>
<li><p>If the ontology has nothing to do with your variable, you may
manually enter a category in the <strong>Category</strong> column. This
will change the <strong>Category Source</strong> to manual and will not
be linked to the ontology.</p></li>
<li><p>Once you have annotated all your variables click <strong>Download
Annotated CSV</strong> to save the annotated metadata file.</p></li>
</ul>
<p><img src="ont_annot.png" width="150%" style="display: block; margin: auto;"></p>
</div>
</div>
<div class="section level2">
<h2 id="loading-data">Loading Data<a class="anchor" aria-label="anchor" href="#loading-data"></a>
</h2>
<hr>
<p>To get started we need to load the data. The
<code>create_exposomicset</code> function is used to create a
<code>MultiAssayExperiment</code> object that contains exposure and
omics data. As a quick introduction, a <code>MultiAssayExperiment</code>
object is a container for storing multiple assays (e.g., omics data) and
their associated metadata:</p>
<div class="figure" style="text-align: center">
<img src="mae_rep.png" alt="**Overview of the MultiAssayExperiment Object Structure.** Subject level data is captured within the colData of the MultiAssayExperiment. Observations are stored within experiments slots in the MultiAssayExperiment and sampleMap is used to link the data ( (MultiAssay Special Interest Group, 2025) )." width="328"><p class="caption">
<strong>Overview of the MultiAssayExperiment Object Structure.</strong>
Subject level data is captured within the colData of the
MultiAssayExperiment. Observations are stored within experiments slots
in the MultiAssayExperiment and sampleMap is used to link the data (
(MultiAssay Special Interest Group, 2025) ).
</p>
</div>
<p>We use the MultiAssayExperiment object to store the exposure and
omics data. The create_expomicset function has several arguments:</p>
<ul>
<li><p>The codebook argument is a data frame that contains information
about the variables in the exposure metadata. The column names must
contain variable where the values are the column names of the exposure
data frame, and category which contains general categories for the
variable names. This is the data frame you created with the ontology
annotation app!</p></li>
<li><p>The exposure argument is a data frame that contains the exposure
and other metadata.</p></li>
<li><p>The omics argument is a list of data frames that contain the
omics data.</p></li>
<li><p>The row_data argument is a list of data frames that contain
information about the rows of each omics data frame.</p></li>
</ul>
<p>We are going to start by loading in example data pulled from the <a href="https://www.sciencedirect.com/science/article/pii/S016041202200349X#b0065" class="external-link">ISGlobal
Exposome Data Challenge 2021</a> (Maitre et al., 2022). Specifically, we
will examine how exposures and omics features relate to asthma
status.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load Libraries</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://BioNomad.github.io/tidyexposomics">tidyexposomics</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Load the example data</span></span>
<span><span class="fu"><a href="../reference/load_example_data.html">load_example_data</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create the expomicset object</span></span>
<span><span class="va">expom</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_exposomicset.html">create_exposomicset</a></span><span class="op">(</span></span>
<span>    codebook <span class="op">=</span> <span class="va">annotated_cb</span>,</span>
<span>    exposure <span class="op">=</span> <span class="va">meta</span>,</span>
<span>    omics <span class="op">=</span> <span class="va">omics_list</span>,</span>
<span>    row_data <span class="op">=</span> <span class="va">fdata</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>In this tutorial we will focus on asthma patients with a lower
socioeconomic status (SES):</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Keeping only samples with asthma status</span></span>
<span><span class="va">expom</span> <span class="op">&lt;-</span> <span class="va">expom</span><span class="op">[</span>, <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">expom</span><span class="op">$</span><span class="va">hs_asthma</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">expom_1</span> <span class="op">&lt;-</span> <span class="va">expom</span><span class="op">[</span>, <span class="va">expom</span><span class="op">$</span><span class="va">FAS_cat_None</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Low"</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/rm.html" class="external-link">rm</a></span><span class="op">(</span><span class="va">expom</span><span class="op">)</span></span></code></pre></div>
<p>We are interested in how the exposome affects health outcomes, so
let’s define which metadata variables represent exposure variables.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Grab exposure variables</span></span>
<span><span class="va">exp_vars</span> <span class="op">&lt;-</span> <span class="va">annotated_cb</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">category</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>        <span class="st">"exposure to oxygen molecular entity"</span>,</span>
<span>        <span class="st">"aerosol"</span>,</span>
<span>        <span class="st">"environmental zone"</span>,</span>
<span>        <span class="st">"main group molecular entity"</span>,</span>
<span>        <span class="st">"transition element molecular entity"</span>,</span>
<span>        <span class="st">"exposure to environmental process"</span>,</span>
<span>        <span class="st">"polyatomic entity"</span></span>
<span>    <span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">variable</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="quality-control">Quality Control<a class="anchor" aria-label="anchor" href="#quality-control"></a>
</h2>
<hr>
<div class="section level3">
<h3 id="missingness">Missingness<a class="anchor" aria-label="anchor" href="#missingness"></a>
</h3>
<p>Oftentimes when collecting data, there are missing values. Let’s use
the <code>plot_missing</code> function to determine where our missing
values are</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/plot_missing.html">plot_missing</a></span><span class="op">(</span></span>
<span>        plot_type <span class="op">=</span> <span class="st">"summary"</span>,</span>
<span>        threshold <span class="op">=</span> <span class="fl">0</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p><img src="tidyexposomics_files/figure-html/unnamed-chunk-10-1.png" width="576"></p>
<p>Here we see that there are 10 variables in the exposure data that are
missing data. Let’s take a look at them:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/plot_missing.html">plot_missing</a></span><span class="op">(</span></span>
<span>        plot_type <span class="op">=</span> <span class="st">"lollipop"</span>,</span>
<span>        threshold <span class="op">=</span> <span class="fl">0</span>,</span>
<span>        layers <span class="op">=</span> <span class="st">"Exposure"</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p><img src="tidyexposomics_files/figure-html/unnamed-chunk-11-1.png" width="384"></p>
<p>Here we see that one variable, <code>hs_wgtgain_None</code>, has
nearly 30% missing values! We can apply a missingness filter using the
<code>filter_missing</code> function. The <code>na_thresh</code>
argument is used to set the threshold for missing values. For example,
if <code>na_thresh = 20</code>, then any variable with more than 20%
missing values will be removed. Here we set the threshold to 5% to
filter out variables with more than 5% missing values.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Filter out variables with too many missing values</span></span>
<span><span class="va">expom_1</span> <span class="op">&lt;-</span> <span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/filter_missing.html">filter_missing</a></span><span class="op">(</span>na_thresh <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="imputation">Imputation<a class="anchor" aria-label="anchor" href="#imputation"></a>
</h3>
<p>Now that we have filtered out the variables with too many missing
values, we can impute the missing values. The
<code>run_impute_missing</code> function is used to impute missing
values. Here we can specify the imputation method for exposure and omics
data separately.</p>
<p>The <code>exposure_impute_method</code> argument is used to set the
imputation method for exposure data, and the
<code>omics_impute_method</code> argument is used to set the imputation
method for omics data. The <code>omics_to_impute</code> argument is used
to specify which omics data to impute. Here we will impute the exposure
data given using the <code>missforest</code> method, but other options
for imputation methods include:</p>
<ul>
<li><p><code>median</code>: Imputes missing values with the median of
the variable.</p></li>
<li><p><code>mean</code>: Imputes missing values with the mean of the
variable.</p></li>
<li><p><code>knn</code>: Uses k-nearest neighbors to impute missing
values.</p></li>
<li><p><code>mice</code>: Uses the Multivariate Imputation by Chained
Equations (MICE) method to impute missing values.</p></li>
<li><p><code>dep</code>: Uses the DEP method to impute missing
values.</p></li>
<li><p><code>missforest</code>: Uses the MissForest method to impute
missing values.</p></li>
<li><p><code>lod_sqrt2</code>: Imputes missing values using the square
root of the lower limit of detection (LOD) for each variable. This is
useful for variables that have a lower limit of detection, such as
chemical exposures.</p></li>
</ul>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Impute missing values</span></span>
<span><span class="va">expom_1</span> <span class="op">&lt;-</span> <span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/run_impute_missing.html">run_impute_missing</a></span><span class="op">(</span></span>
<span>        exposure_impute_method <span class="op">=</span> <span class="st">"missforest"</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="filtering-omics-features">Filtering Omics Features<a class="anchor" aria-label="anchor" href="#filtering-omics-features"></a>
</h3>
<p>We can filter omics features based on variance or expression levels.
The <code>filter_omics</code> function is used to filter omics features.
The method argument is used to set the method for filtering. Here we can
use either:</p>
<ul>
<li><p><strong>Variance</strong>: Filters features based on variance. We
recommend this for omics based on continuous measurements, such as
log-transformed counts, M-values, protein intensities, or metabolite
concentrations.</p></li>
<li><p><strong>Expression</strong>: Filters features based on expression
levels. We recommend this for omics where many values may be near-zero
or zero, such as RNA-seq data.</p></li>
</ul>
<p>The <code>assays</code> argument is used to specify which omics data
to filter. The <code>assay_name</code> argument is used to specify which
assay to filter. The <code>min_var</code>, <code>min_value</code>, and
<code>min_prop</code> arguments are used to set the minimum variance,
minimum expression value, and minimum proportion of samples exceeding
the minimum value, respectively.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># filter omics layers by variance and expression</span></span>
<span><span class="va">expom_1</span> <span class="op">&lt;-</span> <span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/filter_omics.html">filter_omics</a></span><span class="op">(</span></span>
<span>        method <span class="op">=</span> <span class="st">"variance"</span>,</span>
<span>        assays <span class="op">=</span> <span class="st">"Methylation"</span>,</span>
<span>        assay_name <span class="op">=</span> <span class="fl">1</span>,</span>
<span>        min_var <span class="op">=</span> <span class="fl">0.05</span></span>
<span>    <span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/filter_omics.html">filter_omics</a></span><span class="op">(</span></span>
<span>        method <span class="op">=</span> <span class="st">"variance"</span>,</span>
<span>        assays <span class="op">=</span> <span class="st">"Metabolomics"</span>,</span>
<span>        assay_name <span class="op">=</span> <span class="fl">1</span>,</span>
<span>        min_var <span class="op">=</span> <span class="fl">0.1</span></span>
<span>    <span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/filter_omics.html">filter_omics</a></span><span class="op">(</span></span>
<span>        method <span class="op">=</span> <span class="st">"expression"</span>,</span>
<span>        assays <span class="op">=</span> <span class="st">"Gene Expression"</span>,</span>
<span>        assay_name <span class="op">=</span> <span class="fl">1</span>,</span>
<span>        min_value <span class="op">=</span> <span class="fl">1</span>,</span>
<span>        min_prop <span class="op">=</span> <span class="fl">0.3</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="normality-check">Normality Check<a class="anchor" aria-label="anchor" href="#normality-check"></a>
</h3>
<p>When determining variable associations, it is important to check the
normality of the data. The <code>run_normality_check</code> function is
used to check the normality of the data.</p>
<p>The <code>transform_exposure</code> function is used to transform the
data to make it more normal. Here the transform_method is set to
<code>boxcox_best</code> as it will automatically select the best
transformation method based on the data. The
<code>transform_method</code> can be manually set to <code>log2</code>,
<code>sqrt</code>, or <code>x_1_3</code> as well. We specify the
exposure_cols argument to set the columns to transform.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Check variable normality &amp; transform variables</span></span>
<span><span class="va">expom_1</span> <span class="op">&lt;-</span> <span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="co"># Check variable normality</span></span>
<span>    <span class="fu"><a href="../reference/run_normality_check.html">run_normality_check</a></span><span class="op">(</span>action <span class="op">=</span> <span class="st">"add"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="co"># Transform variables</span></span>
<span>    <span class="fu"><a href="../reference/transform_exposure.html">transform_exposure</a></span><span class="op">(</span></span>
<span>        transform_method <span class="op">=</span> <span class="st">"boxcox_best"</span>,</span>
<span>        exposure_cols <span class="op">=</span> <span class="va">exp_vars</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p>To check the normality of the exposure data, we can use the
<code>plot_normality_summary</code> function. This function plots the
normality of the data before and after transformation. The
<code>transformed</code> argument is set to <code>TRUE</code> to plot
the normality status of the transformed data.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/plot_normality_summary.html">plot_normality_summary</a></span><span class="op">(</span></span>
<span>        transformed <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p><img src="tidyexposomics_files/figure-html/unnamed-chunk-16-1.png" width="384"></p>
</div>
<div class="section level3">
<h3 id="principal-component-analysis">Principal Component Analysis<a class="anchor" aria-label="anchor" href="#principal-component-analysis"></a>
</h3>
<p>To identify the spread of the data, we can perform a principal
component analysis (PCA). The <code>run_pca</code> function is used to
perform PCA on samples with observations in each omics data frame. Here
we specify that we would like to log-transform the exposure and omics
data before performing PCA using the <code>log_trans_exp</code> and the
<code>log_trans_omics</code> arguments, respectively. We automatically
identify sample outliers based on the Mahalanobis distance, a measure of
the distance between a point and a distribution.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Perform principal component analysis</span></span>
<span><span class="va">expom_1</span> <span class="op">&lt;-</span> <span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/run_pca.html">run_pca</a></span><span class="op">(</span></span>
<span>        log_trans_exp <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>        log_trans_omics <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>        action <span class="op">=</span> <span class="st">"add"</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot principal component analysis results</span></span>
<span><span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/plot_pca.html">plot_pca</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="tidyexposomics_files/figure-html/unnamed-chunk-18-1.png" width="864" style="display: block; margin: auto;"></p>
<p>Here we see a few sample outliers, and that most variation is
captured in the first two principal components for both features and
samples. We can filter out these samples using the
<code>filter_sample_outliers</code> function.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Filter out sample outliers</span></span>
<span><span class="va">expom_1</span> <span class="op">&lt;-</span> <span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/filter_sample_outliers.html">filter_sample_outliers</a></span><span class="op">(</span></span>
<span>        outliers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"s73"</span>, <span class="st">"s1183"</span>, <span class="st">"s376"</span>, <span class="st">"s899"</span>, <span class="st">"s828"</span>, <span class="st">"s993"</span><span class="op">)</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p>To understand the relationship between the principal components and
exposures we can correlate them using the <code>run_correlation</code>
function. Here we specify that the <code>feature_type</code> is
<code>pcs</code> for principal components, specify a set of exposure
variables, <code>exp_vars</code>, and the number of principal
components, <code>n_pcs.</code> We set <code>correlation_cutoff</code>
to <code>0</code> and <code>pval_cutoff</code> to <code>1</code> to
initially include all correlations.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expom_1</span> <span class="op">&lt;-</span> <span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/run_correlation.html">run_correlation</a></span><span class="op">(</span></span>
<span>        feature_type <span class="op">=</span> <span class="st">"pcs"</span>,</span>
<span>        exposure_cols <span class="op">=</span> <span class="va">exp_vars</span>,</span>
<span>        n_pcs <span class="op">=</span> <span class="fl">20</span>,</span>
<span>        action <span class="op">=</span> <span class="st">"add"</span>,</span>
<span>        correlation_cutoff <span class="op">=</span> <span class="fl">0</span>,</span>
<span>        pval_cutoff <span class="op">=</span> <span class="fl">1</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p>We can visualize these correlations with the
<code>plot_correlation_tile</code> function. We specify we are plotting
the <code>feature_type</code> of <code>pcs</code> to grab the principal
component correlation results. We then set the significance threshold to
0.05 with the <code>pval_cutoff</code> argument.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/plot_correlation_tile.html">plot_correlation_tile</a></span><span class="op">(</span></span>
<span>        feature_type <span class="op">=</span> <span class="st">"pcs"</span>,</span>
<span>        pval_cutoff <span class="op">=</span> <span class="fl">0.05</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p><img src="tidyexposomics_files/figure-html/unnamed-chunk-21-1.png" width="672"></p>
<p>Here we note that particulate matter exposure and PFHXS exposure
(<code>hs_pfhxs_m_Log2</code>) are associated with the most principal
components, indicating these exposures are driving variation in the
data.</p>
</div>
<div class="section level3">
<h3 id="exposure-summary">Exposure Summary<a class="anchor" aria-label="anchor" href="#exposure-summary"></a>
</h3>
<p>We can summarize the exposure data using the
<code>run_summarize_exposures</code> function. This function calculates
summary statistics for each exposure variable, including the number of
values, number of missing values, minimum, maximum, range, sum, median,
mean, standard error, and confidence intervals. The action argument is
set to get to return the summary statistics as a data frame. We specify
the <code>exposure_cols</code> argument to set the columns to
summarize.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Summarize exposure data</span></span>
<span><span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/run_summarize_exposures.html">run_summarize_exposures</a></span><span class="op">(</span></span>
<span>        action <span class="op">=</span> <span class="st">"get"</span>,</span>
<span>        exposure_cols <span class="op">=</span> <span class="va">exp_vars</span></span>
<span>    <span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 6 × 27</span></span></span>
<span><span class="co">##   variable    n_values  n_na   min   max range   sum median  mean    se ci_lower</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> h_pm10_rat…      139     0  2.29  3.86  1.57  446.   3.17  3.21  0.02     3.16</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> h_pm25_rat…      139     0  3.06  4.66  1.6   542.   3.89  3.9   0.02     3.85</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> hs_bpa_mad…      139     0  0    12.8  12.8  <span style="text-decoration: underline;">1</span>242.   8.75  8.94  0.12     8.7 </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">4</span> hs_cs_c_Lo…      139     0  0     2.35  2.35  157.   1.08  1.13  0.04     1.04</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">5</span> hs_cu_m_Lo…      139     0  2.29  2.41  0.12  326.   2.35  2.35  0        2.34</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">6</span> hs_mepa_ma…      139     0  0.86 14.1  13.2   956.   7.49  6.88  0.21     6.47</span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 16 more variables: ci_upper &lt;dbl&gt;, variance &lt;dbl&gt;, sd &lt;dbl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   coef_var &lt;dbl&gt;, period &lt;chr&gt;, location &lt;chr&gt;, description &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   var_type &lt;chr&gt;, transformation &lt;chr&gt;, selected_ontology_label &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   selected_ontology_id &lt;chr&gt;, root_id &lt;chr&gt;, root_label &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   category &lt;chr&gt;, category_source &lt;chr&gt;, transformation_applied &lt;chr&gt;</span></span></span></code></pre>
</div>
<div class="section level3">
<h3 id="exposure-visualization">Exposure Visualization<a class="anchor" aria-label="anchor" href="#exposure-visualization"></a>
</h3>
<hr>
<p>To visualize our exposure data, we can use the
<code>plot_exposures</code> function. This function allows us to plot
the exposure data in a variety of ways. Here we will plot the exposure
data using a boxplot. The <code>exposure_cat</code> argument is used to
set the exposure category to plot. Additionally, we could specify
<code>exposure_cols</code> to only plot certain exposures. The
<code>plot_type</code> argument is used to set the type of plot to
create. Here we use a boxplot, but we could also use a ridge plot.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot exposure data</span></span>
<span><span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/plot_exposures.html">plot_exposures</a></span><span class="op">(</span></span>
<span>        group_by <span class="op">=</span> <span class="st">"e3_sex_None"</span>,</span>
<span>        exposure_cat <span class="op">=</span> <span class="st">"aerosol"</span>,</span>
<span>        plot_type <span class="op">=</span> <span class="st">"boxplot"</span>,</span>
<span>        ylab <span class="op">=</span> <span class="st">"Values"</span>,</span>
<span>        title <span class="op">=</span> <span class="st">"Aerosol Exposure by Sex"</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p><img src="tidyexposomics_files/figure-html/unnamed-chunk-23-1.png" width="624"></p>
<p>Here we see that there are not significant differences in aerosol
exposure between males and females.</p>
</div>
</div>
<div class="section level2">
<h2 id="sample-exposure-association">Sample-Exposure Association<a class="anchor" aria-label="anchor" href="#sample-exposure-association"></a>
</h2>
<hr>
<div class="section level3">
<h3 id="sample-clustering">Sample Clustering<a class="anchor" aria-label="anchor" href="#sample-clustering"></a>
</h3>
<p>The <code>run_cluster_samples</code> function is used to cluster
samples based on the exposure data. The <code>clustering_approach</code>
argument is used to set the clustering approach to automatically
identify clusters. Here we use the dynamic approach, which uses a
dynamic tree cut method to identify clusters. However, you may choose
between these other options as well:</p>
<ul>
<li><p><code>gap</code>: <strong>Gap statistic method
(default)</strong>; estimates optimal <code>k</code> by comparing
within-cluster dispersion to that of reference data.</p></li>
<li><p><code>diana</code>: <strong>Divisive hierarchical clustering
(DIANA)</strong>; chooses <code>k</code> based on the largest drop in
dendrogram height.</p></li>
<li><p><code>elbow</code>: <strong>Elbow method</strong>; detects the
point of maximum curvature in within-cluster sum of squares (WSS) to
determine <code>k</code>.</p></li>
<li><p><code>dynamic</code>: <strong>Dynamic tree cut</strong>;
adaptively detects clusters from a dendrogram structure without needing
to predefine <code>k</code>.</p></li>
<li><p><code>density</code>: <strong>Density-based clustering</strong>
(via <code>densityClust</code>); identifies clusters based on local
density peaks in distance space.</p></li>
</ul>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Sample clustering</span></span>
<span><span class="va">expom_1</span> <span class="op">&lt;-</span> <span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/run_cluster_samples.html">run_cluster_samples</a></span><span class="op">(</span></span>
<span>        exposure_cols <span class="op">=</span> <span class="va">exp_vars</span>,</span>
<span>        clustering_approach <span class="op">=</span> <span class="st">"dynamic"</span>,</span>
<span>        action <span class="op">=</span> <span class="st">"add"</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  ..cutHeight not given, setting it to 1620  ===&gt;  99% of the (truncated) height range in dendro.</span></span>
<span><span class="co">##  ..done.</span></span></code></pre>
<p>We plot the sample clusters using the
<code>plot_sample_clusters</code> function. This function plots z-scored
values of the exposure data for each sample, colored by the cluster
assignment. The <code>exposure_cols</code> argument is used to set the
columns to plot.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot sample clusters</span></span>
<span><span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/plot_sample_clusters.html">plot_sample_clusters</a></span><span class="op">(</span></span>
<span>        exposure_cols <span class="op">=</span> <span class="va">exp_vars</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p><img src="tidyexposomics_files/figure-html/unnamed-chunk-25-1.png" width="170%"></p>
<p>Here we see five clusters, largely driven by particulate
matter/aerosol exposure. Although it is worth noting that maternal PFHXS
exposure (<code>hs_pfhxs_m_Log2</code>) appears to strongly drive Groups
3 and 4.</p>
</div>
<div class="section level3">
<h3 id="exposure-correlations">Exposure Correlations<a class="anchor" aria-label="anchor" href="#exposure-correlations"></a>
</h3>
<p>The <code>run_correlation</code> function can be used to correlate
exposure variables. The <code>feature_type</code> argument is used to
set the type of features to correlate. Here we use exposures to indicate
that we are correlating exposure variables. The
<code>correlation_cutoff</code> argument is used to set the minimum
correlation for the association. Here we use a cutoff of
<code>0.3</code>, but this can be adjusted based on the data.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expom_1</span> <span class="op">&lt;-</span> <span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/run_correlation.html">run_correlation</a></span><span class="op">(</span></span>
<span>        feature_type <span class="op">=</span> <span class="st">"exposures"</span>,</span>
<span>        action <span class="op">=</span> <span class="st">"add"</span>,</span>
<span>        exposure_cols <span class="op">=</span> <span class="va">exp_vars</span>,</span>
<span>        correlation_cutoff <span class="op">=</span> <span class="fl">0.3</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p>To visualize the exposure correlations, we can use the
<code>plot_circos_correlation</code> function. Here we will plot the
circos plot. This function creates a circular plot of the exposure
correlations. The <code>correlation_cutoff</code> argument is used to
set the minimum correlation score for the association. Here we use a
cutoff of 0.3.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/plot_circos_correlation.html">plot_circos_correlation</a></span><span class="op">(</span></span>
<span>        feature_type <span class="op">=</span> <span class="st">"exposures"</span>,</span>
<span>        corr_threshold <span class="op">=</span> <span class="fl">0.3</span>,</span>
<span>        exposure_cols <span class="op">=</span> <span class="va">exp_vars</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p><img src="tidyexposomics_files/figure-html/unnamed-chunk-27-1.png" width="960" style="display: block; margin: auto;"></p>
<p>Here we see that particulate matter exposure variables are highly
correlated and also correlated with maternal pregnancy BPA
(<code>hs_bpa_madj_Log2</code>) and childhood PCB-153
(<code>hs_pcb153_cadj_Log2</code>) exposure.</p>
</div>
<div class="section level3">
<h3 id="exposure-wide-association-exwas">Exposure-wide association (ExWAS)<a class="anchor" aria-label="anchor" href="#exposure-wide-association-exwas"></a>
</h3>
<p>The <code>run_association</code> function is used to run the
association analysis. The <code>source</code> argument is used to set
the source of the data to associate. The <code>outcome</code> argument
is used to set the outcome variable. The <code>feature_set</code>
argument is used to set the features to include in the analysis. The
<code>covariates</code> argument is used to set the covariates to
include in the analysis. The <code>action</code> argument is set to add
to add the results to the existing object. The <code>family</code>
argument is used to set the family of the model. Here we use
<code>binomial</code> for a binary outcome.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Perform ExWAS Analysis</span></span>
<span><span class="va">expom_1</span> <span class="op">&lt;-</span> <span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/run_association.html">run_association</a></span><span class="op">(</span></span>
<span>        source <span class="op">=</span> <span class="st">"exposures"</span>,</span>
<span>        outcome <span class="op">=</span> <span class="st">"hs_asthma"</span>,</span>
<span>        feature_set <span class="op">=</span> <span class="va">exp_vars</span>,</span>
<span>        covariates <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>            <span class="st">"hs_child_age_None"</span>,</span>
<span>            <span class="st">"e3_sex_None"</span>,</span>
<span>            <span class="st">"h_cohort"</span></span>
<span>        <span class="op">)</span>,</span>
<span>        action <span class="op">=</span> <span class="st">"add"</span>,</span>
<span>        family <span class="op">=</span> <span class="st">"binomial"</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p>To visualize the results of the ExWAS analysis, we can use the
<code>plot_association</code> function. This function plots the
association results for the specified features. The terms argument is
used to set the features to plot. The <code>filter_thresh</code>
argument is used to set the threshold for filtering the results. The
<code>filter_col</code> argument is used to set the column to filter on.
Here we use <code>p.value</code> to filter on the p-value of the
association. We can also include the adjusted R^2 using the
<code>r2_col</code> argument.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/plot_association.html">plot_association</a></span><span class="op">(</span></span>
<span>        subtitle <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span></span>
<span>            <span class="st">"Covariates:"</span>,</span>
<span>            <span class="st">"Age,"</span>,</span>
<span>            <span class="st">"Biological Sex, "</span>,</span>
<span>            <span class="st">"Cohort"</span></span>
<span>        <span class="op">)</span>,</span>
<span>        source <span class="op">=</span> <span class="st">"exposures"</span>,</span>
<span>        terms <span class="op">=</span> <span class="va">exp_vars</span>,</span>
<span>        filter_thresh <span class="op">=</span> <span class="fl">0.15</span>,</span>
<span>        filter_col <span class="op">=</span> <span class="st">"p.value"</span>,</span>
<span>        r2_col <span class="op">=</span> <span class="st">"adj_r2"</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p><img src="tidyexposomics_files/figure-html/unnamed-chunk-29-1.png" width="480" style="display: block; margin: auto;"></p>
<p>Here we see that particulate matter and PFNA
(<code>hs_pfna_c_Log2</code>) are associated with asthma status after
adjusting for age, biological sex, and cohort. Although the p-value
threshold for this association is quite generous.</p>
<p>We can also associate our omics features with an outcome of interest
using the <code>run_association</code> function. Here we specify an
additional argument, <code>top_n</code>, which is used to set the top
number of omics features to include per omics layer.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Perform ExWAS Analysis</span></span>
<span><span class="va">expom_1</span> <span class="op">&lt;-</span> <span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/run_association.html">run_association</a></span><span class="op">(</span></span>
<span>        outcome <span class="op">=</span> <span class="st">"hs_asthma"</span>,</span>
<span>        source <span class="op">=</span> <span class="st">"omics"</span>,</span>
<span>        covariates <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>            <span class="st">"hs_child_age_None"</span>,</span>
<span>            <span class="st">"e3_sex_None"</span>,</span>
<span>            <span class="st">"h_cohort"</span></span>
<span>        <span class="op">)</span>,</span>
<span>        top_n <span class="op">=</span> <span class="fl">500</span>,</span>
<span>        action <span class="op">=</span> <span class="st">"add"</span>,</span>
<span>        family <span class="op">=</span> <span class="st">"binomial"</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p>We can visualize the results using a Manhattan plot. The
<code>plot_manhattan</code> function is used to plot the results. The
<code>pval_thresh</code> argument is used to set the p-value threshold
for significance. The <code>vars_to_label</code> argument is used to set
the variables to label on the plot. The <code>min_per_cat</code>
argument is used to set the minimum number of significant features per
category to include in the plot. The <code>panel_sizes</code> argument
is used to set the size of each panel in the plot.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/plot_manhattan.html">plot_manhattan</a></span><span class="op">(</span></span>
<span>        min_per_cat <span class="op">=</span> <span class="fl">0</span>,</span>
<span>        feature_col <span class="op">=</span> <span class="st">"feature_clean"</span>,</span>
<span>        vars_to_label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>            <span class="st">"TC01000261.hg.1"</span>,</span>
<span>            <span class="st">"TC15000063.hg.1"</span>,</span>
<span>            <span class="st">"cg14780466"</span>,</span>
<span>            <span class="st">"hs_pfna_c_Log2"</span></span>
<span>        <span class="op">)</span>,</span>
<span>        panel_sizes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">3</span>, <span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">3</span>, <span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span>,</span>
<span>        facet_angle <span class="op">=</span> <span class="fl">0</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p><img src="tidyexposomics_files/figure-html/unnamed-chunk-31-1.png" width="480" style="display: block; margin: auto;"></p>
</div>
<div class="section level3">
<h3 id="exposome-scores">Exposome Scores<a class="anchor" aria-label="anchor" href="#exposome-scores"></a>
</h3>
<hr>
<p>We can also calculate exposome scores, which are a summary measure of
exposure. The <code>run_exposome_score</code> function is used to
calculate the exposome score. The <code>exposure_cols</code> argument is
used to set the columns to use for the exposome score. The
<code>score_type</code> argument is used to set the type of score to
calculate. Here we could use:</p>
<ul>
<li><p><code>median</code>: Calculates the median of the exposure
variables.</p></li>
<li><p><code>mean</code>: Calculates the mean of the exposure
variables.</p></li>
<li><p><code>sum</code>: Calculates the sum of the exposure
variables.</p></li>
<li><p><code>pca</code>: Calculates the first principal component of the
exposure variables.</p></li>
<li><p><code>irt</code>: Uses Item Response Theory to calculate the
exposome score.</p></li>
<li><p><code>quantile</code>: Calculates the quantile of the exposure
variables.</p></li>
<li><p><code>var</code>: Calculates the variance of the exposure
variables.</p></li>
</ul>
<p>The <code>score_column_name</code> argument is used to set the name
of the column to store the exposome score in. Here we will define a
score for PFAS using a variety of different methods and demonstrate
their use in association with asthma status.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># determine which PFAS to use</span></span>
<span><span class="va">pfas</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"hs_pfhxs_m_Log2"</span>, <span class="st">"hs_pfna_c_Log2"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">expom_1</span> <span class="op">&lt;-</span> <span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/run_exposome_score.html">run_exposome_score</a></span><span class="op">(</span></span>
<span>        exposure_cols <span class="op">=</span> <span class="va">pfas</span>,</span>
<span>        score_type <span class="op">=</span> <span class="st">"median"</span>,</span>
<span>        score_column_name <span class="op">=</span> <span class="st">"exposome_median_score"</span></span>
<span>    <span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/run_exposome_score.html">run_exposome_score</a></span><span class="op">(</span></span>
<span>        exposure_cols <span class="op">=</span> <span class="va">pfas</span>,</span>
<span>        score_type <span class="op">=</span> <span class="st">"pca"</span>,</span>
<span>        score_column_name <span class="op">=</span> <span class="st">"exposome_pca_score"</span></span>
<span>    <span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/run_exposome_score.html">run_exposome_score</a></span><span class="op">(</span></span>
<span>        exposure_cols <span class="op">=</span> <span class="va">pfas</span>,</span>
<span>        score_type <span class="op">=</span> <span class="st">"irt"</span>,</span>
<span>        score_column_name <span class="op">=</span> <span class="st">"exposome_irt_score"</span></span>
<span>    <span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/run_exposome_score.html">run_exposome_score</a></span><span class="op">(</span></span>
<span>        exposure_cols <span class="op">=</span> <span class="va">pfas</span>,</span>
<span>        score_type <span class="op">=</span> <span class="st">"quantile"</span>,</span>
<span>        score_column_name <span class="op">=</span> <span class="st">"exposome_quantile_score"</span></span>
<span>    <span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/run_exposome_score.html">run_exposome_score</a></span><span class="op">(</span></span>
<span>        exposure_cols <span class="op">=</span> <span class="va">pfas</span>,</span>
<span>        score_type <span class="op">=</span> <span class="st">"var"</span>,</span>
<span>        score_column_name <span class="op">=</span> <span class="st">"exposome_var_score"</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p>We can then associate these exposome scores with the outcome of
interest using the <code>run_association</code> function, just like we
did before. However, this time we specify our <code>feature_set</code>
to be the exposome scores we just calculated.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Associate Exposome Scores with Outcome</span></span>
<span><span class="va">expom_1</span> <span class="op">&lt;-</span> <span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/run_association.html">run_association</a></span><span class="op">(</span></span>
<span>        outcome <span class="op">=</span> <span class="st">"hs_asthma"</span>,</span>
<span>        source <span class="op">=</span> <span class="st">"exposures"</span>,</span>
<span>        feature_set <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>            <span class="st">"exposome_median_score"</span>,</span>
<span>            <span class="st">"exposome_pca_score"</span>,</span>
<span>            <span class="st">"exposome_irt_score"</span>,</span>
<span>            <span class="st">"exposome_quantile_score"</span>,</span>
<span>            <span class="st">"exposome_var_score"</span></span>
<span>        <span class="op">)</span>,</span>
<span>        covariates <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>            <span class="st">"hs_child_age_None"</span>,</span>
<span>            <span class="st">"e3_sex_None"</span>,</span>
<span>            <span class="st">"h_cohort"</span></span>
<span>        <span class="op">)</span>,</span>
<span>        action <span class="op">=</span> <span class="st">"add"</span>,</span>
<span>        family <span class="op">=</span> <span class="st">"binomial"</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p>To plot the results of the exposome score association with the
outcome, we can use the <code>plot_association</code> function:</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/plot_association.html">plot_association</a></span><span class="op">(</span></span>
<span>        subtitle <span class="op">=</span> <span class="st">"Covariates: Age, Biological Sex, Cohort"</span>,</span>
<span>        source <span class="op">=</span> <span class="st">"exposures"</span>,</span>
<span>        terms <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>            <span class="st">"exposome_median_score"</span>,</span>
<span>            <span class="st">"exposome_pca_score"</span>,</span>
<span>            <span class="st">"exposome_irt_score"</span>,</span>
<span>            <span class="st">"exposome_quantile_score"</span>,</span>
<span>            <span class="st">"exposome_var_score"</span></span>
<span>        <span class="op">)</span>,</span>
<span>        filter_col <span class="op">=</span> <span class="st">"p.value"</span>,</span>
<span>        filter_thresh <span class="op">=</span> <span class="fl">0.05</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p><img src="tidyexposomics_files/figure-html/unnamed-chunk-34-1.png" width="480" style="display: block; margin: auto;"></p>
<p>Here we see that most scores are associated with an increased
probability of asthma. The item response theory (IRT) score is the only
one that is significant.</p>
</div>
</div>
<div class="section level2">
<h2 id="differential-abundance">Differential Abundance<a class="anchor" aria-label="anchor" href="#differential-abundance"></a>
</h2>
<hr>
<div class="section level3">
<h3 id="differential-abundance-1">Differential Abundance<a class="anchor" aria-label="anchor" href="#differential-abundance-1"></a>
</h3>
<p>We also provide functionality to assess differentially abundant
features associated with a particular outcome across several omics
layers. The <code>run_differential_abundance</code> function is used to
run the differential abundance analysis. The <code>formula</code>
argument is used to set the formula for the model. The
<code>method</code> argument is used to set the method to use for the
differential abundance analysis. Here we use the
<code>limma_trend</code> method, which is a popular method for
differential abundance analysis. The <code>scaling_method</code>
argument is used to set the scaling method to use. Here we use none, but
we could also use quantile or TMM. The <code>action</code> argument is
set to add to add the results to the existing object.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Run differential abundance analysis</span></span>
<span><span class="va">expom_1</span> <span class="op">&lt;-</span> <span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/run_differential_abundance.html">run_differential_abundance</a></span><span class="op">(</span></span>
<span>        formula <span class="op">=</span> <span class="op">~</span> <span class="va">hs_asthma</span> <span class="op">+</span> <span class="va">hs_child_age_None</span> <span class="op">+</span> <span class="va">e3_sex_None</span> <span class="op">+</span> <span class="va">h_cohort</span>,</span>
<span>        method <span class="op">=</span> <span class="st">"limma_trend"</span>,</span>
<span>        scaling_method <span class="op">=</span> <span class="st">"none"</span>,</span>
<span>        action <span class="op">=</span> <span class="st">"add"</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p>Using a volcano plot, we can visualize the results of the
differential abundance analysis. The <code>plot_volcano</code> function
is used to plot the results. The <code>pval_col</code> argument is used
to set the column containing the p-values. The <code>pval_thresh</code>
argument is used to set the p-value threshold for significance. The
<code>logFC_thresh</code> argument is used to set the log fold change
threshold for significance. The <code>top_n_label</code> argument is
used to set the number of top features to label on the plot. The
<code>nrow</code> argument is used to set the number of rows in the
facet layout. We can use the <code>feature_col</code> to specify which
feature name column to plot. Here we set it to
<code>feature_clean</code> as this column contains the gene/metabolite
name information per feature.</p>
<p><strong>Note:</strong> we set the <code>pval_col</code> to
<code>P.Value</code> for the purposes of this example, but we recommend
keeping the default of <code>adj.P.Val</code> to use the adjusted
p-values.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot Differential Abundance Results</span></span>
<span><span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/plot_volcano.html">plot_volcano</a></span><span class="op">(</span></span>
<span>        top_n_label <span class="op">=</span> <span class="fl">3</span>,</span>
<span>        feature_col <span class="op">=</span> <span class="st">"feature_clean"</span>,</span>
<span>        logFC_thresh <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span>,</span>
<span>        pval_thresh <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>        pval_col <span class="op">=</span> <span class="st">"P.Value"</span>,</span>
<span>        logFC_col <span class="op">=</span> <span class="st">"logFC"</span>,</span>
<span>        nrow <span class="op">=</span> <span class="fl">1</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p><img src="tidyexposomics_files/figure-html/unnamed-chunk-36-1.png" width="672" style="display: block; margin: auto;"></p>
</div>
<div class="section level3">
<h3 id="sensitivity-analysis">Sensitivity Analysis<a class="anchor" aria-label="anchor" href="#sensitivity-analysis"></a>
</h3>
<p>Depending on pre-processing steps, the results of the differential
abundance analysis may vary. The <code>sensitivity_analysis</code>
function is used to perform a sensitivity analysis to determine the
robustness of the results. Here we determine if a feature is still
differentially abundant if different minimum values, proportions,
scaling methods are used, the inclusion of covariates, and after
bootstrapping. We then define a stability score based on the number of
times a feature is found to be differentially abundant under different
conditions as well as the consistency of the effect size:</p>
<p>Where:</p>
<p><span class="math display">\[ Stability\ Score = \frac{\sum_i{(p_i
&lt; \alpha)}}{N} * \frac{1}{1 +
\frac{\sigma_{\beta}}{\mu_{|\beta|}}}\]</span></p>
<p>Where:</p>
<ul>
<li><p><span class="math inline">\(p_i\)</span> is the p-value for the
<span class="math inline">\(i^{th}\)</span> test</p></li>
<li><p><span class="math inline">\(\alpha\)</span> is the significance
threshold</p></li>
<li><p><span class="math inline">\(\beta\)</span> is the log fold
change</p></li>
<li><p><span class="math inline">\(N\)</span> is the number of
tests</p></li>
<li><p><span class="math inline">\(\sigma_{\beta}\)</span> is the
standard deviation of the effect size estimates</p></li>
<li><p><span class="math inline">\(\mu_{|\beta|}\)</span> is the mean of
the absolute value of the effect size estimates.</p></li>
</ul>
<p>The first term captures the proportion of tests that are significant,
while the second term captures the consistency of the effect size
estimates. The stability score ranges from 0 to 1.</p>
<p>A stability score of 1 indicates that the feature is always found to
be differentially abundant, while a stability score of 0 indicates that
the feature is never found to be differentially abundant. Besides these,
we provide other score metrics as well:</p>
<ul>
<li><p><code>presence_rate</code>: Proportion of runs in which the
feature’s p-value is below the specified threshold (selection
frequency).</p></li>
<li><p><code>effect_consistency</code>: Inverse of the coefficient of
variation of log fold-changes; measures effect size stability across
runs.</p></li>
<li><p><code>stability_score</code>: Hybrid score combining
presence_rate and effect_consistency, capturing reproducibility and
signal strength.</p></li>
<li><p><code>mean_log_p</code>: Average of negative log-transformed
p-values; represents overall statistical signal strength.</p></li>
<li><p><code>logp_weighted_score</code>: Product of
<code>mean_log_p</code> and effect_consistency; highlights consistently
strong features.</p></li>
<li><p><code>sd_logFC</code>: Standard deviation of log fold-change
estimates; quantifies variability of effect sizes.</p></li>
<li><p><code>iqr_logFC</code>: Interquartile range of log fold-changes;
provides a robust measure of effect size spread.</p></li>
<li><p><code>cv_logFC</code>: Coefficient of variation of log
fold-changes; reflects relative variability of effect size.</p></li>
<li><p><code>sign_flip_freq</code>: Proportion of runs where the sign of
the effect size differs from the overall average direction.</p></li>
<li><p><code>sd_log_p</code>: Standard deviation of log-transformed
p-values; indicates variability in statistical signal.</p></li>
</ul>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Perform Sensitivity Analysis</span></span>
<span><span class="va">expom_1</span> <span class="op">&lt;-</span> <span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/run_sensitivity_analysis.html">run_sensitivity_analysis</a></span><span class="op">(</span></span>
<span>        base_formula <span class="op">=</span> <span class="op">~</span> <span class="va">hs_asthma</span> <span class="op">+</span> <span class="va">hs_child_age_None</span> <span class="op">+</span> <span class="va">h_cohort</span>,</span>
<span>        methods <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"limma_trend"</span><span class="op">)</span>,</span>
<span>        scaling_methods <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"none"</span><span class="op">)</span>,</span>
<span>        covariates_to_remove <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>            <span class="st">"hs_child_age_None"</span>,</span>
<span>            <span class="st">"e3_sex_None"</span>,</span>
<span>            <span class="st">"h_cohort"</span></span>
<span>        <span class="op">)</span>,</span>
<span>        pval_col <span class="op">=</span> <span class="st">"P.Value"</span>,</span>
<span>        logfc_col <span class="op">=</span> <span class="st">"logFC"</span>,</span>
<span>        logFC_threshold <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span>,</span>
<span>        pval_threshold <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>        stability_metric <span class="op">=</span> <span class="st">"stability_score"</span>,</span>
<span>        bootstrap_n <span class="op">=</span> <span class="fl">10</span>,</span>
<span>        action <span class="op">=</span> <span class="st">"add"</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p>Here we plot the sensitivity analysis results using the
<code>plot_sensitivity_summary</code> function. The
<code>stability_score_thresh</code> argument is used to set the
stability score threshold for significance. Here we use a threshold of
<code>0.31</code>, but this can again be adjusted as needed.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot sensitivity analysis results</span></span>
<span><span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/plot_sensitivity_summary.html">plot_sensitivity_summary</a></span><span class="op">(</span></span>
<span>        stability_score_thresh <span class="op">=</span> <span class="fl">0.31</span>,</span>
<span>        stability_metric <span class="op">=</span> <span class="st">"stability_score"</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p><img src="tidyexposomics_files/figure-html/unnamed-chunk-38-1.png" width="768" style="display: block; margin: auto;"></p>
</div>
</div>
<div class="section level2">
<h2 id="multi-omics-integration">Multi-Omics Integration<a class="anchor" aria-label="anchor" href="#multi-omics-integration"></a>
</h2>
<hr>
<div class="section level3">
<h3 id="multi-omics-integration-1">Multi-Omics Integration<a class="anchor" aria-label="anchor" href="#multi-omics-integration-1"></a>
</h3>
<p>While differential abundance analysis per omics layer can deliver
insights into how each omic is associated with a particular outcome, we
may want to leverage methods which integrate multiple omics layers. The
<code>run_multiomics_integration</code> function is used to integrate
multiple omics layers. Here we use either <code>MCIA</code>,
<code>RGCCA</code>, <code>MOFA</code>, or the <code>DIABLO</code> method
to integrate omics layers:</p>
<ul>
<li><p><code>MCIA</code>: Multiple Co-inertia Analysis, a method that
uses canonical correlation analysis to integrate multiple omics layers.
We use the nipalsMCIA algorithm to compute the co-inertia scores from
the <a href="https://www.bioconductor.org/packages/release/bioc/html/nipalsMCIA.html" class="external-link"><code>nipalsMCIA</code>
package</a> (nipalsMCIA).</p></li>
<li><p><code>RGCCA</code> : Generalized canonical correlation for
flexible multi-block integration implemented using the <a href="https://rgcca-factory.github.io/RGCCA/" class="external-link"><code>RGCCA</code>
package</a> (Regularized and Sparse Generalized Ca…).</p></li>
<li><p><code>MOFA</code>: Multi-Omics Factor Analysis, a method that
uses factor analysis to integrate multiple omics layers. MOFA is
implemented using the <a href="https://www.bioconductor.org/packages/release/bioc/html/MOFA2.html" class="external-link"><code>MOFA2</code>
package</a> (MOFA2).</p></li>
<li><p><code>DIABLO</code>: DIABLO: Supervised multi-block PLS for
outcome-aligned latent factors implemented using the <a href="https://www.bioconductor.org/packages/devel/bioc/html/mixOmics.html" class="external-link"><code>mixOmics</code>
package</a> (mixOmics).</p></li>
</ul>
<p>Here we are interested in integrating our omics layers with the end
goal of identifying multi-omics features that are associated with asthma
status. So, we will use the <code>DIABLO</code> method.</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Perform multi-omics integration</span></span>
<span><span class="va">expom_1</span> <span class="op">&lt;-</span> <span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/run_multiomics_integration.html">run_multiomics_integration</a></span><span class="op">(</span></span>
<span>        method <span class="op">=</span> <span class="st">"DIABLO"</span>,</span>
<span>        n_factors <span class="op">=</span> <span class="fl">5</span>,</span>
<span>        outcome <span class="op">=</span> <span class="st">"hs_asthma"</span>,</span>
<span>        action <span class="op">=</span> <span class="st">"add"</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p>We can then use <code>plot_factor_summary</code> to visualize which
omics contribute most to which factors.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot multi-omics factor summary</span></span>
<span><span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/plot_factor_summary.html">plot_factor_summary</a></span><span class="op">(</span>midpoint <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span></code></pre></div>
<p><img src="tidyexposomics_files/figure-html/unnamed-chunk-40-1.png" width="480" style="display: block; margin: auto;"></p>
<p>Here we see that these factors are largely driven by features in the
methylation and gene expression assays.</p>
</div>
<div class="section level3">
<h3 id="factor-analysis">Factor Analysis<a class="anchor" aria-label="anchor" href="#factor-analysis"></a>
</h3>
<p>These methods are designed to identify factors that we can then
associate with an outcome variable. Here we will use the run_association
function to identify factors that are associated with asthma status
after controlling for child age, sex, and cohort.</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Identify factors that correlate with the outcome</span></span>
<span><span class="va">expom_1</span> <span class="op">&lt;-</span> <span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/run_association.html">run_association</a></span><span class="op">(</span></span>
<span>        source <span class="op">=</span> <span class="st">"factors"</span>,</span>
<span>        outcome <span class="op">=</span> <span class="st">"hs_asthma"</span>,</span>
<span>        feature_set <span class="op">=</span> <span class="va">exp_vars</span>,</span>
<span>        covariates <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>            <span class="st">"hs_child_age_None"</span>,</span>
<span>            <span class="st">"e3_sex_None"</span>,</span>
<span>            <span class="st">"h_cohort"</span></span>
<span>        <span class="op">)</span>,</span>
<span>        action <span class="op">=</span> <span class="st">"add"</span>,</span>
<span>        family <span class="op">=</span> <span class="st">"binomial"</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p>Now let’s see if any of our factors are associated with asthma status
after adjusting for child age, sex, and cohort:</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/plot_association.html">plot_association</a></span><span class="op">(</span></span>
<span>        source <span class="op">=</span> <span class="st">"factors"</span>,</span>
<span>        subtitle <span class="op">=</span> <span class="st">"Covariates: Age, Sex, Cohort"</span>,</span>
<span>        filter_col <span class="op">=</span> <span class="st">"p_adjust"</span>,</span>
<span>        filter_thresh <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>        r2_col <span class="op">=</span> <span class="st">"adj_r2"</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p><img src="tidyexposomics_files/figure-html/unnamed-chunk-42-1.png" width="480"></p>
<p>We see that several factors are associated with our outcome. Factors
have loading scores which indicate the strength of the association
between the factor and the features. Here we can extract the top
features, those in the 90th percentile, per omic, associated with our
factors of interest. We set the <code>pval_col</code> and
<code>pval_thresh</code> to filter our association results to grab the
factors that pass our thresholds. However, we could specify specific
factors with the <code>factors</code> argument too.</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Extract top features that contribute to a factor</span></span>
<span><span class="va">expom_1</span> <span class="op">&lt;-</span> <span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/extract_top_factor_features.html">extract_top_factor_features</a></span><span class="op">(</span></span>
<span>        method <span class="op">=</span> <span class="st">"percentile"</span>,</span>
<span>        pval_col <span class="op">=</span> <span class="st">"p_adjust"</span>,</span>
<span>        pval_thresh <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>        percentile <span class="op">=</span> <span class="fl">0.95</span>,</span>
<span>        action <span class="op">=</span> <span class="st">"add"</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p>We can visualize the top features associated with each factor using
the <code>plot_top_factor_features</code> function. The
<code>top_n</code> argument is used to set the number of top features to
plot. The factors argument is used to set the factors to plot.</p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot top factor features</span></span>
<span><span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/plot_top_factor_features.html">plot_top_factor_features</a></span><span class="op">(</span></span>
<span>        top_n <span class="op">=</span> <span class="fl">15</span>,</span>
<span>        factors <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>            <span class="st">"Methylation comp3"</span>,</span>
<span>            <span class="st">"Gene Expression comp3"</span>,</span>
<span>            <span class="st">"Methylation comp1"</span></span>
<span>        <span class="op">)</span>,</span>
<span>        feature_col <span class="op">=</span> <span class="st">"feature_clean"</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p><img src="tidyexposomics_files/figure-html/unnamed-chunk-44-1.png" width="576" style="display: block; margin: auto;"></p>
<p>To identify the common features across factors, we can use the
<code>run_factor_overlap</code> function. This function will identify
the features that are shared across the top factor features.</p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Determine which features drive multiple factors</span></span>
<span><span class="va">expom_1</span> <span class="op">&lt;-</span> <span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/run_factor_overlap.html">run_factor_overlap</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>Here we see that there are 703 features that are shared across the
factors.</p>
</div>
</div>
<div class="section level2">
<h2 id="exposure-omics-association">Exposure-Omics Association<a class="anchor" aria-label="anchor" href="#exposure-omics-association"></a>
</h2>
<hr>
<div class="section level3">
<h3 id="exposure-omics-association-1">Exposure-Omics Association<a class="anchor" aria-label="anchor" href="#exposure-omics-association-1"></a>
</h3>
<p>Now we have the option to correlate either the top factor features,
differentially abundant features, or user-specified omics features (by
using a variable map, a data frame with two columns, exp_name for the
name of the omics assay, and variable for the name of the molecular
feature) with exposures.</p>
<p>Here we will correlate features driving multiple latent factors with
exposures. To grab these features, we will grab them from the metadata
in our <code>MultiAssayExperiment</code> object. The
<code>correlation_cutoff</code> is used to set the minimum correlation
score, while the <code>pval_cutoff</code> is used to set the maximum
p-value for the association.</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Grab top common factor features and ensure</span></span>
<span><span class="co"># feature is renamed to variable for the variable_map</span></span>
<span><span class="va">top_factor_features</span> <span class="op">&lt;-</span> <span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/extract_results.html">extract_results</a></span><span class="op">(</span>result <span class="op">=</span> <span class="st">"multiomics_integration"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://purrr.tidyverse.org/reference/pluck.html" class="external-link">pluck</a></span><span class="op">(</span><span class="st">"common_top_factor_features"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span></span>
<span>        variable <span class="op">=</span> <span class="va">feature</span>,</span>
<span>        <span class="va">exp_name</span></span>
<span>    <span class="op">)</span></span>
<span></span>
<span><span class="co"># Correlate top factor features with exposures</span></span>
<span><span class="va">expom_1</span> <span class="op">&lt;-</span> <span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="co"># Perform correlation analysis between factor features</span></span>
<span>    <span class="co"># and exposures</span></span>
<span>    <span class="fu"><a href="../reference/run_correlation.html">run_correlation</a></span><span class="op">(</span></span>
<span>        feature_type <span class="op">=</span> <span class="st">"omics"</span>,</span>
<span>        variable_map <span class="op">=</span> <span class="va">top_factor_features</span>,</span>
<span>        exposure_cols <span class="op">=</span> <span class="va">exp_vars</span>,</span>
<span>        action <span class="op">=</span> <span class="st">"add"</span>,</span>
<span>        correlation_cutoff <span class="op">=</span> <span class="fl">0.2</span>,</span>
<span>        pval_cutoff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>        cor_pval_column <span class="op">=</span> <span class="st">"p.value"</span></span>
<span>    <span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="co"># Perform correlation analysis between factor features</span></span>
<span>    <span class="fu"><a href="../reference/run_correlation.html">run_correlation</a></span><span class="op">(</span></span>
<span>        feature_type <span class="op">=</span> <span class="st">"omics"</span>,</span>
<span>        variable_map <span class="op">=</span> <span class="va">top_factor_features</span>,</span>
<span>        feature_cors <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>        action <span class="op">=</span> <span class="st">"add"</span>,</span>
<span>        correlation_cutoff <span class="op">=</span> <span class="fl">0.2</span>,</span>
<span>        pval_cutoff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>        cor_pval_column <span class="op">=</span> <span class="st">"p.value"</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p>We can plot the results of the exposure-omics association analysis
using the <code>plot_correlation_summary</code> function. Here we set
the mode to <code>summary</code>, which will plot the number of
associations per exposure and feature type. The
<code>feature_type</code> argument is used to set the type of features
to plot. Here we use <code>omics</code>, which are the top factor
features.</p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/plot_correlation_summary.html">plot_correlation_summary</a></span><span class="op">(</span></span>
<span>        mode <span class="op">=</span> <span class="st">"summary"</span>,</span>
<span>        feature_type <span class="op">=</span> <span class="st">"omics"</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p><img src="tidyexposomics_files/figure-html/unnamed-chunk-47-1.png" width="1248" style="display: block; margin: auto;"></p>
<p>Here we note that gene expression features have a greater number of
associations with exposures. Among exposures, we see that aerosols have
the greatest number of associations with features, which may reflect the
larger number of aerosol variables.</p>
<p>It may also be useful to identify which exposures are correlated with
similar molecular features. We can do this with the
<code>plot_circos_correlation</code> function. This function will plot a
circos plot of the exposures and their shared features. The
<code>feature_type</code> argument is used to set the correlation
results to use for the analysis. Here we use the <code>omics</code>
feature set, which plots the feature-exposure correlation information.
The cutoff argument is used to set the minimum number of shared features
to plot. Here we set the <code>shared_cutoff</code> to 1, which means
that only exposures that share at least one feature will be plotted.</p>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot Shared Feature Correlations Between Exposures</span></span>
<span><span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/plot_circos_correlation.html">plot_circos_correlation</a></span><span class="op">(</span></span>
<span>        feature_type <span class="op">=</span> <span class="st">"omics"</span>,</span>
<span>        shared_cutoff <span class="op">=</span> <span class="fl">1</span>,</span>
<span>        midpoint <span class="op">=</span> <span class="fl">3</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p><img src="tidyexposomics_files/figure-html/unnamed-chunk-48-1.png" width="960"></p>
<p>Here we see that particulate matter exposures are correlated with
many of the same features. However, we also see that maternal methyl
paraben exposure (<code>hs_mepa_madj_Log2</code>) is correlated with
many of the same features, suggesting that this exposure may have
similar effects on the molecular features.</p>
</div>
<div class="section level3">
<h3 id="network-analysis">Network Analysis<a class="anchor" aria-label="anchor" href="#network-analysis"></a>
</h3>
<p>The <code>run_create_network</code> function is used to create a
network of exposures and omics features. The correlation results are
used to create the network. The <code>feature_type</code> argument is
used to set the type of features to use for the network. We could choose
any of the following:</p>
<ul>
<li><p><code>degs</code>: Differentially abundant features correlated
with exposures.</p></li>
<li><p><code>factors</code>: Factor features correlated with
exposures.</p></li>
<li><p><code>omics</code>: User specified omics features correlated with
exposures.</p></li>
</ul>
<p>Here we create networks based on two correlation tables generated
above:</p>
<ul>
<li><p><code>omics</code>: Correlations between omics and exposure
variables.</p></li>
<li><p><code>omics_feature_cor</code>: Correlation just between the
omics features.</p></li>
</ul>
<p>This allows us to quantify a bipartite graph between exposures and
omics features and a graph just between the omics features.</p>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expom_1</span> <span class="op">&lt;-</span> <span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/run_create_network.html">run_create_network</a></span><span class="op">(</span></span>
<span>        feature_type <span class="op">=</span> <span class="st">"omics_feature_cor"</span>,</span>
<span>        action <span class="op">=</span> <span class="st">"add"</span></span>
<span>    <span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/run_create_network.html">run_create_network</a></span><span class="op">(</span></span>
<span>        feature_type <span class="op">=</span> <span class="st">"omics"</span>,</span>
<span>        action <span class="op">=</span> <span class="st">"add"</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p>To plot the network we can use the <code>plot_network</code>
function. The network argument is used to set the type of network to
plot. The <code>top_n_nodes</code> argument is used to set the number of
nodes to plot. The <code>node_color_var</code> argument is used to set
the variable to color the nodes by. The <code>label</code> argument is
used to set whether or not to label the nodes. We can label the top n
nodes, where the default is 5 nodes based on centrality. Additionally,
we can choose to label certain nodes using the
<code>nodes_to_label</code> argument. We can also include the network
statistics using the <code>include_stats</code> argument.</p>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/plot_network.html">plot_network</a></span><span class="op">(</span></span>
<span>        network <span class="op">=</span> <span class="st">"omics"</span>,</span>
<span>        top_n_nodes <span class="op">=</span> <span class="fl">100</span>,</span>
<span>        include_stats <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>        cor_thresh <span class="op">=</span> <span class="fl">0.2</span>,</span>
<span>        node_color_var <span class="op">=</span> <span class="st">"group"</span>,</span>
<span>        label <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>        label_top_n <span class="op">=</span> <span class="fl">5</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p><img src="tidyexposomics_files/figure-html/unnamed-chunk-50-1.png" width="960"></p>
<p>Here we note that the network is highly connected, with particulate
matter and methyl paraben exposures having the most connections to omics
features. To briefly explain the metrics mentioned:</p>
<ul>
<li><p><strong>Nodes</strong>: Number of nodes included in the
graph.</p></li>
<li><p><strong>Edges</strong>: Number of connections between nodes in
the graph.</p></li>
<li><p><strong>Components</strong>: Number of disconnected
subgraphs.</p></li>
<li><p><strong>Diameter</strong>: The longest shortest path between any
two nodes.</p></li>
<li><p><strong>Mean Distance</strong>: The average shortest path between
any two nodes.</p></li>
<li><p><strong>Modularity</strong>: A measure of the division in the
graph, where higher modularity indicates that nodes within the same
community are more densely connected compared to nodes in different
communities.</p></li>
</ul>
</div>
<div class="section level3">
<h3 id="exposure-omics-impact-network-analysis">Exposure-Omics Impact Network Analysis<a class="anchor" aria-label="anchor" href="#exposure-omics-impact-network-analysis"></a>
</h3>
<p>The bipartite graph above describes how exposures are associated with
omics features. However, this does not describe if exposures are
associated with features that are more central in the feature-only graph
(i.e. are exposures associated with features that are more central or
peripheral to the graph?). To answer this question, we can use the
run_exposure_impact function to calculate centrality metrics for the
omics features associated with a given exposure. Centrality is computed
on the feature-only graph. Centrality metrics are used to identify
features that are more central to the network and may be more important
in the context of the exposure-omics relationships. The centrality
metrics included are:</p>
<ul>
<li><p><strong>Mean Betweenness Centrality</strong>: Measures how often
a node lies on the shortest path between other nodes where high values
indicate potential intermediates.</p></li>
<li><p><strong>Mean Closeness Centrality</strong>: Measures how close a
node is to all others in the network where high values indicate faster
access to all nodes.</p></li>
<li><p><strong>Mean Degree</strong>: The average number of direct
connections or edges a node has.</p></li>
<li><p><strong>Mean Eigenvector Centrality</strong>: Measures how
well-connected a node is and how well-connected its neighbors are, where
high values suggest influence in a well-connected cluster.</p></li>
</ul>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Run exposure-omics impact analysis</span></span>
<span><span class="va">expom_1</span> <span class="op">&lt;-</span> <span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/run_exposure_impact.html">run_exposure_impact</a></span><span class="op">(</span>feature_type <span class="op">=</span> <span class="st">"omics"</span><span class="op">)</span></span></code></pre></div>
<p>To plot the results of the exposure-omics impact analysis, we can use
the <code>plot_exposure_impact</code> function. We set
<code>feature_type</code> to <code>omics</code>, which means that we are
plotting the network metrics of the omics features correlated with
exposures. The <code>min_per_group</code> argument is used to set the
minimum number of features per group to plot. This is useful for
filtering out exposures that do not have enough associated features.</p>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot exposure-omics impact results</span></span>
<span><span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/plot_exposure_impact.html">plot_exposure_impact</a></span><span class="op">(</span></span>
<span>        feature_type <span class="op">=</span> <span class="st">"omics"</span>,</span>
<span>        min_per_group <span class="op">=</span> <span class="fl">10</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p><img src="tidyexposomics_files/figure-html/unnamed-chunk-52-1.png" width="576" style="display: block; margin: auto;"></p>
<p>Here we see that maternal triclosan (<code>hs_trcs_madj_Log2</code>)
exposure is associated with omics features that are central to our
network, despite not having the greatest number of features associated
with it. This suggests that maternal triclosan exposure may have a
significant impact on the omics features, even if it is not associated
with as many features as some of the other exposures.</p>
</div>
</div>
<div class="section level2">
<h2 id="enrichment-analysis">Enrichment Analysis<a class="anchor" aria-label="anchor" href="#enrichment-analysis"></a>
</h2>
<hr>
<p>The benefit of grouping our exposures into categories is that we can
perform enrichment analysis on the associated features to determine how
broad categories of exposures are associated with features. The
<code>run_enrichment</code> function is used to perform enrichment
analysis. Here we could enrich the following feature types:</p>
<ul>
<li><p><code>degs</code>: Differentially abundant features.</p></li>
<li><p><code>degs_robust</code>: Robust differentially abundant features
from the sensitivity analysis.</p></li>
<li><p><code>omics</code>: User chosen features.</p></li>
<li><p><code>factor_features</code>: Multi-omics factor features either
from <code>factor_type = “common_top_factor_features”</code> or
<code>“top_factor_features”</code>.</p></li>
<li><p><code>degs_cor</code>: Differentially abundant features
correlated with a set of exposures.</p></li>
<li><p><code>omics_cor</code>: User chosen features correlated with a
set of exposures.</p></li>
<li><p><code>factor_features_cor</code>: Multi-omics factor features
correlated with a set of exposures.</p></li>
</ul>
<p>Here we will run enrichment analysis on our chosen factor features
correlated with exposures. We specify <code>feature_col</code> to
represent the column in our feature metadata with IDs that can be mapped
(i.e. gene names). We will be performing Gene ontology enrichment
powered by the <a href="https://www.bioconductor.org/packages/release/bioc/html/fenr.html" class="external-link"><code>fenr</code>
package</a> (fenr ). Note that we specify a
<code>clustering_approach.</code> This will cluster our enrichment terms
by the molecular feature overlap.</p>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Run enrichment analysis on factor features correlated with exposures</span></span>
<span><span class="va">expom_1</span> <span class="op">&lt;-</span> <span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/run_enrichment.html">run_enrichment</a></span><span class="op">(</span></span>
<span>        feature_type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"omics_cor"</span><span class="op">)</span>,</span>
<span>        feature_col <span class="op">=</span> <span class="st">"feature_clean"</span>,</span>
<span>        db <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GO"</span><span class="op">)</span>,</span>
<span>        species <span class="op">=</span> <span class="st">"goa_human"</span>,</span>
<span>        fenr_col <span class="op">=</span> <span class="st">"gene_symbol"</span>,</span>
<span>        padj_method <span class="op">=</span> <span class="st">"none"</span>,</span>
<span>        pval_thresh <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>        min_set <span class="op">=</span> <span class="fl">1</span>,</span>
<span>        max_set <span class="op">=</span> <span class="fl">800</span>,</span>
<span>        clustering_approach <span class="op">=</span> <span class="st">"diana"</span>,</span>
<span>        action <span class="op">=</span> <span class="st">"add"</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="enrichment-visualizations">Enrichment Visualizations<a class="anchor" aria-label="anchor" href="#enrichment-visualizations"></a>
</h3>
<p>To visualize our enrichment results we provide several options:</p>
<ul>
<li><p>dot`plot: A dot plot showing the top enriched terms. The size of
the dots represents the number of features associated with the term,
while the color represents the significance of the term.</p></li>
<li><p><code>cnet</code>: A network plot showing the relationship
between features and enriched terms.</p></li>
<li><p><code>network</code>: A network plot showing the relationship
between enriched terms.</p></li>
<li><p><code>heatmap</code>: A heatmap showing the relationship between
features and enriched terms.</p></li>
<li><p><code>summary</code>: A summary figure of the enrichment
results.</p></li>
</ul>
<div class="section level4">
<h4 id="enrichment-summary">Enrichment Summary<a class="anchor" aria-label="anchor" href="#enrichment-summary"></a>
</h4>
<p>To summarize the enrichment results, we can use the
<code>plot_enrichment</code> function with the <code>plot_type</code>
argument set to <code>summary.</code> This will plot a summary of the
enrichment results, showing:</p>
<ul>
<li><p>The number of exposure categories per enrichment term
group.</p></li>
<li><p>The number of features driving the enrichment term
group.</p></li>
<li><p>A p-value distribution of the enrichment term group.</p></li>
<li><p>The number of terms in the enrichment term group.</p></li>
<li><p>The total number of terms per experiment name.</p></li>
<li><p>The overlap in enrichment terms between experiments (i.e. between
gene expression and methylation).</p></li>
</ul>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/plot_enrichment.html">plot_enrichment</a></span><span class="op">(</span></span>
<span>        feature_type <span class="op">=</span> <span class="st">"omics_cor"</span>,</span>
<span>        plot_type <span class="op">=</span> <span class="st">"summary"</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p><img src="tidyexposomics_files/figure-html/unnamed-chunk-54-1.png" width="1248"></p>
<p>Here we see that it is just the features associated with “polyatomic
entity” exposures that seem to be enriched. Additionally, there appears
to be no overlap in terms between methylation and gene expression
results.</p>
</div>
<div class="section level4">
<h4 id="dotplot">DotPlot<a class="anchor" aria-label="anchor" href="#dotplot"></a>
</h4>
<p>By setting the <code>plot_type</code> to dotplot we can create a
dotplot to show which omics and which exposure categories are associated
with which terms. By specifying the <code>top_n_genes</code> we can add
the most frequent features in that particular enrichment term group.</p>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/plot_enrichment.html">plot_enrichment</a></span><span class="op">(</span></span>
<span>        feature_type <span class="op">=</span> <span class="st">"omics_cor"</span>,</span>
<span>        plot_type <span class="op">=</span> <span class="st">"dotplot"</span>,</span>
<span>        top_n <span class="op">=</span> <span class="fl">15</span>,</span>
<span>        add_top_genes <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>        top_n_genes <span class="op">=</span> <span class="fl">5</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p><img src="tidyexposomics_files/figure-html/unnamed-chunk-55-1.png" width="1056"></p>
</div>
<div class="section level4">
<h4 id="term-network-plot">Term Network Plot<a class="anchor" aria-label="anchor" href="#term-network-plot"></a>
</h4>
<p>We can set the <code>plot_type</code> to <code>network</code> to
understand how our enrichment terms are individually connected.</p>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/plot_enrichment.html">plot_enrichment</a></span><span class="op">(</span></span>
<span>        feature_type <span class="op">=</span> <span class="st">"omics_cor"</span>,</span>
<span>        plot_type <span class="op">=</span> <span class="st">"network"</span>,</span>
<span>        label_top_n <span class="op">=</span> <span class="fl">2</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p><img src="tidyexposomics_files/figure-html/unnamed-chunk-56-1.png" width="576"></p>
<p>At the individual term level, we see that they differ by omics layer,
with the gene expression driving terms related to protein binding and
the methylation data driving terms related to lipid metabolism.</p>
</div>
<div class="section level4">
<h4 id="heatmap">Heatmap<a class="anchor" aria-label="anchor" href="#heatmap"></a>
</h4>
<p>Setting the <code>plot_type</code> to <code>heatmap</code> can help
us understand which genes are driving the enrichment terms. We have the
additional benefit of being able to color our tiles by the Log_2_Fold
Change from our differential abundance testing. Here we will examine
group 5, given it seems to be driven by the most terms and multiple
omics layers.</p>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/plot_enrichment.html">plot_enrichment</a></span><span class="op">(</span></span>
<span>        feature_type <span class="op">=</span> <span class="st">"omics_cor"</span>,</span>
<span>        go_groups <span class="op">=</span> <span class="st">"Group_5"</span>,</span>
<span>        plot_type <span class="op">=</span> <span class="st">"heatmap"</span>,</span>
<span>        heatmap_fill <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>        feature_col <span class="op">=</span> <span class="st">"feature_clean"</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p><img src="tidyexposomics_files/figure-html/unnamed-chunk-57-1.png" width="768"></p>
<p>Here we see that methylation features have the strongest fold change
with respect to asthma status. Interestingly, we see that methylation of
<em>CYP2E1</em>, an enzyme responsible for toxicant metabolism, is
downregulated in asthma and correlated with exposures in the polyatomic
entity category containing chemicals like triclosan, BPA and several
PFAS.</p>
</div>
<div class="section level4">
<h4 id="cnet-plot">Cnet Plot<a class="anchor" aria-label="anchor" href="#cnet-plot"></a>
</h4>
<p>Another way to visualize this information is with the
<code>cnet</code> plot, where the enrichment terms are connected to the
genes driving them.</p>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/plot_enrichment.html">plot_enrichment</a></span><span class="op">(</span></span>
<span>        feature_type <span class="op">=</span> <span class="st">"omics_cor"</span>,</span>
<span>        go_groups <span class="op">=</span> <span class="st">"Group_5"</span>,</span>
<span>        plot_type <span class="op">=</span> <span class="st">"cnet"</span>,</span>
<span>        feature_col <span class="op">=</span> <span class="st">"feature_clean"</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p><img src="tidyexposomics_files/figure-html/unnamed-chunk-58-1.png" width="576"></p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="custom-analysis">Custom Analysis<a class="anchor" aria-label="anchor" href="#custom-analysis"></a>
</h2>
<p>We provide functionality to access the underlying data in the
MultiAssayExperiment object and to construct tibbles for your own
analysis:</p>
<ul>
<li><p><code>pivot_sample</code>: Pivot the sample data to a tibble with
samples as rows and exposures as columns.</p></li>
<li><p><code>pivot_feature</code>: Pivot the feature metadata to a
tibble with features as rows and feature metadata as columns.</p></li>
<li><p><code>pivot_exp</code>: Pivot the sample and experiment assay
data to a tibble with samples as rows and sample metadata as columns.
Additionally, there will be a column for values for specified features
in specified assays.</p></li>
</ul>
<div class="section level3">
<h3 id="pivot-sample">Pivot Sample<a class="anchor" aria-label="anchor" href="#pivot-sample"></a>
</h3>
<p>Let’s check out the <code>pivot_sample</code> function. This function
pivots the sample data to a tibble with samples as rows and exposures as
columns.</p>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Pivot sample data to a tibble</span></span>
<span><span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/pivot_sample.html">pivot_sample</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 6 × 357</span></span></span>
<span><span class="co">##   .sample h_abs_ratio_preg_Log h_no2_ratio_preg_Log hs_no2_dy_hs_h_Log</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>                <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> s224                 1.01                    3.21               3.32</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> s873                -<span style="color: #BB0000;">0.005</span><span style="color: #BB0000; text-decoration: underline;">13</span>                 2.48               3.32</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> s846                 0.841                   2.90               3.36</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">4</span> s1019               -<span style="color: #BB0000;">0.066</span><span style="color: #BB0000; text-decoration: underline;">3</span>                  2.89               3.08</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">5</span> s1099               -<span style="color: #BB0000;">0.137</span>                   2.13               2.65</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">6</span> s725                 0.354                   3.18               2.69</span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 353 more variables: hs_no2_wk_hs_h_Log &lt;dbl&gt;, hs_no2_yr_hs_h_Log &lt;dbl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   hs_pm25abs_dy_hs_h_Log &lt;dbl&gt;, hs_pm25abs_wk_hs_h_Log &lt;dbl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   hs_pm25abs_yr_hs_h_Log &lt;dbl&gt;, h_accesslines300_preg_dic0 &lt;dbl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   h_accesspoints300_preg_Log &lt;dbl&gt;, h_builtdens300_preg_Sqrt &lt;dbl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   h_connind300_preg_Sqrt &lt;dbl&gt;, h_fdensity300_preg_Log &lt;dbl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   h_frichness300_preg_None &lt;dbl&gt;, h_landuseshan300_preg_None &lt;dbl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   h_popdens_preg_Sqrt &lt;dbl&gt;, h_walkability_mean_preg_None &lt;dbl&gt;, …</span></span></span></code></pre>
<p>We could use this functionality to count the number of asthmatics per
sex:</p>
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/pivot_sample.html">pivot_sample</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">hs_asthma</span>, <span class="va">e3_sex_None</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/context.html" class="external-link">n</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 4 × 3</span></span></span>
<span><span class="co">## <span style="color: #949494;"># Groups:   hs_asthma [2]</span></span></span>
<span><span class="co">##   hs_asthma e3_sex_None     n</span></span>
<span><span class="co">##       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span>         0 female         67</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span>         0 male           52</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span>         1 female         10</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">4</span>         1 male           10</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="pivot-feature">Pivot Feature<a class="anchor" aria-label="anchor" href="#pivot-feature"></a>
</h3>
<p>The <code>pivot_feature</code> function pivots the feature metadata
to a tibble with features as rows and feature metadata as columns. This
can be useful for exploring the feature metadata in a more flexible
way.</p>
<div class="sourceCode" id="cb61"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Pivot feature data to a tibble</span></span>
<span><span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/pivot_feature.html">pivot_feature</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 6 × 44</span></span></span>
<span><span class="co">##   .exp_name     .feature transcript_cluster_id probeset_id seqname strand  start</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> Gene Express… TC01000… TC01000001.hg.1       TC01000001… chr1    +       <span style="text-decoration: underline;">11</span>869</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> Gene Express… TC01000… TC01000002.hg.1       TC01000002… chr1    +       <span style="text-decoration: underline;">29</span>554</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> Gene Express… TC01000… TC01000003.hg.1       TC01000003… chr1    +       <span style="text-decoration: underline;">69</span>091</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">4</span> Gene Express… TC01000… TC01000005.hg.1       TC01000005… chr1    +      <span style="text-decoration: underline;">317</span>811</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">5</span> Gene Express… TC01000… TC01000007.hg.1       TC01000007… chr1    +      <span style="text-decoration: underline;">321</span>146</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">6</span> Gene Express… TC01000… TC01000009.hg.1       TC01000009… chr1    +      <span style="text-decoration: underline;">367</span>640</span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 37 more variables: stop &lt;int&gt;, total_probes &lt;int&gt;, gene_assignment &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   mrna_assignment &lt;chr&gt;, notes &lt;chr&gt;, phase &lt;chr&gt;, TC_size &lt;dbl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   TSS_Affy &lt;int&gt;, EntrezeGeneID_Affy &lt;chr&gt;, GeneSymbol_Affy &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   GeneSymbolDB &lt;chr&gt;, GeneSymbolDB2 &lt;chr&gt;, mrna_ID &lt;chr&gt;, mrna_DB &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   mrna_N &lt;int&gt;, notesYN &lt;chr&gt;, geneYN &lt;chr&gt;, genes_N &lt;dbl&gt;, CallRate &lt;dbl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   fil1 &lt;chr&gt;, feature_clean &lt;chr&gt;, Forward_Sequence &lt;chr&gt;, SourceSeq &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   Random_Loci &lt;chr&gt;, Methyl27_Loci &lt;chr&gt;, UCSC_RefGene_Name &lt;chr&gt;, …</span></span></span></code></pre>
<p>We could use this functionality to count the number of features per
omics layer, or to filter features based on their metadata. For example,
we can count the number of features per omics layer:</p>
<div class="sourceCode" id="cb63"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Count the number of features per omic layer</span></span>
<span><span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/pivot_feature.html">pivot_feature</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">.exp_name</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/context.html" class="external-link">n</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 3 × 2</span></span></span>
<span><span class="co">##   .exp_name           n</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> Gene Expression <span style="text-decoration: underline;">28</span>714</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> Metabolomics       99</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> Methylation     <span style="text-decoration: underline;">19</span>591</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="pivot-experiment">Pivot Experiment<a class="anchor" aria-label="anchor" href="#pivot-experiment"></a>
</h3>
<p>Now if we want to grab assay data from a particular experiment, we
can do that with the <code>pivot_exp.</code> Let’s try grabbing assay
values for the <code>cg25330361</code> probe (methylation probe for
<em>CYP2E1</em>) in the <code>Methylation</code> experiment:</p>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Pivot experiment data to a tibble</span></span>
<span><span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/pivot_exp.html">pivot_exp</a></span><span class="op">(</span></span>
<span>        omics_name <span class="op">=</span> <span class="st">"Methylation"</span>,</span>
<span>        features <span class="op">=</span> <span class="st">"cg25330361"</span></span>
<span>    <span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 6 × 375</span></span></span>
<span><span class="co">##   exp_name    .feature  .sample counts h_abs_ratio_preg_Log h_no2_ratio_preg_Log</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>                <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>                <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> Methylation cg253303… s224     0.826              1.01                    3.21</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> Methylation cg253303… s873     1.16              -<span style="color: #BB0000;">0.005</span><span style="color: #BB0000; text-decoration: underline;">13</span>                 2.48</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> Methylation cg253303… s1019    0.974             -<span style="color: #BB0000;">0.066</span><span style="color: #BB0000; text-decoration: underline;">3</span>                  2.89</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">4</span> Methylation cg253303… s725     1.06               0.354                   3.18</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">5</span> Methylation cg253303… s1025    0.656              0.054<span style="text-decoration: underline;">5</span>                  2.76</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">6</span> Methylation cg253303… s242     0.968              0.338                   2.92</span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 369 more variables: hs_no2_dy_hs_h_Log &lt;dbl&gt;, hs_no2_wk_hs_h_Log &lt;dbl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   hs_no2_yr_hs_h_Log &lt;dbl&gt;, hs_pm25abs_dy_hs_h_Log &lt;dbl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   hs_pm25abs_wk_hs_h_Log &lt;dbl&gt;, hs_pm25abs_yr_hs_h_Log &lt;dbl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   h_accesslines300_preg_dic0 &lt;dbl&gt;, h_accesspoints300_preg_Log &lt;dbl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   h_builtdens300_preg_Sqrt &lt;dbl&gt;, h_connind300_preg_Sqrt &lt;dbl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   h_fdensity300_preg_Log &lt;dbl&gt;, h_frichness300_preg_None &lt;dbl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   h_landuseshan300_preg_None &lt;dbl&gt;, h_popdens_preg_Sqrt &lt;dbl&gt;, …</span></span></span></code></pre>
<p>We can use this functionality to create custom plots or analyses
based on the exposure and feature data. For example, we can plot the
methylation levels of <em>CYP2E1</em> by asthma status:</p>
<div class="sourceCode" id="cb67"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot the distribution of a specific feature across samples</span></span>
<span><span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/pivot_exp.html">pivot_exp</a></span><span class="op">(</span></span>
<span>        omics_name <span class="op">=</span> <span class="st">"Methylation"</span>,</span>
<span>        features <span class="op">=</span> <span class="st">"cg25330361"</span></span>
<span>    <span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span>
<span>        x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">hs_asthma</span><span class="op">)</span>,</span>
<span>        y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="va">counts</span> <span class="op">+</span> <span class="fl">1</span><span class="op">)</span>,</span>
<span>        color <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">hs_asthma</span><span class="op">)</span>,</span>
<span>        fill <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">hs_asthma</span><span class="op">)</span></span>
<span>    <span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">geom_boxplot</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_jitter.html" class="external-link">geom_jitter</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggpubr</span><span class="fu">::</span><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/geom_pwc.html" class="external-link">geom_pwc</a></span><span class="op">(</span></span>
<span>        label <span class="op">=</span> <span class="st">"{p.adj.format}{p.adj.signif}"</span></span>
<span>    <span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_minimal</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggpubr</span><span class="fu">::</span><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/rotate_axis_text.html" class="external-link">rotate_x_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggsci</span><span class="fu">::</span><span class="fu"><a href="https://nanx.me/ggsci/reference/scale_cosmic.html" class="external-link">scale_color_cosmic</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggsci</span><span class="fu">::</span><span class="fu"><a href="https://nanx.me/ggsci/reference/scale_cosmic.html" class="external-link">scale_fill_cosmic</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span></span>
<span>        x <span class="op">=</span> <span class="st">""</span>,</span>
<span>        y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/expression.html" class="external-link">expression</a></span><span class="op">(</span><span class="va">Log</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span> <span class="op">*</span> <span class="st">"Abd."</span><span class="op">)</span>,</span>
<span>        fill <span class="op">=</span> <span class="st">"Asthma Status"</span>,</span>
<span>        color <span class="op">=</span> <span class="st">"Asthma Status"</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p><img src="tidyexposomics_files/figure-html/unnamed-chunk-64-1.png" width="384" style="display: block; margin: auto;"></p>
</div>
</div>
<div class="section level2">
<h2 id="pipeline-summary">Pipeline Summary<a class="anchor" aria-label="anchor" href="#pipeline-summary"></a>
</h2>
<p>To summarize the steps we have taken in this analysis, we can use the
<code>run_pipeline_summary</code> function. This function will provide a
summary of the steps taken in the analysis. We can set
<code>console_print</code> to <code>TRUE</code> to print the summary to
the console. Setting <code>include_notes</code> to <code>TRUE</code>
will include notes on the steps taken in the analysis.</p>
<div class="sourceCode" id="cb68"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Run the pipeline summary</span></span>
<span><span class="va">expom_1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/run_pipeline_summary.html">run_pipeline_summary</a></span><span class="op">(</span></span>
<span>        console_print <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>        include_notes <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 1. filter_missing - Filtered exposure variables and omics features with more than 5% missing values. QC plots generated for exposure and omics data.2. run_impute_missing - 3. filter_omics_Methylation - Filtered omics features from 'Methylation'</span></span>
<span><span class="co">##         Using method = 'variance': 40253 removed of 59844 (67.3%).4. filter_omics_Metabolomics - Filtered omics features from 'Metabolomics'</span></span>
<span><span class="co">##         Using method = 'variance': 78 removed of 177 (44.1%).5. filter_omics_Gene Expression - Filtered omics features from 'Gene Expression'</span></span>
<span><span class="co">##         Using method = 'expression': 24 removed of 28738 (0.1%).6. run_normality_check - Assessed normality of 180 numeric exposure variables. 29 were normally distributed (p &gt; 0.05), 151 were not.7. transform_exposure - Applied 'boxcox_best' transformation to 22 exposure variables. 6 passed normality (Shapiro-Wilk p &gt; 0.05, 27.3%).8. run_pca - Outliers:  s73, s1183, s376, s899, s828, s9939. filter_sample_outliers - Outliers:  s73, s1183, s376, s899, s828, s99310. run_correlation_pcs - Correlated pcs features with exposures.11. run_cluster_samples - Optimal number of clusters for samples: 512. run_correlation_exposures - Correlated exposures features with exposures.13. run_association - Performed association analysis using source: exposures14. run_association - Performed association analysis using source: exposures15. run_exposome_score_exposome_median_score - Exposome score computed using method: 'median'16. run_exposome_score_exposome_pca_score - Exposome score computed using method: 'pca'17. run_exposome_score_exposome_irt_score - Exposome score computed using method: 'irt'18. run_exposome_score_exposome_quantile_score - Exposome score computed using method: 'quantile'19. run_exposome_score_exposome_var_score - Exposome score computed using method: 'var'20. run_association - Performed association analysis using source: exposures21. run_differential_abundance - Performed differential abundance analysis across all assays.22. run_sensitivity_analysis - Ran sensitivity analysis across 10 bootstrap iterations and 1 methods and 1 scaling strategies. Covariates removed in model variations: hs_child_age_None, e3_sex_None, h_cohort23. run_multiomics_integration - Performed multi-omics integration using DIABLO with 5 latent factors. Scaling was enabled.24. run_association - Performed association analysis using source: exposures25. extract_top_factor_features - Selected 9664 features contributing to specified factors.26. run_factor_overlap - Annotated based on 'stability_score' score &gt; 0.341.27. run_correlation_omics - Correlated omics features with exposures.28. run_correlation_omics - Correlated omics features with exposures.29. run_create_network - Created undirected network from correlation results for'omics_feature_cor'.30. run_create_network - Created undirected network from correlation results for'omics_feature_cor'.31. run_exposure_impact - Computed exposure impact using omics correlation network.32. run_enrichment - Performed GO enrichment on omics_cor features.</span></span></code></pre>
<div class="section level3">
<h3 id="saving-results">Saving Results<a class="anchor" aria-label="anchor" href="#saving-results"></a>
</h3>
<p>We can export the results in our <code>MultiAssayExperiment</code> to
an Excel spreadsheet using the <code>extract_results_excel</code>
function. Here we add all of our results to the Excel file, but we can
choose certain results by changing the <code>result_types</code>
argument.</p>
<div class="sourceCode" id="cb70"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/extract_results_excel.html">extract_results_excel</a></span><span class="op">(</span><span class="va">expom_1</span>,</span>
<span>    file <span class="op">=</span> <span class="st">"./supplementary_results.xlsx"</span>,</span>
<span>    result_types <span class="op">=</span> <span class="st">"all"</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h3>
<div class="sourceCode" id="cb71"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.4.1 (2024-06-14 ucrt)</span></span>
<span><span class="co">## Platform: x86_64-w64-mingw32/x64</span></span>
<span><span class="co">## Running under: Windows 11 x64 (build 26100)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">## [1] LC_COLLATE=English_United States.utf8 </span></span>
<span><span class="co">## [2] LC_CTYPE=English_United States.utf8   </span></span>
<span><span class="co">## [3] LC_MONETARY=English_United States.utf8</span></span>
<span><span class="co">## [4] LC_NUMERIC=C                          </span></span>
<span><span class="co">## [5] LC_TIME=English_United States.utf8    </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: America/New_York</span></span>
<span><span class="co">## tzcode source: internal</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] tidyexposomics_0.99.0 lubridate_1.9.4       forcats_1.0.0        </span></span>
<span><span class="co">##  [4] stringr_1.5.1         dplyr_1.1.4           purrr_1.1.0          </span></span>
<span><span class="co">##  [7] readr_2.1.5           tidyr_1.3.1           tibble_3.3.0         </span></span>
<span><span class="co">## [10] ggplot2_3.5.2         tidyverse_2.0.0       BiocStyle_2.32.1     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] fs_1.6.6                    matrixStats_1.5.0          </span></span>
<span><span class="co">##   [3] naniar_1.1.0                httr_1.4.7                 </span></span>
<span><span class="co">##   [5] RColorBrewer_1.1-3          doParallel_1.0.17          </span></span>
<span><span class="co">##   [7] ggsci_3.2.0                 dynamicTreeCut_1.63-1      </span></span>
<span><span class="co">##   [9] tools_4.4.1                 doRNG_1.8.6.2              </span></span>
<span><span class="co">##  [11] backports_1.5.0             utf8_1.2.6                 </span></span>
<span><span class="co">##  [13] R6_2.6.1                    vegan_2.7-1                </span></span>
<span><span class="co">##  [15] lazyeval_0.2.2              mgcv_1.9-1                 </span></span>
<span><span class="co">##  [17] permute_0.9-8               GetoptLong_1.0.5           </span></span>
<span><span class="co">##  [19] withr_3.0.2                 gridExtra_2.3              </span></span>
<span><span class="co">##  [21] preprocessCore_1.66.0       progressr_0.15.1           </span></span>
<span><span class="co">##  [23] cli_3.6.5                   Biobase_2.64.0             </span></span>
<span><span class="co">##  [25] textshaping_1.0.1           factoextra_1.0.7           </span></span>
<span><span class="co">##  [27] RGCCA_3.0.3                 labeling_0.4.3             </span></span>
<span><span class="co">##  [29] sass_0.4.10                 randomForest_4.7-1.2       </span></span>
<span><span class="co">##  [31] ggridges_0.5.6              pbapply_1.7-4              </span></span>
<span><span class="co">##  [33] pkgdown_2.1.3               systemfonts_1.2.3          </span></span>
<span><span class="co">##  [35] foreign_0.8-86              R.utils_2.13.0             </span></span>
<span><span class="co">##  [37] dichromat_2.0-0.1           sessioninfo_1.2.3          </span></span>
<span><span class="co">##  [39] parallelly_1.45.1           itertools_0.1-3            </span></span>
<span><span class="co">##  [41] limma_3.60.6                RSQLite_2.4.2              </span></span>
<span><span class="co">##  [43] rstudioapi_0.17.1           FNN_1.1.4.1                </span></span>
<span><span class="co">##  [45] generics_0.1.4              shape_1.4.6.1              </span></span>
<span><span class="co">##  [47] vroom_1.6.5                 car_3.1-3                  </span></span>
<span><span class="co">##  [49] dendextend_1.19.1           Matrix_1.7-0               </span></span>
<span><span class="co">##  [51] S4Vectors_0.42.1            abind_1.4-8                </span></span>
<span><span class="co">##  [53] R.methodsS3_1.8.2           lifecycle_1.0.4            </span></span>
<span><span class="co">##  [55] edgeR_4.2.2                 yaml_2.3.10                </span></span>
<span><span class="co">##  [57] snakecase_0.11.1            carData_3.0-5              </span></span>
<span><span class="co">##  [59] SummarizedExperiment_1.34.0 BiocFileCache_2.12.0       </span></span>
<span><span class="co">##  [61] recipes_1.3.1               SparseArray_1.4.8          </span></span>
<span><span class="co">##  [63] Rtsne_0.17                  blob_1.2.4                 </span></span>
<span><span class="co">##  [65] grid_4.4.1                  crayon_1.5.3               </span></span>
<span><span class="co">##  [67] lattice_0.22-6              magick_2.8.7               </span></span>
<span><span class="co">##  [69] pillar_1.11.0               knitr_1.50                 </span></span>
<span><span class="co">##  [71] ComplexHeatmap_2.20.0       GenomicRanges_1.56.2       </span></span>
<span><span class="co">##  [73] rjson_0.2.23                corpcor_1.6.10             </span></span>
<span><span class="co">##  [75] future.apply_1.20.0         mixOmics_6.28.0            </span></span>
<span><span class="co">##  [77] codetools_0.2-20            glue_1.8.0                 </span></span>
<span><span class="co">##  [79] ggvenn_0.1.10               beepr_2.0                  </span></span>
<span><span class="co">##  [81] data.table_1.17.8           MultiAssayExperiment_1.30.3</span></span>
<span><span class="co">##  [83] vctrs_0.6.5                 png_0.1-8                  </span></span>
<span><span class="co">##  [85] testthat_3.2.3              gtable_0.3.6               </span></span>
<span><span class="co">##  [87] assertthat_0.2.1            cachem_1.1.0               </span></span>
<span><span class="co">##  [89] gower_1.0.2                 xfun_0.52                  </span></span>
<span><span class="co">##  [91] S4Arrays_1.4.1              prodlim_2025.04.28         </span></span>
<span><span class="co">##  [93] tidygraph_1.3.1             pracma_2.4.4               </span></span>
<span><span class="co">##  [95] survival_3.6-4              audio_0.1-11               </span></span>
<span><span class="co">##  [97] timeDate_4041.110           iterators_1.0.14           </span></span>
<span><span class="co">##  [99] nipalsMCIA_1.2.1            hardhat_1.4.1              </span></span>
<span><span class="co">## [101] lava_1.8.1                  statmod_1.5.0              </span></span>
<span><span class="co">## [103] ipred_0.9-15                nlme_3.1-164               </span></span>
<span><span class="co">## [105] fenr_1.2.1                  bit64_4.6.0-1              </span></span>
<span><span class="co">## [107] filelock_1.0.3              GenomeInfoDb_1.40.1        </span></span>
<span><span class="co">## [109] bslib_0.9.0                 Deriv_4.2.0                </span></span>
<span><span class="co">## [111] rpart_4.1.23                DBI_1.2.3                  </span></span>
<span><span class="co">## [113] colorspace_2.1-1            BiocGenerics_0.50.0        </span></span>
<span><span class="co">## [115] Hmisc_5.2-3                 nnet_7.3-19                </span></span>
<span><span class="co">## [117] tidyselect_1.2.1            bit_4.6.0                  </span></span>
<span><span class="co">## [119] compiler_4.4.1              curl_6.4.0                 </span></span>
<span><span class="co">## [121] tidyHeatmap_1.12.2          rvest_1.0.4                </span></span>
<span><span class="co">## [123] httr2_1.2.1                 htmlTable_2.4.3            </span></span>
<span><span class="co">## [125] xml2_1.3.8                  desc_1.4.3                 </span></span>
<span><span class="co">## [127] DelayedArray_0.30.1         plotly_4.11.0              </span></span>
<span><span class="co">## [129] bookdown_0.43               checkmate_2.3.2            </span></span>
<span><span class="co">## [131] scales_1.4.0                rappdirs_0.3.3             </span></span>
<span><span class="co">## [133] digest_0.6.37               rmarkdown_2.29             </span></span>
<span><span class="co">## [135] XVector_0.44.0              htmltools_0.5.8.1          </span></span>
<span><span class="co">## [137] pkgconfig_2.0.3             base64enc_0.1-3            </span></span>
<span><span class="co">## [139] SimDesign_2.20.0            MatrixGenerics_1.16.0      </span></span>
<span><span class="co">## [141] dbplyr_2.5.0                fastmap_1.2.0              </span></span>
<span><span class="co">## [143] rlang_1.1.6                 GlobalOptions_0.1.2        </span></span>
<span><span class="co">## [145] htmlwidgets_1.6.4           UCSC.utils_1.0.0           </span></span>
<span><span class="co">## [147] ggh4x_0.3.1                 farver_2.1.2               </span></span>
<span><span class="co">## [149] jquerylib_0.1.4             jsonlite_2.0.0             </span></span>
<span><span class="co">## [151] BiocParallel_1.38.0         dcurver_0.9.2              </span></span>
<span><span class="co">## [153] R.oo_1.27.1                 ModelMetrics_1.2.2.2       </span></span>
<span><span class="co">## [155] magrittr_2.0.3              Formula_1.2-5              </span></span>
<span><span class="co">## [157] GenomeInfoDbData_1.2.12     patchwork_1.3.1            </span></span>
<span><span class="co">## [159] Rcpp_1.1.0                  ggnewscale_0.5.2           </span></span>
<span><span class="co">## [161] viridis_0.6.5               visdat_0.6.0               </span></span>
<span><span class="co">## [163] stringi_1.8.7               pROC_1.18.5                </span></span>
<span><span class="co">## [165] ggraph_2.2.1                brio_1.1.5                 </span></span>
<span><span class="co">## [167] zlibbioc_1.50.0             MASS_7.3-60.2              </span></span>
<span><span class="co">## [169] plyr_1.8.9                  parallel_4.4.1             </span></span>
<span><span class="co">## [171] listenv_0.9.1               ggrepel_0.9.6              </span></span>
<span><span class="co">## [173] graphlayouts_1.2.2          splines_4.4.1              </span></span>
<span><span class="co">## [175] hms_1.1.3                   circlize_0.4.16            </span></span>
<span><span class="co">## [177] locfit_1.5-9.12             igraph_2.1.4               </span></span>
<span><span class="co">## [179] ggpubr_0.6.1                ggsignif_0.6.4             </span></span>
<span><span class="co">## [181] rngtools_1.5.2              reshape2_1.4.4             </span></span>
<span><span class="co">## [183] stats4_4.4.1                GPArotation_2025.3-1       </span></span>
<span><span class="co">## [185] tidybulk_1.16.0             evaluate_1.0.4             </span></span>
<span><span class="co">## [187] ttservice_0.5.3             BiocManager_1.30.26        </span></span>
<span><span class="co">## [189] tzdb_0.5.0                  foreach_1.5.2              </span></span>
<span><span class="co">## [191] missForest_1.5              tweenr_2.0.3               </span></span>
<span><span class="co">## [193] polyclip_1.10-7             future_1.58.0              </span></span>
<span><span class="co">## [195] clue_0.3-66                 mirt_1.44.0                </span></span>
<span><span class="co">## [197] BiocBaseUtils_1.6.0         ggforce_0.5.0              </span></span>
<span><span class="co">## [199] janitor_2.2.1               broom_1.0.9                </span></span>
<span><span class="co">## [201] RSpectra_0.16-2             rstatix_0.7.2              </span></span>
<span><span class="co">## [203] viridisLite_0.4.2           class_7.3-22               </span></span>
<span><span class="co">## [205] ragg_1.4.0                  rARPACK_0.11-0             </span></span>
<span><span class="co">## [207] memoise_2.0.1               ellipse_0.5.0              </span></span>
<span><span class="co">## [209] IRanges_2.38.1              densityClust_0.3.3         </span></span>
<span><span class="co">## [211] cluster_2.1.6               timechange_0.3.0           </span></span>
<span><span class="co">## [213] globals_0.18.0              caret_7.0-1</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h3>
<p>Chung, M. K., House, J. S., Akhtari, F. S., Makris, K. C., Langston,
M. A., Islam, K. T., Holmes, P., Chadeau-Hyam, M., Smirnov, A. I., Du,
X., Thessen, A. E., Cui, Y., Zhang, K., Manrai, A. K., Motsinger-Reif,
A., Patel, C. J., &amp; Members of the Exposomics Consortium. (2024).
Decoding the exposome: data science methodologies and implications in
exposome-wide association studies (ExWASs). <em>Exposome, 4</em>(1),
osae001. <a href="https://doi.org/10.1093/exposome/osae001" class="external-link uri">https://doi.org/10.1093/exposome/osae001</a></p>
<p>fenr. (n.d.). Bioconductor. Retrieved August 18, 2025, from <a href="https://www.bioconductor.org/packages/release/bioc/html/fenr.html" class="external-link uri">https://www.bioconductor.org/packages/release/bioc/html/fenr.html</a></p>
<p>Maitre, L., Guimbaud, J.-B., Warembourg, C., Güil-Oumrait, N.,
Petrone, P. M., Chadeau-Hyam, M., Vrijheid, M., Basagaña, X., Gonzalez,
J. R., &amp; Exposome Data Challenge Participant Consortium. (2022).
State-of-the-art methods for exposure-health studies: Results from the
exposome data challenge event. <em>Environment International,
168</em>(107422), 107422. <a href="https://doi.org/10.1016/j.envint.2022.107422" class="external-link uri">https://doi.org/10.1016/j.envint.2022.107422</a></p>
<p>Miller, G. W., &amp; Banbury Exposomics Consortium. (2025).
Integrating exposomics into biomedicine. <em>Science, 388</em>(6745),
356–358. <a href="https://doi.org/10.1126/science.adr0544" class="external-link uri">https://doi.org/10.1126/science.adr0544</a></p>
<p>mixOmics. (n.d.). Bioconductor. Retrieved August 18, 2025, from <a href="https://www.bioconductor.org/packages/devel/bioc/html/mixOmics.html" class="external-link uri">https://www.bioconductor.org/packages/devel/bioc/html/mixOmics.html</a></p>
<p>MOFA2. (n.d.). Bioconductor. Retrieved August 18, 2025, from <a href="https://www.bioconductor.org/packages/release/bioc/html/MOFA2.html" class="external-link uri">https://www.bioconductor.org/packages/release/bioc/html/MOFA2.html</a></p>
<p>MultiAssay Special Interest Group. (2025, April 15).
<em>MultiAssayExperiment: The Integrative Bioconductor Container.</em>
<a href="https://www.bioconductor.org/packages/release/bioc/vignettes/MultiAssayExperiment/inst/doc/MultiAssayExperiment.html" class="external-link uri">https://www.bioconductor.org/packages/release/bioc/vignettes/MultiAssayExperiment/inst/doc/MultiAssayExperiment.html</a></p>
<p>nipalsMCIA. (n.d.). Bioconductor. Retrieved August 18, 2025, from <a href="https://www.bioconductor.org/packages/release/bioc/html/nipalsMCIA.html" class="external-link uri">https://www.bioconductor.org/packages/release/bioc/html/nipalsMCIA.html</a></p>
<p>Regularized and Sparse Generalized Canonical Correlation Analysis for
Multiblock Data. (n.d.). Retrieved August 18, 2025, from <a href="https://rgcca-factory.github.io/RGCCA/" class="external-link uri">https://rgcca-factory.github.io/RGCCA/</a></p>
<p>Wild, C. P. (2005). Complementing the genome with an “exposome”: the
outstanding challenge of environmental exposure measurement in molecular
epidemiology. <em>Cancer Epidemiology, Biomarkers &amp; Prevention,
14</em>(8), 1847–1850. <a href="https://doi.org/10.1158/1055-9965.EPI-05-0456" class="external-link uri">https://doi.org/10.1158/1055-9965.EPI-05-0456</a></p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Jason Laird, JHU Discovery Award.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
