<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Introduction to tidyexposomics • tidyexposomics</title>
<!-- mathjax math --><script src="https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/tex-mml-chtml.js" integrity="sha256-MASABpB4tYktI2Oitl4t+78w/lyA+D7b/s9GEP0JOGI=" crossorigin="anonymous"></script><script>
  window.MathJax = {
    chtml: {
      fontURL: "https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/output/chtml/fonts/woff-v2"
    }
  };
</script><script src="../lightswitch.js"></script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/Atkinson_Hyperlegible-0.4.10/font.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Introduction to tidyexposomics">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">tidyexposomics</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/tidyexposomics.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/faq.html">FAQ</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch">
<li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul>
</li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">



<link href="tidyexposomics_files/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet">
<script src="tidyexposomics_files/htmlwidgets-1.6.4/htmlwidgets.js"></script><script src="tidyexposomics_files/d3-3.3.8/d3.min.js"></script><script src="tidyexposomics_files/dagre-0.4.0/dagre-d3.min.js"></script><link href="tidyexposomics_files/mermaid-0.3.0/dist/mermaid.css" rel="stylesheet">
<script src="tidyexposomics_files/mermaid-0.3.0/dist/mermaid.slim.min.js"></script><link href="tidyexposomics_files/DiagrammeR-styles-0.2/styles.css" rel="stylesheet">
<script src="tidyexposomics_files/chromatography-0.1/chromatography.js"></script><script src="tidyexposomics_files/DiagrammeR-binding-1.0.11/DiagrammeR.js"></script><div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Introduction to tidyexposomics</h1>
            
      

      <div class="d-none name"><code>tidyexposomics.Rmd</code></div>
    </div>

    
    
<p>Exposure to environmental factors is a major determinant of health
and disease. The exposome is a term that represents the totality of
environmental exposures that drive disease and overall health <a href="https://pubmed.ncbi.nlm.nih.gov/16103423/" class="external-link">(Wild et al. 2005)</a>.
To tackle the daunting task of characterizing the relationship between a
multiplicity of exposures and disease outcomes, epidemiological studies
have moved towards exposure wide association studies (ExWAS) where sets
of exposures are associated with an outcome <a href="https://academic.oup.com/exposome/article/4/1/osae001/7574628" class="external-link">(Chung
et al. 2024)</a>. The advent of high-throughput technologies has enabled
profiling of several layers of biological information. These layers too
can be integrated to better understand the relationship between
exposures and disease outcomes.</p>
<p>The <code>tidyexposomics</code> package is designed to facilitate the
integration of exposure and omics data to identify exposure-omics
associations. We structure our commands to fit into the
<code>tidyverse</code> framework, where commands are designed to be
simplified and intuitive. Here we provide functionality to perform
quality control, sample and exposure association analysis, differential
abundance analysis, multi-omics integration, and functional enrichment
analysis.</p>
<div class="figure" style="text-align: center">
<img src="overview.png" alt="**tidyexposomics pipeline overview.** Exposure and omics data are stored in a MultiAssayExperiment. Data then undergoes quality control, sample and exposure associations, differential abundance analysis, multi-omics integration, and functional enrichment analysis. The pipeline is designed to be flexible and modular, allowing users to customize their analysis." width="120%"><p class="caption">
<strong>tidyexposomics pipeline overview.</strong> Exposure and omics
data are stored in a MultiAssayExperiment. Data then undergoes quality
control, sample and exposure associations, differential abundance
analysis, multi-omics integration, and functional enrichment analysis.
The pipeline is designed to be flexible and modular, allowing users to
customize their analysis.
</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install the current development version</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"BioNomad/tidyexposomics"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="command-structure">Command Structure<a class="anchor" aria-label="anchor" href="#command-structure"></a>
</h2>
<hr>
<p>To make the package more user-friendly, we have named our functions
to be more intuitive. For example, we use the following naming
conventions:</p>
<p><img src="command_str.png" width="80%" style="display: block; margin: auto;"></p>
<p>We provide functionality to either <code>add</code> results to the
existing object storing the omics/exposure data or to return the direct
results using the <code>get</code> option using the <code>action</code>
argument. We suggest adding results, as we also include a
<code><a href="../reference/run_pipeline_summary.html">run_pipeline_summary()</a></code> function to generate a diagram of
the workflow. This is useful for keeping track of the pipeline
steps.</p>
</div>
<div class="section level2">
<h2 id="exposure-meta-data-and-ontology-annotation">Exposure Meta data and Ontology Annotation<a class="anchor" aria-label="anchor" href="#exposure-meta-data-and-ontology-annotation"></a>
</h2>
<hr>
<p>So before starting an exposomics data analysis we recommend having a
codebook about your exposure variables. Some suggestions:</p>
<ul>
<li><p><strong>Variable Name</strong>: The name of the variable in the
data set.</p></li>
<li><p><strong>Variable Description</strong>: A concise description of
what the variable measures, including units (e.g., “urinary bisphenol A
(ng/mL)”)</p></li>
<li><p><strong>Variable Type</strong>: The type of variable, such as
continuous, categorical, or binary.</p></li>
<li><p><strong>Variable Period</strong>: The period of time over which
the variable was measured, such as “lifetime”, “year”, “month”, or
“day”.</p></li>
<li><p><strong>Variable Location</strong>: The location where the
variable was measured, such as “home”, “work”, or “school”.</p></li>
<li><p><strong>Variable Ontology</strong>: The ontology term associated
with the variable.</p></li>
</ul>
<p>Now the ontology is very important and we recommend:</p>
<ul>
<li><p><a href="https://www.ebi.ac.uk/ols4/ontologies/ecto" class="external-link">Environment
Exposure Ontology</a> to annotate your exposure variables.</p></li>
<li><p><a href="https://www.ebi.ac.uk/ols4/ontologies/hp" class="external-link">Human
Phenotype Ontology</a> to annotate your outcome variables and phenotypic
data.</p></li>
<li><p><a href="https://www.ebi.ac.uk/ols4/ontologies/chebi" class="external-link">Chemical
Entities of Biological Interest</a> to annotate your chemical exposure
variables.</p></li>
</ul>
<p><strong>Now why should you bother doing this?</strong></p>
<ul>
<li><p><strong>Interpretability</strong>: Ontology labels clarify
ambiguous or inconsistently named variables.</p></li>
<li><p><strong>Harmonization</strong>: You can compare and combine
variables across datasets when they map to the same term.</p></li>
<li><p><strong>Grouping</strong>: Ontologies allow you to collapse
fine-grained exposures into broader categories.</p></li>
<li><p><strong>Integration</strong>: Many public tools, knowledge
graphs, and repositories (e.g., EBI OLS, GO, or ChEBI) are
ontology-aware. This can make your results more interoperable and
reusable.</p></li>
</ul>
<p>To help you annotate your exposure variables, we provide a
lightweight shiny app that will help you decide which ontology terms to
use and how to annotate your variable:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Launch the shiny app to annotate exposure variables</span></span>
<span><span class="fu"><a href="../reference/ont_annot_app.html">ont_annot_app</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><strong>Now to use the app:</strong></p>
<ul>
<li><p>Click “Browse” to select your exposure metadata file.</p></li>
<li><p>Then you can click the variable you’d like to link to an
annotation term and search in the “Choose Ontology Term”
dropdown.</p></li>
<li><p>After you select a term you will see a short description of the
the term.</p></li>
<li><p>After your are done click “Apply Annotate” to save the
annotation.</p></li>
<li><p>Now you can group exposures into larger categories by selecting
each line and then your ontology, and root depth level (where a lower
number means a more general term).</p></li>
<li><p>Then you can click “Apply Categorization” to apply the changes to
the selected rows.</p></li>
<li><p>If the ontology has nothing to do with your variable, you may
manually enter a category in the “Category” column. This will change the
“Category Source” to “manual” and will not be linked to the
ontology.</p></li>
<li><p>Once you have annotated all your variables click “Download
Annotated CSV” to save the annotated metadata file.</p></li>
</ul>
<p><img src="ont_annot.png" width="150%" style="display: block; margin: auto;"></p>
</div>
<div class="section level2">
<h2 id="loading-data">Loading Data<a class="anchor" aria-label="anchor" href="#loading-data"></a>
</h2>
<hr>
<p>To get started we need to load the data. The
<code>create_expomicset</code> function is used to create a
MultiAssayExperiment object that contains exposure and omics data. As a
quick introduction, a MultiAssayExperiment object is a container for
storing multiple assays (e.g., omics data) and their associated
metadata:</p>
<div class="figure" style="text-align: center">
<img src="mae_rep.png" alt="**Overview of the MultiAssayExperiment Object Structure.** Subject level data is captured within the `colData` of the MultiAssayExperiment. Observations are stored within `experiments` slots in the MultiAssayExperiment and `sampleMap` is used to link the data ( [MultiAssay Special Interest Group 2024](https://www.bioconductor.org/packages/release/bioc/vignettes/MultiAssayExperiment/inst/doc/MultiAssayExperiment.html) )." width="60%"><p class="caption">
<strong>Overview of the MultiAssayExperiment Object Structure.</strong>
Subject level data is captured within the <code>colData</code> of the
MultiAssayExperiment. Observations are stored within
<code>experiments</code> slots in the MultiAssayExperiment and
<code>sampleMap</code> is used to link the data ( <a href="https://www.bioconductor.org/packages/release/bioc/vignettes/MultiAssayExperiment/inst/doc/MultiAssayExperiment.html" class="external-link">MultiAssay
Special Interest Group 2024</a> ).
</p>
</div>
<p>We use the MultiAssayExperiment object to store the exposure and
omics data. The <code>create_expomicset</code> function has several
arguments:</p>
<ul>
<li><p>The <code>var_info</code> argument is a data frame that contains
information about the variables in the exposure meta data. The column
names must contain <code>variable</code> where the values are the column
names of the exposure data frame, and <code>category</code> which
contains general categories for the variable names. This is the data
frame you created with the ontology annotation app!</p></li>
<li><p>The <code>exposure</code> argument is a data frame that contains
the exposure and other metadata.</p></li>
<li><p>The <code>omics</code> argument is a list of data frames that
contain the omics data.</p></li>
<li><p>The <code>row_data</code> argument is a list of data frames that
contain information about the rows of each omics data frame.</p></li>
</ul>
<p>We are going to start by loading in example data pulled from the <a href="https://www.sciencedirect.com/science/article/pii/S016041202200349X?via%3Dihub" class="external-link">ISGlobal
Exposome data challenge 2021</a>. Specifically, we will examine how
exposures and omics features relate to asthma status.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load Libraries</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">tidyexposomics</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create list of omics data frames</span></span>
<span><span class="va">omics_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  <span class="st">"Gene Expression"</span> <span class="op">=</span> <span class="va">exp</span>,</span>
<span>  <span class="st">"Metabolomics"</span> <span class="op">=</span> <span class="va">met</span>,</span>
<span>  <span class="st">"Proteomics"</span> <span class="op">=</span> <span class="va">prot</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create list of omics row data data frames</span></span>
<span><span class="va">fdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  <span class="st">"Gene Expression"</span> <span class="op">=</span> <span class="va">exp_fdata</span>,</span>
<span>  <span class="st">"Metabolomics"</span> <span class="op">=</span> <span class="va">met_fdata</span>,</span>
<span>  <span class="st">"Proteomics"</span> <span class="op">=</span> <span class="va">prot_fdata</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create the expomicset object</span></span>
<span><span class="va">expom</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_expomicset.html">create_expomicset</a></span><span class="op">(</span></span>
<span>  var_info <span class="op">=</span> <span class="va">des</span>,</span>
<span>  exposure <span class="op">=</span> <span class="va">meta</span>,</span>
<span>  omics <span class="op">=</span> <span class="va">omics_list</span>,</span>
<span>  row_data <span class="op">=</span> <span class="va">fdata</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Define our exposure variables</span></span>
<span><span class="va">exp_vars</span> <span class="op">&lt;-</span> <span class="va">des</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"exposure|chemical"</span>,<span class="va">category</span>,ignore.case <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">variable</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code>Ensuring all omics datasets are matrices with column names...
Creating SummarizedExperiment objects with ordered samples and matching rowData...
Creating MultiAssayExperiment object...
MultiAssayExperiment created successfully with ordered samples and rowData.</code></pre>
</div>
<div class="section level2">
<h2 id="quality-control">Quality Control<a class="anchor" aria-label="anchor" href="#quality-control"></a>
</h2>
<hr>
<div class="section level3">
<h3 id="missingness">Missingness<a class="anchor" aria-label="anchor" href="#missingness"></a>
</h3>
<p>Oftentimes when collecting data, there are missing values. The
<code>filter_missing</code> function is used to filter out variables
with too many missing values. The <code>na_thresh</code> argument is
used to set the threshold for missing values. For example, if
<code>na_thresh = 20</code>, then any variable with more than 20%
missing values will be removed. Here we set the threshold to 1% to
filter out variables with more than 1% missing values.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Filter out variables with too many missing values</span></span>
<span><span class="va">expom</span> <span class="op">&lt;-</span> <span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/filter_missing.html">filter_missing</a></span><span class="op">(</span>na_thresh <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<pre><code>Missing Data Filter threshold: 1%
Filtered metadata variables: h_parity
Filtered rows with high missingness in Gene Expression: 0
Filtered rows with high missingness in Metabolomics: 0
Filtered rows with high missingness in Proteomics: 0</code></pre>
<p>We can plot the number of missing values using the
<code>plot_missing_summary</code> function. This function plots the
number of missing values for each variable in the exposure data. The
<code>threshold</code> argument is used to set the threshold for missing
values.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Check the number of missing values</span></span>
<span><span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/plot_missing_summary.html">plot_missing_summary</a></span><span class="op">(</span>threshold <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p><img src="tidyexposomics_files/figure-html/unnamed-chunk-11-1.png" width="70%" style="display: block; margin: auto;"></p>
<p>Here we see that only our exposure data has missing values, and that
the majority of variables have less than 1% missing values.</p>
</div>
<div class="section level3">
<h3 id="imputation">Imputation<a class="anchor" aria-label="anchor" href="#imputation"></a>
</h3>
<p>Now that we have filtered out the variables with too many missing
values, we can impute the missing values. The
<code>run_impute_missing</code> function is used to impute missing
values. While we impute using <code>median</code> here for the speed, we
recommend trying the other imputation options:</p>
<ul>
<li><p><code>mean</code>: Imputes missing values with the mean of the
variable.</p></li>
<li><p><code>knn</code>: Uses k-nearest neighbors to impute missing
values.</p></li>
<li><p><code>mice</code>: Uses the Multivariate Imputation by Chained
Equations (MICE) method to impute missing values.</p></li>
<li><p><code>dep</code>: Uses the DEP method to impute missing
values.</p></li>
<li><p><code>missforest</code>: Uses the MissForest method to impute
missing values.</p></li>
<li><p><code>lod_sqrt2</code>: Imputes missing values using the square
root of the lower limit of detection (LOD) for each variable. This is
useful for variables that have a lower limit of detection, such as
chemical exposures.</p></li>
</ul>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Impute missing values</span></span>
<span><span class="va">expom</span> <span class="op">&lt;-</span> <span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/run_impute_missing.html">run_impute_missing</a></span><span class="op">(</span></span>
<span>    exposure_impute_method <span class="op">=</span> <span class="st">"median"</span>,</span>
<span>    omics_impute_method <span class="op">=</span> <span class="st">"median"</span><span class="op">)</span></span></code></pre></div>
<pre><code>Imputing exposure data using method: median
Imputing omics dataset: Gene Expression using method: median
Imputing omics dataset: Metabolomics using method: median
Imputing omics dataset: Proteomics using method: median</code></pre>
</div>
<div class="section level3">
<h3 id="pca-analysis">PCA Analysis<a class="anchor" aria-label="anchor" href="#pca-analysis"></a>
</h3>
<p>To identify the spread of the data, we can perform a principal
component analysis (PCA). The <code>run_pca</code> function is used to
perform PCA on samples with observations in each omics data frame. The
<code>plot_pca</code> function is used to plot the PCA results. We
automatically identify sample outliers based on the Mahalanobis
distance, a measure of the distance between a point and a
distribution.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Perform principal component analysis</span></span>
<span><span class="va">expom</span> <span class="op">&lt;-</span> <span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>    <span class="fu"><a href="../reference/run_pca.html">run_pca</a></span><span class="op">(</span>action <span class="op">=</span> <span class="st">"add"</span><span class="op">)</span> </span></code></pre></div>
<pre><code>Identifying common samples...
Subsetting exposure data...
Subsetting omics data...
Performing PCA on Feature Space...
Performing PCA on Sample Space...
[1] "Outliers detected: s936, s764, s588, s857, s378, s411, s918"</code></pre>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot principal component analysis results</span></span>
<span><span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/plot_pca.html">plot_pca</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="tidyexposomics_files/figure-html/unnamed-chunk-14-1.png" width="864" style="display: block; margin: auto;"></p>
<p>Here we see have a few sample outliers, and that most variation is
captured in the first two principal components for both features and
samples. We can filter out these samples using the
<code>filter_sample_outliers</code> function.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Filter out sample outliers</span></span>
<span><span class="va">expom</span> <span class="op">&lt;-</span> <span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/filter_sample_outliers.html">filter_sample_outliers</a></span><span class="op">(</span></span>
<span>    outliers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"s411"</span>,<span class="st">"s378"</span>, <span class="st">"s588"</span>,</span>
<span>                 <span class="st">"s764"</span>, <span class="st">"s857"</span>, <span class="st">"s918"</span>,</span>
<span>                 <span class="st">"s936"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="normality-check">Normality Check<a class="anchor" aria-label="anchor" href="#normality-check"></a>
</h3>
<p>When determining variable associations, it is important to check the
normality of the data. The <code>run_check_normality</code> function is
used to check the normality of the data. The
<code>transform_exposure</code> function is used to transform the data
to make it more normal. Here the <code>transform_method</code> is set to
<code>boxcox_best</code> as it will automatically select the best
transformation method based on the data. The
<code>transform_method</code> can be manually set to <code>log2</code>,
<code>sqrt</code>, or <code>x_1_3</code> as well.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Check variable normality &amp; transform variables</span></span>
<span><span class="va">expom</span> <span class="op">&lt;-</span> <span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="co"># Check variable normality</span></span>
<span>  <span class="fu"><a href="../reference/run_normality_check.html">run_normality_check</a></span><span class="op">(</span>action <span class="op">=</span> <span class="st">"add"</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  </span>
<span>  <span class="co"># Transform variables </span></span>
<span>  <span class="fu"><a href="../reference/transform_exposure.html">transform_exposure</a></span><span class="op">(</span>transform_method <span class="op">=</span> <span class="st">"boxcox_best"</span>,</span>
<span>                     exposure_cols <span class="op">=</span> <span class="va">exp_vars</span><span class="op">)</span> </span></code></pre></div>
<pre><code>Checking Normality Using Shapiro-Wilk Test
0 Exposure Variables are Normally Distributed
37 Exposure Variables are NOT Normally Distributed
Applying the boxcox_best transformation.</code></pre>
<p>To check the normality of the data, we can use the
<code>plot_normality_summary</code> function. This function plots the
normality of the data before and after transformation. The
<code>transformed</code> argument is set to <code>TRUE</code> to plot
the transformed data.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expom</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="../reference/plot_normality_summary.html">plot_normality_summary</a></span><span class="op">(</span></span>
<span>    transformed <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<p><img src="tidyexposomics_files/figure-html/unnamed-chunk-17-1.png" width="60%"></p>
</div>
<div class="section level3">
<h3 id="exposure-summary">Exposure Summary<a class="anchor" aria-label="anchor" href="#exposure-summary"></a>
</h3>
<p>We can summarize the exposure data using the
<code>run_summarize_exposures</code> function. This function calculates
summary statistics for each exposure variable, including the number of
values, number of missing values, minimum, maximum, range, sum, median,
mean, standard error, and confidence intervals. The <code>action</code>
argument is set to <code>get</code> to return the summary statistics as
a data frame. If <code>action = "add"</code>, the results are added to
the metadata of the MultiAssayExperiment object.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Summarize exposure data</span></span>
<span><span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/run_summarize_exposures.html">run_summarize_exposures</a></span><span class="op">(</span>action <span class="op">=</span> <span class="st">"get"</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 6 × 17</span></span>m</span>
<span><span class="co">##   variable    n_values  n_na   min   max range   sum median  mean    se ci_lower</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> h_PM             269     0  5.24 82.1  76.9  <span style="text-decoration: underline;">4</span>176.  12.3  15.5   0.67    14.2 </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> h_clf_preg       269     0  0.03  6.77  6.73  912.   3.32  3.39  0.14     3.12</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> h_no2_rati…      269     0  2.11  4.14  2.04  835.   3.12  3.1   0.03     3.05</span>05</span>
<span><span class="co">## <span style="color: #BCBCBC;">4</span> h_pm10_rat…      269     0  3.34  6.91  3.57 <span style="text-decoration: underline;">1</span>351.   4.94  5.02  0.05     4.93</span>93</span>
<span><span class="co">## <span style="color: #BCBCBC;">5</span> h_pm25_rat…      269     0  3.06  4.57  1.5  <span style="text-decoration: underline;">1</span>039.   3.83  3.86  0.02     3.83</span>83</span>
<span><span class="co">## <span style="color: #BCBCBC;">6</span> h_walkabil…      269     0  0.39  0.74  0.35  143.   0.52  0.53  0        0.53</span>53</span>
<span><span class="co">## <span style="color: #949494;"># ℹ 6 more variables: ci_upper &lt;dbl&gt;, variance &lt;dbl&gt;, sd &lt;dbl&gt;, coef_var &lt;dbl&gt;,</span></span>9m</span>
<span><span class="co">## <span style="color: #949494;">#   category &lt;fct&gt;, sub_category &lt;fct&gt;</span></span></span></code></pre>
</div>
<div class="section level3">
<h3 id="exposure-visualization">Exposure Visualization<a class="anchor" aria-label="anchor" href="#exposure-visualization"></a>
</h3>
<hr>
<p>To visualize our exposure data, we can use the
<code>plot_exposures</code> function. This function allows us to plot
the exposure data in a variety of ways. Here we will plot the exposure
data using a boxplot. The <code>exposure_cols</code> argument is used to
set the columns to plot. Additionally, we could specify,
<code>exposure_cat</code> to only plot exposures of a certain category.
The <code>panel_sizes</code> argument is used to set the size of each
panel in the plot. The <code>plot_type</code> argument is used to set
the type of plot to create. Here we use a boxplot, but we could also use
a ridge plot. The <code>ylab</code> argument is used to set the y-axis
label.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot exposure data</span></span>
<span><span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/plot_exposures.html">plot_exposures</a></span><span class="op">(</span></span>
<span>    panel_sizes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">3</span>,<span class="fl">4</span><span class="op">)</span>,</span>
<span>    exposure_cols <span class="op">=</span> <span class="va">exp_vars</span>,</span>
<span>    plot_type <span class="op">=</span> <span class="st">"boxplot"</span>,</span>
<span>    ylab <span class="op">=</span> <span class="st">"Values"</span><span class="op">)</span></span></code></pre></div>
<p><img src="tidyexposomics_files/figure-html/unnamed-chunk-19-1.png" width="960"></p>
</div>
</div>
<div class="section level2">
<h2 id="sample-exposure-association">Sample-Exposure Association<a class="anchor" aria-label="anchor" href="#sample-exposure-association"></a>
</h2>
<hr>
<div class="section level3">
<h3 id="sample-clustering">Sample Clustering<a class="anchor" aria-label="anchor" href="#sample-clustering"></a>
</h3>
<p>The <code>run_cluster_samples</code> function is used to cluster
samples based on the exposure data. The <code>clustering_approach</code>
argument is used to set the clustering approach to automatically
identify clusters. Here we use the <code>dynamic</code> approach, which
uses a dynamic tree cut method to identify clusters. However, you may
choose between these other options as well:</p>
<ul>
<li><p><code>gap</code>: <strong>Gap statistic method
(default)</strong>; estimates optimal <code>k</code> by comparing
within-cluster dispersion to that of reference data.</p></li>
<li><p><code>diana</code>: <strong>Divisive hierarchical clustering
(DIANA)</strong>; chooses <code>k</code> based on the largest drop in
dendrogram height.</p></li>
<li><p><code>elbow</code>: <strong>Elbow method</strong>; detects the
point of maximum curvature in within-cluster sum of squares (WSS) to
determine <code>k</code>.</p></li>
<li><p><code>dynamic</code>: <strong>Dynamic tree cut</strong>;
adaptively detects clusters from a dendrogram structure without needing
to predefine <code>k</code>.</p></li>
<li><p><code>density</code>: <strong>Density-based clustering</strong>
(via <code>densityClust</code>); identifies clusters based on local
density peaks in distance space.</p></li>
</ul>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Sample clustering</span></span>
<span><span class="va">expom</span> <span class="op">&lt;-</span> <span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/run_cluster_samples.html">run_cluster_samples</a></span><span class="op">(</span>exposure_cols <span class="op">=</span> <span class="va">exp_vars</span>,</span>
<span>                  clustering_approach <span class="op">=</span> <span class="st">"dynamic"</span>,</span>
<span>                  action <span class="op">=</span> <span class="st">"add"</span><span class="op">)</span> </span></code></pre></div>
<pre><code>Starting clustering analysis...
 ..cutHeight not given, setting it to 8730  ===&gt;  99% of the (truncated) height range in dendro.
 ..done.
Optimal number of clusters for samples: 6</code></pre>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot sample clusters</span></span>
<span><span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/plot_sample_clusters.html">plot_sample_clusters</a></span><span class="op">(</span></span>
<span>    exposure_cols <span class="op">=</span> <span class="va">exp_vars</span><span class="op">)</span></span></code></pre></div>
<p><img src="tidyexposomics_files/figure-html/unnamed-chunk-21-1.png" width="170%"></p>
<p>Here we see seven clusters, largely driven by particulate matter and
chemical exposure.</p>
</div>
<div class="section level3">
<h3 id="exposure-correlations">Exposure Correlations<a class="anchor" aria-label="anchor" href="#exposure-correlations"></a>
</h3>
<p>The <code>correlate_exposures</code> function is used to correlate
exposures with each other. The <code>exposure_cols</code> argument is
used to set the columns to correlate. The
<code>correlation_cutoff</code> argument is used to set the minimum
correlation score for the association. Here we use a cutoff of 0.3, but
this can be adjusted based on the data.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expom</span> <span class="op">&lt;-</span> <span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/run_correlation.html">run_correlation</a></span><span class="op">(</span>feature_type <span class="op">=</span> <span class="st">"exposures"</span>,</span>
<span>                  action <span class="op">=</span> <span class="st">"add"</span>,</span>
<span>                  correlation_cutoff <span class="op">=</span> <span class="fl">0.3</span><span class="op">)</span></span></code></pre></div>
<p>To visualize the exposure correlations, we can use the
<code>plot_circos_correlation</code> or
<code>plot_heatmap_correlate_exposure</code> functions. Here we will
plot the circos plot. This function creates a circular plot of the
exposure correlations. The <code>correlation_cutoff</code> argument is
used to set the minimum correlation score for the association. Here we
use a cutoff of 0.3, but this can be adjusted based on the data.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/plot_circos_correlation.html">plot_circos_correlation</a></span><span class="op">(</span>feature_type <span class="op">=</span> <span class="st">"exposures"</span>,</span>
<span>                          corr_threshold <span class="op">=</span> <span class="fl">0.3</span>,</span>
<span>                          exposure_cols <span class="op">=</span> <span class="va">exp_vars</span><span class="op">)</span></span></code></pre></div>
<p><img src="tidyexposomics_files/figure-html/unnamed-chunk-23-1.png" width="960" style="display: block; margin: auto;"></p>
</div>
<div class="section level3">
<h3 id="exwas-association">ExWAS Association<a class="anchor" aria-label="anchor" href="#exwas-association"></a>
</h3>
<p>The <code>perform_exwas</code> function is used to perform ExWAS
analysis. The <code>source</code> argument is used to set the source of
the data. Here we use <code>exposures</code> to indicate that we are
using the exposure data. The <code>outcome</code> argument is used to
set the outcome variable. The <code>feature_set</code> argument is used
to set the features to include in the analysis. The
<code>covariates</code> argument is used to set the covariates to
include in the analysis. The <code>family</code> argument is used to set
the family of the model. Here we use <code>binomial</code> for a binary
outcome.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Perform ExWAS Analysis</span></span>
<span><span class="va">expom</span> <span class="op">&lt;-</span> <span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/run_association.html">run_association</a></span><span class="op">(</span></span>
<span>    source <span class="op">=</span> <span class="st">"exposures"</span>,</span>
<span>    outcome <span class="op">=</span> <span class="st">"hs_asthma"</span>,</span>
<span>    feature_set <span class="op">=</span> <span class="va">exp_vars</span><span class="op">[</span></span>
<span>      <span class="op">!</span><span class="va">exp_vars</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"hs_asthma"</span>,</span>
<span>                       <span class="st">"hs_child_age"</span>,</span>
<span>                       <span class="st">"e3_sex"</span><span class="op">)</span><span class="op">]</span>,</span>
<span>    covariates <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"hs_child_age"</span>,</span>
<span>                       <span class="st">"e3_sex"</span><span class="op">)</span>,</span>
<span>    action <span class="op">=</span> <span class="st">"add"</span>,</span>
<span>    family <span class="op">=</span> <span class="st">"binomial"</span><span class="op">)</span></span></code></pre></div>
<pre><code>Running GLMs...</code></pre>
<p>To visualize the results of the ExWAS analysis, we can use the
<code>plot_associate_exposure_outcome</code> function. This function
plots the association results for the exposure data. The
<code>source</code> argument is used to set the source of the data. Here
we use <code>exposures</code> to indicate that we are using the exposure
data. The <code>subtitle</code> argument is used to set the subtitle of
the plot. The <code>filter_thresh</code> argument is used to set the
p-value threshold for significance.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/plot_association.html">plot_association</a></span><span class="op">(</span></span>
<span>    subtitle <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"Covariates:"</span>,<span class="st">"Age,"</span>,<span class="st">"Sex"</span><span class="op">)</span>,</span>
<span>    source <span class="op">=</span> <span class="st">"exposures"</span>,</span>
<span>    filter_thresh <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span></span></code></pre></div>
<p><img src="tidyexposomics_files/figure-html/unnamed-chunk-25-1.png" width="60%" style="display: block; margin: auto;"></p>
<p>Here we see that particulate matter, PBDE, and disinfection byproduct
levels are associated with asthma status after adjusting for age and
sex. We can also associate our omics features with an outcome of
interest using the <code><a href="../reference/run_association.html">run_association()</a></code> function.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Perform ExWAS Analysis</span></span>
<span><span class="va">expom</span>  <span class="op">&lt;-</span> <span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>    <span class="fu"><a href="../reference/run_association.html">run_association</a></span><span class="op">(</span></span>
<span>        outcome <span class="op">=</span> <span class="st">"hs_asthma"</span>,</span>
<span>        source <span class="op">=</span> <span class="st">"omics"</span>,</span>
<span>        covariates <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"hs_child_age"</span>,</span>
<span>                       <span class="st">"e3_sex"</span><span class="op">)</span>,</span>
<span>        top_n <span class="op">=</span> <span class="fl">500</span>,</span>
<span>        action <span class="op">=</span> <span class="st">"add"</span>,</span>
<span>        family <span class="op">=</span> <span class="st">"binomial"</span><span class="op">)</span></span></code></pre></div>
<pre><code>Log2-Transforming each assay in MultiAssayExperiment...
Scaling each assay in MultiAssayExperiment...
Running GLMs...</code></pre>
<p>We can visualize the results using a Manhattan plot. The
<code>plot_manhattan</code> function is used to plot the results. The
<code>pval_thresh</code> argument is used to set the p-value threshold
for significance. The <code>vars_to_label</code> argument is used to set
the variables to label on the plot. The <code>min_per_cat</code>
argument is used to set the minimum number of significant features per
category to include in the plot. The <code>panel_sizes</code> argument
is used to set the size of each panel in the plot.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/plot_manhattan.html">plot_manhattan</a></span><span class="op">(</span></span>
<span>    min_per_cat <span class="op">=</span> <span class="fl">0</span>,</span>
<span>    vars_to_label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"TC17001579.hg.1"</span>,<span class="st">"TC02003949.hg.1"</span>,<span class="st">"HMGN2P28"</span><span class="op">)</span>,</span>
<span>    panel_sizes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">4</span>,<span class="fl">2</span>,<span class="fl">1</span>,<span class="fl">3</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="tidyexposomics_files/figure-html/unnamed-chunk-27-1.png" width="864" style="display: block; margin: auto;"></p>
</div>
<div class="section level3">
<h3 id="exposome-scores">Exposome Scores<a class="anchor" aria-label="anchor" href="#exposome-scores"></a>
</h3>
<hr>
<p>We can also calculate exposome scores, which are a summary measure of
exposure. The <code>run_exposome_score</code> function is used to
calculate the exposome score. The <code>exposure_cols</code> argument is
used to set the columns to use for the exposome score. The
<code>score_type</code> argument is used to set the type of score to
calculate. Here we could use:</p>
<ul>
<li><p><code>median</code>: Calculates the median of the exposure
variables.</p></li>
<li><p><code>mean</code>: Calculates the mean of the exposure
variables.</p></li>
<li><p><code>sum</code>: Calculates the sum of the exposure
variables.</p></li>
<li><p><code>pca</code>: Calculates the first principal component of the
exposure variables.</p></li>
<li><p><code>irt</code>: Uses Item Response Theory to calculate the
exposome score.</p></li>
<li><p><code>quantile</code>: Calculates the quantile of the exposure
variables.</p></li>
</ul>
<p>The <code>score_column_name</code> argument is used to set the name
of the column to store the exposome score in.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Calculate Exposome Scores</span></span>
<span><span class="va">expom</span> <span class="op">&lt;-</span> <span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/run_exposome_score.html">run_exposome_score</a></span><span class="op">(</span>exposure_cols <span class="op">=</span> <span class="va">exp_vars</span>,</span>
<span>                     score_type <span class="op">=</span> <span class="st">"median"</span>,</span>
<span>                     score_column_name <span class="op">=</span> <span class="st">"exposome_median_score"</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/run_exposome_score.html">run_exposome_score</a></span><span class="op">(</span>exposure_cols <span class="op">=</span> <span class="va">exp_vars</span>,</span>
<span>                     score_type <span class="op">=</span> <span class="st">"pca"</span>,</span>
<span>                     score_column_name <span class="op">=</span> <span class="st">"exposome_pca_score"</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/run_exposome_score.html">run_exposome_score</a></span><span class="op">(</span>exposure_cols <span class="op">=</span> <span class="va">exp_vars</span>,</span>
<span>                     score_type <span class="op">=</span> <span class="st">"irt"</span>,</span>
<span>                     score_column_name <span class="op">=</span> <span class="st">"exposome_irt_score"</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/run_exposome_score.html">run_exposome_score</a></span><span class="op">(</span>exposure_cols <span class="op">=</span> <span class="va">exp_vars</span>,</span>
<span>                     score_type <span class="op">=</span> <span class="st">"quantile"</span>,</span>
<span>                     score_column_name <span class="op">=</span> <span class="st">"exposome_quantile_score"</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/run_exposome_score.html">run_exposome_score</a></span><span class="op">(</span>exposure_cols <span class="op">=</span> <span class="va">exp_vars</span>,</span>
<span>                     score_type <span class="op">=</span> <span class="st">"var"</span>,</span>
<span>                     score_column_name <span class="op">=</span> <span class="st">"exposome_var_score"</span><span class="op">)</span></span></code></pre></div>
<pre><code>Extracting exposure data...
Extracting exposure data...
Extracting exposure data...
Extracting exposure data...
Extracting exposure data...
Calculating median exposure scores...
Calculating PCA exposure scores...
Calculating IRT exposure scores...
Iteration: 52, Log-Lik: -21076.825, Max-Change: 0.00010
Calculating quantile exposure scores...
Calculating variance exposure scores...</code></pre>
<p>We can then associate these exposome scores with the outcome of
interest using the <code>run_association</code> function, just like we
did before. However, this time we specify our <code>feature_set</code>
to be the exposome scores we just calculated.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Associate Exposome Scores with Outcome</span></span>
<span><span class="va">expom</span> <span class="op">&lt;-</span> <span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/run_association.html">run_association</a></span><span class="op">(</span></span>
<span>    outcome <span class="op">=</span> <span class="st">"hs_asthma"</span>,</span>
<span>    source <span class="op">=</span> <span class="st">"exposures"</span>,</span>
<span>    feature_set <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"exposome_median_score"</span>,</span>
<span>                  <span class="st">"exposome_pca_score"</span>,</span>
<span>                  <span class="st">"exposome_irt_score"</span>,</span>
<span>                  <span class="st">"exposome_quantile_score"</span>,</span>
<span>                  <span class="st">"exposome_var_score"</span><span class="op">)</span>,</span>
<span>    covariates <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"hs_child_age"</span>,</span>
<span>                       <span class="st">"e3_sex"</span><span class="op">)</span>,</span>
<span>    action <span class="op">=</span> <span class="st">"add"</span>,</span>
<span>    family <span class="op">=</span> <span class="st">"binomial"</span><span class="op">)</span></span></code></pre></div>
<pre><code>Running GLMs...</code></pre>
<p>To plot the results of the exposome score association with the
outcome, we can use the <code>plot_association</code> function:</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/plot_association.html">plot_association</a></span><span class="op">(</span></span>
<span>    subtitle <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"Covariates:"</span>,<span class="st">"Age,"</span>,<span class="st">"Sex"</span><span class="op">)</span>,</span>
<span>    source <span class="op">=</span> <span class="st">"exposures"</span>,</span>
<span>    terms <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"exposome_median_score"</span>,</span>
<span>                        <span class="st">"exposome_pca_score"</span>,</span>
<span>                        <span class="st">"exposome_irt_score"</span>,</span>
<span>                        <span class="st">"exposome_quantile_score"</span>,</span>
<span>                        <span class="st">"exposome_var_score"</span><span class="op">)</span>,</span>
<span>    filter_thresh <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p><img src="tidyexposomics_files/figure-html/unnamed-chunk-30-1.png" width="60%" style="display: block; margin: auto;"></p>
<p>Here we see that most scores are associated with a decreased
probability of asthma.</p>
</div>
</div>
<div class="section level2">
<h2 id="differential-abundance">Differential Abundance<a class="anchor" aria-label="anchor" href="#differential-abundance"></a>
</h2>
<hr>
<div class="section level3">
<h3 id="differential-abundance-1">Differential Abundance<a class="anchor" aria-label="anchor" href="#differential-abundance-1"></a>
</h3>
<p>We also provide functionality to assess differentially abundant
features associated with a particular outcome across several omics’
layers. The <code>run_differential_abundance</code> function is used to
run the differential abundance analysis. The <code>formula</code>
argument is used to set the formula for the model. The
<code>method</code> argument is used to set the method to use for the
differential abundance analysis. Here we use the <code>limma_voom</code>
method, which is a popular method for differential abundance analysis.
The <code>minimum_counts</code> and <code>minimum_proportion</code>
arguments are used to filter out features with low counts or low
proportions. The <code>scaling_method</code> argument is used to set the
scaling method to use. Here we use <code>none</code>, but we could also
use <code>quantile</code> or <code>TMM</code>. The <code>action</code>
argument is set to <code>add</code> to add the results to the existing
object.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Run differential abundance analysis</span></span>
<span><span class="va">expom</span> <span class="op">&lt;-</span> <span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/run_differential_abundance.html">run_differential_abundance</a></span><span class="op">(</span></span>
<span>    formula <span class="op">=</span> <span class="op">~</span> <span class="va">hs_asthma</span> <span class="op">+</span> <span class="va">hs_child_age</span> <span class="op">+</span> <span class="va">e3_sex</span>,</span>
<span>    method <span class="op">=</span> <span class="st">"limma_voom"</span>,</span>
<span>    minimum_counts <span class="op">=</span> <span class="fl">1</span>,</span>
<span>    minimum_proportion <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>    scaling_method <span class="op">=</span> <span class="st">"none"</span>,</span>
<span>    action <span class="op">=</span> <span class="st">"add"</span><span class="op">)</span> </span></code></pre></div>
<pre><code>Running differential abundance testing...
Processing assay: Gene Expression
No group or design set. Assuming all samples belong to one group.
...
Differential abundance testing completed.</code></pre>
<p>Using a volcano plot, we can visualize the results of the
differential abundance analysis. The <code>plot_volcano</code> function
is used to plot the results. The <code>pval_thresh</code> argument is
used to set the p-value threshold for significance. The
<code>logFC_thresh</code> argument is used to set the log fold change
threshold for significance. The <code>top_n_label</code> argument is
used to set the number of top features to label on the plot. The
<code>nrow</code> argument is used to set the number of rows in the
facet layout.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot Differential Abundance Results</span></span>
<span><span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/plot_volcano.html">plot_volcano</a></span><span class="op">(</span></span>
<span>    top_n_label <span class="op">=</span> <span class="fl">3</span>,</span>
<span>    logFC_thresh <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span>,</span>
<span>    pval_thresh <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>    nrow <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p><img src="tidyexposomics_files/figure-html/unnamed-chunk-32-1.png" width="672" style="display: block; margin: auto;"></p>
</div>
<div class="section level3">
<h3 id="sensitivity-analysis">Sensitivity Analysis<a class="anchor" aria-label="anchor" href="#sensitivity-analysis"></a>
</h3>
<p>Depending on pre-processing steps, the results of the differential
abundance analysis may vary. The <code>sensitivity_analysis</code>
function is used to perform a sensitivity analysis to determine the
robustness of the results. Here we determine if a feature is still
differentially abundant if different minimum values, proportions,
scaling methods are used, the inclusion of covariates, and after
bootstrapping. We then define a stability score based on the number of
times a feature is found to be differentially abundant under different
conditions as well as the consistency of the effect size:</p>
<p><span class="math display">\[ Stability\ Score = \frac{\sum_i{(p_i
&lt; \alpha)}}{N} * \frac{1}{1 +
\frac{\sigma_{\beta}}{\mu_{|\beta|}}}\]</span></p>
<p>Where:</p>
<ul>
<li><p><span class="math inline">\(p_i\)</span> is the p-value for the
<span class="math inline">\(i^{th}\)</span> test</p></li>
<li><p><span class="math inline">\(\alpha\)</span> is the significance
threshold</p></li>
<li><p><span class="math inline">\(N\)</span> is the number of
tests</p></li>
<li><p><span class="math inline">\(\sigma_{\beta}\)</span> is the
standard deviation of the effect size estimates</p></li>
<li><p><span class="math inline">\(\mu_{|\beta|}\)</span> is the mean of
the absolute value of the effect size estimates.</p></li>
</ul>
<p>The first term captures the proportion of tests that are significant,
while the second term captures the consistency of the effect size
estimates. The stability score ranges from 0 to 1.</p>
<p>A stability score of 1 indicates that the feature is always found to
be differentially abundant, while a stability score of 0 indicates that
the feature is never found to be differentially abundant. Besides these,
we provide other score metrics as well:</p>
<ul>
<li><p><code>presence_rate</code>: Proportion of runs in which the
feature’s p-value is below the specified threshold (selection
frequency).</p></li>
<li><p><code>effect_consistency</code>: Inverse of the coefficient of
variation of log fold-changes; measures effect size stability across
runs.</p></li>
<li><p><code>stability_score</code>: Hybrid score combining
<code>presence_rate</code> and <code>effect_consistency</code>,
capturing reproducibility and signal strength.</p></li>
<li><p><code>mean_log_p</code>: Average of negative log-transformed
p-values; represents overall statistical signal strength.</p></li>
<li><p><code>logp_weighted_score</code>: Product of
<code>mean_log_p</code> and <code>effect_consistency</code>; highlights
consistently strong features.</p></li>
<li><p><code>sd_logFC</code>: Standard deviation of log fold-change
estimates; quantifies variability of effect sizes.</p></li>
<li><p><code>iqr_logFC</code>: Interquartile range of log fold-changes;
provides a robust measure of effect size spread.</p></li>
<li><p><code>cv_logFC</code>: Coefficient of variation of log
fold-changes; reflects relative variability of effect size.</p></li>
<li><p><code>sign_flip_freq</code>: Proportion of runs where the sign of
the effect size differs from the overall average direction.</p></li>
<li><p><code>sd_log_p</code>: Standard deviation of log-transformed
p-values; indicates variability in statistical signal.</p></li>
</ul>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Perform Sensitivity Analysis</span></span>
<span><span class="va">expom</span> <span class="op">&lt;-</span> <span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/run_sensitivity_analysis.html">run_sensitivity_analysis</a></span><span class="op">(</span></span>
<span>    base_formula <span class="op">=</span> <span class="op">~</span> <span class="va">hs_asthma</span> <span class="op">+</span> <span class="va">hs_child_age</span> <span class="op">+</span> <span class="va">e3_sex</span>, </span>
<span>    methods <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"limma_voom"</span><span class="op">)</span>,</span>
<span>    scaling_methods <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"none"</span><span class="op">)</span>,</span>
<span>    min_counts_range <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">5</span><span class="op">)</span>,</span>
<span>    min_proportion_range <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.1</span>,<span class="fl">0.3</span><span class="op">)</span>,</span>
<span>    covariates_to_remove <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"hs_child_age"</span> , <span class="st">"e3_sex"</span><span class="op">)</span>,</span>
<span>    pval_col <span class="op">=</span> <span class="st">"adj.P.Val"</span>,</span>
<span>    logfc_col <span class="op">=</span> <span class="st">"logFC"</span>,</span>
<span>    pval_threshold <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>    stability_metric <span class="op">=</span> <span class="st">"stability_score"</span>,</span>
<span>    bootstrap_n <span class="op">=</span> <span class="fl">10</span>,</span>
<span>    action <span class="op">=</span> <span class="st">"add"</span><span class="op">)</span> </span></code></pre></div>
<pre><code>Running differential abundance testing...
Processing assay: Gene Expression
...
Feature stability analysis completed.
Number of features above threshold of 0:
-----------------------------------------
Proteomics: 50/3126
Gene Expression: 996/28738
Metabolomics: 0/177
Sensitivity analysis completed.</code></pre>
<p>Here we plot the sensitivity analysis results using the
<code>plot_sensitivity_summary</code> function. The
<code>stability_score_thresh</code> argument is used to set the
stability score threshold for significance. Here we use a threshold of
0.25, but this can be adjusted based on the data. Here we set the
threshold to be quite liberal based given this is highly subsampled
data.</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot sensitivity analysis results</span></span>
<span><span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/plot_sensitivity_summary.html">plot_sensitivity_summary</a></span><span class="op">(</span>stability_score_thresh <span class="op">=</span> <span class="fl">0.25</span>,</span>
<span>                           stability_metric <span class="op">=</span> <span class="st">"stability_score"</span><span class="op">)</span></span></code></pre></div>
<p><img src="tidyexposomics_files/figure-html/unnamed-chunk-34-1.png" width="80%" style="display: block; margin: auto;"></p>
</div>
</div>
<div class="section level2">
<h2 id="multi-omics-integration">Multi-Omics Integration<a class="anchor" aria-label="anchor" href="#multi-omics-integration"></a>
</h2>
<hr>
<div class="section level3">
<h3 id="multi-omics-integration-1">Multi-Omics Integration<a class="anchor" aria-label="anchor" href="#multi-omics-integration-1"></a>
</h3>
<p>While differential abundance analysis per omic can deliver insights
into how each omic is associated with a particular outcome, we may want
to leverage methods which integrate multiple omics layers. The
<code>run_multiomics_integration</code> function is used to integrate
multiple omics layers. Here we use either <code>MCIA</code>,
<code>MCCA</code> or the <code>MOFA</code> method to integrate omics
layers:</p>
<ul>
<li><p><code>MCIA</code>: Multiple Co-inertia Analysis, a method that
uses canonical correlation analysis to integrate multiple omics layers.
We use the nipalsMCIA algorithm to compute the co-inertia scores from
the <a href="https://www.bioconductor.org/packages/release/bioc/html/nipalsMCIA.html" class="external-link">nipalsMCIA
package</a>.</p></li>
<li><p><code>MCCA</code>: Multi-Omics Canonical Correlation Analysis, a
method that uses canonical correlation analysis to integrate multiple
omics layers. We use the <a href="https://github.com/bnaras/PMA" class="external-link">PMA
package</a> to compute the canonical correlation score.</p></li>
<li><p><code>MOFA</code>: Multi-Omics Factor Analysis, a method that
uses factor analysis to integrate multiple omics layers. MOFA is
implemented using the <a href="https://www.bioconductor.org/packages/release/bioc/html/MOFA2.html" class="external-link">MOFA2
package</a>.</p></li>
</ul>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Perform Multi-Omics Integration</span></span>
<span><span class="va">expom</span> <span class="op">&lt;-</span> <span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/run_multiomics_integration.html">run_multiomics_integration</a></span><span class="op">(</span>method <span class="op">=</span> <span class="st">"MCIA"</span>,</span>
<span>                             action <span class="op">=</span> <span class="st">"add"</span><span class="op">)</span> </span></code></pre></div>
<pre><code>Scaling each assay in MultiAssayExperiment...
Running multi-omics integration using MCIA...
Applying MCIA with NIPALS...
Performing column-level pre-processing...
Column pre-processing completed.
Performing block-level preprocessing...
Block pre-processing completed.
Computing order 1 scores
Computing order 2 scores
Computing order 3 scores
Computing order 4 scores
Computing order 5 scores
Computing order 6 scores
Computing order 7 scores
Computing order 8 scores
Computing order 9 scores
Computing order 10 scores</code></pre>
<p>We can then use <code><a href="../reference/plot_factor_summary.html">plot_factor_summary()</a></code> to visualize which
omics contribute most to which factors.</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot multi-omics factor summary</span></span>
<span><span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/plot_factor_summary.html">plot_factor_summary</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="tidyexposomics_files/figure-html/unnamed-chunk-36-1.png" width="576" style="display: block; margin: auto;"></p>
</div>
<div class="section level3">
<h3 id="factor-analysis">Factor Analysis<a class="anchor" aria-label="anchor" href="#factor-analysis"></a>
</h3>
<p>These methods are designed to identify factors that we can then
associate with an outcome variable. Here we will use the
<code>associate_factor_outcome</code> function to identify factors that
are associated with asthma status after controlling for the child age
and sex. Additionally, we will set the p-value threshold,
<code>pval_thresh</code>, to 1 to include all factors. The
<code>action</code> argument is set to <code>add</code> to add the
results to the existing object. Here we print the results to the
console, but this can be set to <code>FALSE</code> to suppress the
output.</p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Identify factors that correlate with the outcome</span></span>
<span><span class="va">expom</span> <span class="op">&lt;-</span> <span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/run_association.html">run_association</a></span><span class="op">(</span></span>
<span>    source <span class="op">=</span> <span class="st">"factors"</span>,</span>
<span>    outcome <span class="op">=</span> <span class="st">"hs_asthma"</span>,</span>
<span>    feature_set <span class="op">=</span> <span class="va">exp_vars</span><span class="op">[</span></span>
<span>      <span class="op">!</span><span class="va">exp_vars</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"hs_asthma"</span>,</span>
<span>                       <span class="st">"hs_child_age"</span>,</span>
<span>                       <span class="st">"e3_sex"</span><span class="op">)</span><span class="op">]</span>,</span>
<span>    covariates <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"hs_child_age"</span>,</span>
<span>                       <span class="st">"e3_sex"</span><span class="op">)</span>,</span>
<span>    action <span class="op">=</span> <span class="st">"add"</span>,</span>
<span>    family <span class="op">=</span> <span class="st">"binomial"</span>,</span>
<span>    print <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code>term    p.value 
&lt;chr&gt;   &lt;dbl&gt;
V1  0.17083208          
V2  0.79768883          
V3  0.97476037          
V4  0.59766388          
V5  0.22535931          
V6  0.05436188          
V7  0.59501506          
V8  0.83149295          
V9  0.14050018          
V10 0.47106052</code></pre>
<p>We see that the first, sixth and ninth factor are associated with
asthma status at a p-value below 0.2. Factors have loading scores which
indicate the strength of the association between the factor and the
features. Here we can extract the top features, those in the 90th
percentile, per omic, associated with our factors of interest.</p>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Extract top features that contribute to a factor</span></span>
<span><span class="va">expom</span> <span class="op">&lt;-</span> <span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/extract_top_factor_features.html">extract_top_factor_features</a></span><span class="op">(</span>factors <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"V1"</span>, <span class="st">"V6"</span>, <span class="st">"V9"</span><span class="op">)</span>, </span>
<span>                              method <span class="op">=</span> <span class="st">"percentile"</span>,</span>
<span>                              percentile <span class="op">=</span> <span class="fl">0.9</span>,</span>
<span>                              action <span class="op">=</span> <span class="st">"add"</span><span class="op">)</span> </span></code></pre></div>
<pre><code>Extracting top contributing features for specified factors...
Using MCIA block loadings...
Selected 9612 features contributing to specified factors.</code></pre>
<p>We can visualize the top features associated with each factor using
the <code>plot_top_factor_features</code> function. The
<code>top_n</code> argument is used to set the number of top features to
plot. The <code>factors</code> argument is used to set the factors to
plot.</p>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot top factor features</span></span>
<span><span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/plot_top_factor_features.html">plot_top_factor_features</a></span><span class="op">(</span></span>
<span>    top_n <span class="op">=</span> <span class="fl">15</span>,</span>
<span>    factors <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"V1"</span>, <span class="st">"V6"</span>, <span class="st">"V9"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="tidyexposomics_files/figure-html/unnamed-chunk-39-1.png" width="480" style="display: block; margin: auto;"></p>
<p>Immediately, we can see that these factors share some common
features, such as the <code>CTSW</code> gene. We can also see that the
factors are associated with different omics layers. To identify the
common features across factors, we can use the
<code>run_factor_overlap</code> function. This function will identify
the features that are shared across the specified factors.</p>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expom</span> <span class="op">&lt;-</span> <span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/run_factor_overlap.html">run_factor_overlap</a></span><span class="op">(</span>stability_score <span class="op">=</span> <span class="fl">0.25</span>,</span>
<span>                     robust_comparison <span class="op">=</span> <span class="cn">T</span>,</span>
<span>                     score_col <span class="op">=</span> <span class="st">"stability_score"</span>,</span>
<span>                     pval_thresh <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>                     pval_col <span class="op">=</span> <span class="st">"p.value"</span>,</span>
<span>                     logfc_thresh <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="fl">1.1</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code>Found 817 common features across factors.</code></pre>
<p>Let’s visualize how these features overlap across factors. The
<code>plot_factor_overlap</code> function is used to plot the overlap of
features across factors.</p>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/plot_factor_overlap.html">plot_factor_overlap</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="tidyexposomics_files/figure-html/unnamed-chunk-41-1.png" width="1152" style="display: block; margin: auto;"></p>
<p>Here we see that there are 817 features that are shared across the
factors. However, only a subset, ~ 200 features, are differentially
abundant.</p>
</div>
</div>
<div class="section level2">
<h2 id="exposure-omics-association">Exposure-Omics Association<a class="anchor" aria-label="anchor" href="#exposure-omics-association"></a>
</h2>
<hr>
<div class="section level3">
<h3 id="exposure-omics-association-1">Exposure-Omics Association<a class="anchor" aria-label="anchor" href="#exposure-omics-association-1"></a>
</h3>
<p>Now we have the option to correlate either the top factor features,
differentially abundant features, or user specified omics features (by
using a variable map, a data frame with two columns,
<code>exp_name</code> for the name of the omics assay, and
<code>feature</code> for the name of the molecular feature) with
exposures. Here we will correlate the differentially abundant features
with exposures. Setting <code>robust</code> to <code>TRUE</code> will
use features with a stability score over <code>score_thresh</code>. We
can also set thresholds for the differential expression log fold change
and p-value to filter the results. The <code>correlation_cutoff</code>
is used to set the minimum correlation score, while the
<code>pval_cutoff</code> is used to set the maximum p-value for the
association.</p>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Correlate top differentially abundant features with exposures</span></span>
<span><span class="va">expom</span> <span class="op">&lt;-</span> <span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/run_correlation.html">run_correlation</a></span><span class="op">(</span>feature_type <span class="op">=</span> <span class="st">"degs"</span>,</span>
<span>                  exposure_cols <span class="op">=</span> <span class="va">exp_vars</span>,</span>
<span>                  action <span class="op">=</span> <span class="st">"add"</span>,</span>
<span>                  correlation_cutoff <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>                  deg_logfc_col <span class="op">=</span> <span class="st">"logFC"</span>,</span>
<span>                  deg_logfc_thresh <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span>,</span>
<span>                  pval_cutoff <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>                  robust <span class="op">=</span> <span class="cn">T</span>,</span>
<span>                  score_col <span class="op">=</span> <span class="st">"stability_score"</span>,</span>
<span>                  score_thresh <span class="op">=</span> <span class="fl">0.25</span><span class="op">)</span></span></code></pre></div>
<pre><code>Performing correlation analysis on summarized experiment...
Running Spearman correlation analysis...
Correlation analysis completed.
Processing experiment: Proteomics
  - Processing batch 1 of 1 (48 features)...
Performing correlation analysis on summarized experiment...
Running Spearman correlation analysis...
Correlation analysis completed.</code></pre>
<p>We can plot the results of the exposure-omics association analysis
using the <code>plot_bar_correlate_summary</code> function. Here we set
the mode to <code>summary</code>, which will plot the number of
associations per exposure and feature type. The
<code>feature_type</code> argument is used to set the type of features
to plot. Here we use <code>degs</code>, which are the differentially
abundant features. The <code>exposure_cols</code> argument is used to
set the columns to plot.</p>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/plot_correlation_summary.html">plot_correlation_summary</a></span><span class="op">(</span>mode <span class="op">=</span> <span class="st">"summary"</span><span class="op">)</span></span></code></pre></div>
<p><img src="tidyexposomics_files/figure-html/unnamed-chunk-43-1.png" width="1152" style="display: block; margin: auto;"></p>
<p>Here we note that transcriptomic features have a greater number of
associations with exposures. Among exposures, we see that outdoor
exposures have the greatest number of associations with features. It may
also be useful to identify which exposures are correlated with similar
molecular features. We can do this with the
<code>plot_circos_exposure_shared_features</code> function. This
function will plot a circos plot of the exposures and their shared
features. The <code>geneset</code> argument is used to set the geneset
to use for the analysis. Here we use the <code>degs</code> geneset,
which plots the differentially abundant feature - exposure correlation
information. The cutoff argument is used to set the minimum number of
shared features to plot. Here we set the cutoff to 0, which means that
all shared features will be plotted.</p>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot Shared Feature Correlations Between Exposures</span></span>
<span><span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/plot_circos_correlation.html">plot_circos_correlation</a></span><span class="op">(</span>feature_type <span class="op">=</span> <span class="st">"degs"</span>,</span>
<span>                          shared_cutoff <span class="op">=</span> <span class="fl">10</span>,</span>
<span>                          midpoint <span class="op">=</span> <span class="fl">200</span><span class="op">)</span></span></code></pre></div>
<p><img src="tidyexposomics_files/figure-html/unnamed-chunk-44-1.png" width="960"></p>
</div>
<div class="section level3">
<h3 id="network-analysis">Network Analysis<a class="anchor" aria-label="anchor" href="#network-analysis"></a>
</h3>
<p>The <code>run_create_network</code> function is used to create a
network of exposures and omics features. The correlation results are
used to create the network. The <code>feature_type</code> argument is
used to set the type of features to use for the network. Here we use
<code>degs</code>, which are the differentially abundant features.
However we could choose anyo f the following:</p>
<ul>
<li><p><code>degs</code>: Differentially abundant features correlated
with exposures.</p></li>
<li><p><code>factors</code>: Factor features correlated with
exposures.</p></li>
<li><p><code>omics</code>: User specified omics features correlated with
exposures.</p></li>
</ul>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expom</span> <span class="op">&lt;-</span> <span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/run_create_network.html">run_create_network</a></span><span class="op">(</span></span>
<span>    feature_type <span class="op">=</span> <span class="st">"degs"</span>,</span>
<span>    action <span class="op">=</span> <span class="st">"add"</span><span class="op">)</span></span></code></pre></div>
<pre><code>Creating network from correlation results...
Network added to metadata as: network_degs</code></pre>
<p>To plot the network we can use the <code>plot_network</code>
function. The <code>network</code> argument is used to set the type of
network to plot. The <code>top_n_nodes</code> argument is used to set
the number of nodes to plot. The <code>node_color_var</code> argument is
used to set the variable to color the nodes by. The <code>label</code>
argument is used to set whether or not to label the nodes. We can label
the top n nodes, where the default is 5 nodes based on centrality.
Addtionally, we can chose to label certain nodes using the
<code>nodes_to_label</code> arguement. The <code>facet_var</code>
argument is used to set the variable to facet the plot by. The
<code>foreground</code> argument is used to set the color of the facets.
We can also include the network statistics using the
<code>include_stats</code> argument.</p>
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expom</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/plot_network.html">plot_network</a></span><span class="op">(</span></span>
<span>        network <span class="op">=</span> <span class="st">"degs"</span>,</span>
<span>        top_n_nodes <span class="op">=</span> <span class="fl">50</span>,</span>
<span>        include_stats <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>        cor_thresh <span class="op">=</span><span class="fl">0.090</span>, </span>
<span>        node_color_var <span class="op">=</span> <span class="st">"group"</span>,</span>
<span>        label <span class="op">=</span> <span class="cn">T</span>,</span>
<span>        label_top_n <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span></code></pre></div>
<p><img src="tidyexposomics_files/figure-html/unnamed-chunk-46-1.png" width="960"></p>
<pre><code>Extracting graph...
Filtering nodes based on correlation threshold...
Filtering top 50 nodes based on centrality...</code></pre>
<p>Here we note that the network is highly connected, with chemical
exposures and particulate matter exposures having the most connections
to omics features. It should be noted these networks will look different
based on the type of correlation analysis you perform:</p>
<ul>
<li><p><code>degs</code>: Correlation between differentially abundant
features and exposures. Meaning that connections between exposures and
between omics will be missing.</p></li>
<li><p><code>factors</code>: Correlation between factors and exposures.
Meaning that connections between factor features and between exposures
will be missing.</p></li>
<li><p><code>omics</code>: Correlation between omics features and
exposures. Where there will be connections between omics and
features.</p></li>
</ul>
</div>
<div class="section level3">
<h3 id="exposure-omics-impact-network-analysis">Exposure-Omics Impact Network Analysis<a class="anchor" aria-label="anchor" href="#exposure-omics-impact-network-analysis"></a>
</h3>
<p>When examining an outcome variable, the number of omics features
associated with that variable have different coexpression relationships
with each other. Features with high coexpression have higher degree and
therefore may be more important in the network. The
<code>run_exposure_impact</code> function is used to run an
exposure-omics impact analysis. Here we calculate the degree of features
associated with exposures. We can then see is a particular exposure
associated with higher degree features. The
<code>stability_threshold</code> argument is used to set the stability
threshold for the features. Here we use a threshold 0.25, which means
that only features with a stability score above 0.25 will be included in
the analysis.</p>
<div class="sourceCode" id="cb61"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Run exposure-omics impact analaysis</span></span>
<span><span class="va">expom</span> <span class="op">&lt;-</span> <span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/run_exposure_impact.html">run_exposure_impact</a></span><span class="op">(</span>feature_type <span class="op">=</span> <span class="st">"degs"</span>,</span>
<span>                      stability_threshold <span class="op">=</span> <span class="fl">0.25</span>,</span>
<span>                      robust <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span></code></pre></div>
<p>To plot the results of the exposure-omics impact analysis, we can use
the <code>plot_exposure_impact</code> function. We set
<code>feature_type</code> to <code>degs</code>, which means that we are
plotting the degree of the differentially abundant features correlated
with exposures. The <code>min_per_group</code> argument is used to set
the minimum number of features per group to plot. This is useful for
filtering out exposures that do not have enough associated features. The
<code>ncol</code> argument is used to set the number of columns in the
plot, while the <code>widths</code> argument is used to set the widths
of the columns in the plot.</p>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot exposure-omics impact results</span></span>
<span><span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/plot_exposure_impact.html">plot_exposure_impact</a></span><span class="op">(</span></span>
<span>    feature_type <span class="op">=</span> <span class="st">"degs"</span>,</span>
<span>    min_per_group <span class="op">=</span> <span class="fl">10</span>,</span>
<span>    ncol <span class="op">=</span> <span class="fl">2</span>,</span>
<span>    widths <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">3</span>,<span class="fl">1</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="tidyexposomics_files/figure-html/unnamed-chunk-48-1.png" width="1152" style="display: block; margin: auto;"></p>
<p>Here we see that BPA (<code>hs_bpa_madj</code>) and methyl paraben
(<code>hs_mepa_madj</code>) exposure, while not associated with as many
differentially abundant features as Monoisobutyl phthalate
(<code>hs_mibp_cadj</code>), the features that these exposures are
associated with are have about the same degree, suggesting that they may
still be an important exposure to consider in the context of asthma.
Additionally, we note that more than half of the features associated
with asthma are associated with the exposures we profiled, while the
remaining are not associated with any exposures. This suggests that we
may be missing exposures to profile or that these features are not
environmentally responsive.</p>
</div>
</div>
<div class="section level2">
<h2 id="enrichment-analysis">Enrichment Analysis<a class="anchor" aria-label="anchor" href="#enrichment-analysis"></a>
</h2>
<hr>
<p>The benefit of grouping our exposures into categories is that we can
perform enrichment analysis on the associated features to determine how
broad categories of exposures are associated with features. The
<code>run_enrichment</code> function is used to perform enrichment
analysis. Here we use the <code>deg_exp_cor</code> geneset to perform
enrichment on differentially abundant features correlated with
exposures. However, we could enrich the following gene sets:</p>
<ul>
<li><p><code>deg</code>: Differentially abundant features.</p></li>
<li><p><code>deg_exp_cor</code>: Differentially abundant features
correlated with exposures.</p></li>
<li><p><code>factor_exp_cor</code>: Top factor features correlated with
exposures.</p></li>
<li><p><code>factor</code>: Top actor features.</p></li>
</ul>
<div class="sourceCode" id="cb63"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Perform Functional Enrichment Analysis</span></span>
<span><span class="va">expom</span> <span class="op">&lt;-</span> <span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/run_enrichment.html">run_enrichment</a></span><span class="op">(</span></span>
<span>    geneset <span class="op">=</span> <span class="st">"deg_exp_cor"</span>,</span>
<span>    feature_col <span class="op">=</span> <span class="st">"gene"</span>,</span>
<span>    clustering_approach <span class="op">=</span> <span class="st">"dynamic"</span>,</span>
<span>    pval_threshold <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>    pval_col <span class="op">=</span> <span class="st">"p.value"</span>,</span>
<span>    pvalueCutoff <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>    pAdjustMethod <span class="op">=</span> <span class="st">"none"</span>,</span>
<span>    qvalueCutoff <span class="op">=</span> <span class="fl">1</span>,</span>
<span>    logfc_threshold <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span>,</span>
<span>    action<span class="op">=</span><span class="st">"add"</span><span class="op">)</span></span></code></pre></div>
<pre><code>Working on: Gene Expression
Working on: Proteomics
Determining Number of GO Term Clusters...
 ..cutHeight not given, setting it to 485  ===&gt;  99% of the (truncated) height range in dendro.
 ..done.
Optimal number of clusters for samples: 11</code></pre>
<div class="section level3">
<h3 id="plot-enrichment-results">Plot Enrichment Results<a class="anchor" aria-label="anchor" href="#plot-enrichment-results"></a>
</h3>
<p>To visualize our enrichment results, we can use the
<code>plot_dotplot_enrichment</code> function. The <code>geneset</code>
argument is used to set the geneset to plot. The <code>top_n</code>
argument is used to set the number of top enrichment terms to plot. The
<code>n_per_group</code> argument is used to set the number of
enrichment terms per group to plot. The <code>add_top_genes</code>
argument is used to add the top genes associated with each enrichment
term. The <code>top_n_genes</code> argument is used to set the number of
top genes to plot.</p>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot functional enrichment results</span></span>
<span><span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>    <span class="fu"><a href="../reference/plot_dotplot_enrichment.html">plot_dotplot_enrichment</a></span><span class="op">(</span>geneset <span class="op">=</span> <span class="st">"deg_exp_cor"</span>,</span>
<span>                            top_n <span class="op">=</span> <span class="fl">5</span>,</span>
<span>                            n_per_group <span class="op">=</span> <span class="fl">5</span>,</span>
<span>                            add_top_genes <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                            top_n_genes <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span></code></pre></div>
<p><img src="tidyexposomics_files/figure-html/unnamed-chunk-50-1.png" width="1344"></p>
<p>You’ll notice that enrichment terms are organized into groups by omic
layer, the exposure category these features are associated with, and the
Jaccard similarity of the GO term genesets. We then provide the top 5
groups of enrichment terms based on a prioritization score calculated
by:</p>
<p><span class="math display">\[ Prioritization\ Score = \frac{1}{N} *
\sum_{i=1}^{N} -log_{10}(p_i) * n_{genes} \]</span></p>
<p>Where: - <span class="math inline">\(N\)</span> is the number of
enrichment terms in the group - <span class="math inline">\(p_i\)</span>
is the p-value for the <span class="math inline">\(i^{th}\)</span>
enrichment term in the group - <span class="math inline">\(n_{genes}\)</span> is the number of genes in the
enrichment term</p>
<p>So essentially, this is the mean of the negative Log<sub>10</sub> of
the p-value multiplied by the number of genes in the term. Here we see
that that outdoor exposures and chemicals are broadly associated with
all of the top enrichment term groups.</p>
</div>
<div class="section level3">
<h3 id="exposures-driving-enrichment">Exposures Driving Enrichment<a class="anchor" aria-label="anchor" href="#exposures-driving-enrichment"></a>
</h3>
<p>We can also visualize the exposures driving the enrichment terms
using the <code>plot_go_group_exposures</code> function. The
<code>go_groups</code> argument is used to set the GO groups to plot.
The <code>feature_col</code> argument is used to set the column
containing the features to plot. We specify <code>feature_col</code>
because the gene expression assay rownames were probe IDs and not gene
names. This highlights the importance of specifying the correct feature
column when plotting enrichment results. Here we will plot the exposures
driving the enrichment terms for a specific GO group, “Group_1”.</p>
<div class="sourceCode" id="cb66"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/plot_go_group_exposures.html">plot_go_group_exposures</a></span><span class="op">(</span></span>
<span>    go_groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Group_1"</span><span class="op">)</span>,</span>
<span>    feature_col <span class="op">=</span> <span class="st">"gene"</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<p><img src="tidyexposomics_files/figure-html/unnamed-chunk-51-1.png" width="672" style="display: block; margin: auto;"></p>
<p>Here we note that similar exposures associated with both proteomics
and gene expression features drive the enrichment terms in this
group.</p>
</div>
</div>
<div class="section level2">
<h2 id="custom-analysis">Custom Analysis<a class="anchor" aria-label="anchor" href="#custom-analysis"></a>
</h2>
<p>Now what if there are analyses, summary statistics, or plots that you
would like to create on your own? We provide functionality to access the
underlying data in the <code>MultiAssayExperiment</code> object and
construct tibbles for your own analysis:</p>
<ul>
<li><p><code>pivot_sample</code>: Pivot the sample data to a tibble with
samples as rows and exposures as columns.</p></li>
<li><p><code>pivot_feature</code>: Pivot the feature meta data to a
tibble with features as rows and feature meta data as columns.</p></li>
<li><p><code>pivot_exp</code>: Pivot the sample and experiment assay
data to a tibble with samples as rows and sample meta data as columns.
Additionally, there will be a column for values for specified features
in specified assays.</p></li>
</ul>
<div class="section level3">
<h3 id="pivot-sample">Pivot Sample<a class="anchor" aria-label="anchor" href="#pivot-sample"></a>
</h3>
<p>Let’s check out the <code>pivot_sample</code> function. This function
pivots the sample data to a tibble with samples as rows and exposures as
columns.</p>
<div class="sourceCode" id="cb67"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Pivot sample data to a tibble</span></span>
<span><span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/pivot_sample.html">pivot_sample</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 6 × 269</span></span>m</span>
<span><span class="co">##   .sample  h_PM hs_greenyn300_h h_greenyn300_preg hs_participation_3cat  </span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>                  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> s548    13.9  1               1                 None                   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> s555     7.48 1               1                 None                   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> s368    12.7  0               0                 1 organisation         </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">4</span> s175     8.85 1               1                 None                   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">5</span> s241    14.7  1               1                 None                   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">6</span> s600    11.0  1               1                 2 or more organisations</span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 264 more variables: hs_smk_parents &lt;fct&gt;, hs_asthma &lt;ord&gt;,</span></span>9m</span>
<span><span class="co">## <span style="color: #949494;">#   hs_child_age &lt;dbl&gt;, e3_sex &lt;fct&gt;, PC1 &lt;dbl&gt;, PC2 &lt;dbl&gt;, PC3 &lt;dbl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   PC4 &lt;dbl&gt;, PC5 &lt;dbl&gt;, PC6 &lt;dbl&gt;, PC7 &lt;dbl&gt;, PC8 &lt;dbl&gt;, PC9 &lt;dbl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   PC10 &lt;dbl&gt;, PC11 &lt;dbl&gt;, PC12 &lt;dbl&gt;, PC13 &lt;dbl&gt;, PC14 &lt;dbl&gt;, PC15 &lt;dbl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   PC16 &lt;dbl&gt;, PC17 &lt;dbl&gt;, PC18 &lt;dbl&gt;, PC19 &lt;dbl&gt;, PC20 &lt;dbl&gt;, PC21 &lt;dbl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   PC22 &lt;dbl&gt;, PC23 &lt;dbl&gt;, PC24 &lt;dbl&gt;, PC25 &lt;dbl&gt;, PC26 &lt;dbl&gt;, PC27 &lt;dbl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   PC28 &lt;dbl&gt;, PC29 &lt;dbl&gt;, PC30 &lt;dbl&gt;, PC31 &lt;dbl&gt;, PC32 &lt;dbl&gt;, PC33 &lt;dbl&gt;, …</span></span>9m</span></code></pre>
<p>We could use this functionality to count the number of asthmatics per
sex:</p>
<div class="sourceCode" id="cb69"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/pivot_sample.html">pivot_sample</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">hs_asthma</span>,<span class="va">e3_sex</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/context.html" class="external-link">n</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> </span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 4 × 3</span></span>m</span>
<span><span class="co">## <span style="color: #949494;"># Groups:   hs_asthma [2]</span></span></span>
<span><span class="co">##   hs_asthma     e3_sex     n</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;ord&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> non_asthmatic female   125</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> non_asthmatic male     115</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> asthma        female    10</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">4</span> asthma        male      19</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="pivot-feature">Pivot Feature<a class="anchor" aria-label="anchor" href="#pivot-feature"></a>
</h3>
<p>The <code>pivot_feature</code> function pivots the feature meta data
to a tibble with features as rows and feature meta data as columns. This
can be useful for exploring the feature metadata in a more flexible
way.</p>
<div class="sourceCode" id="cb71"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Pivot feature data to a tibble</span></span>
<span><span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/pivot_feature.html">pivot_feature</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 6 × 30</span></span>m</span>
<span><span class="co">##   .exp_name     .feature transcript_cluster_id probeset_id seqname strand  start</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> Gene Express… TC01000… TC01000001.hg.1       TC01000001… chr1    +       <span style="text-decoration: underline;">11</span>869</span>24m869</span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> Gene Express… TC01000… TC01000002.hg.1       TC01000002… chr1    +       <span style="text-decoration: underline;">29</span>554</span>24m554</span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> Gene Express… TC01000… TC01000003.hg.1       TC01000003… chr1    +       <span style="text-decoration: underline;">69</span>091</span>24m091</span>
<span><span class="co">## <span style="color: #BCBCBC;">4</span> Gene Express… TC01000… TC01000005.hg.1       TC01000005… chr1    +      <span style="text-decoration: underline;">317</span>811</span>24m811</span>
<span><span class="co">## <span style="color: #BCBCBC;">5</span> Gene Express… TC01000… TC01000007.hg.1       TC01000007… chr1    +      <span style="text-decoration: underline;">321</span>146</span>24m146</span>
<span><span class="co">## <span style="color: #BCBCBC;">6</span> Gene Express… TC01000… TC01000009.hg.1       TC01000009… chr1    +      <span style="text-decoration: underline;">367</span>640</span>24m640</span>
<span><span class="co">## <span style="color: #949494;"># ℹ 23 more variables: stop &lt;int&gt;, total_probes &lt;int&gt;, gene_assignment &lt;chr&gt;,</span></span>9m</span>
<span><span class="co">## <span style="color: #949494;">#   mrna_assignment &lt;chr&gt;, notes &lt;chr&gt;, phase &lt;chr&gt;, TC_size &lt;dbl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   TSS_Affy &lt;int&gt;, EntrezeGeneID_Affy &lt;chr&gt;, GeneSymbol_Affy &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   GeneSymbolDB &lt;chr&gt;, GeneSymbolDB2 &lt;chr&gt;, mrna_ID &lt;chr&gt;, mrna_DB &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   mrna_N &lt;int&gt;, notesYN &lt;chr&gt;, geneYN &lt;chr&gt;, genes_N &lt;dbl&gt;, CallRate &lt;dbl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   fil1 &lt;chr&gt;, gene &lt;chr&gt;, Class &lt;chr&gt;, Class_2 &lt;chr&gt;</span></span></span></code></pre>
<p>We could use this functionality to count the number of features per
omic layer, or to filter features based on their metadata. For example,
we can count the number of features per omic layer:</p>
<div class="sourceCode" id="cb73"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Count the number of features per omic layer</span></span>
<span><span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>    <span class="fu"><a href="../reference/pivot_feature.html">pivot_feature</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">.exp_name</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/context.html" class="external-link">n</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 3 × 2</span></span>m</span>
<span><span class="co">##   .exp_name           n</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> Gene Expression <span style="text-decoration: underline;">28</span>738</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> Metabolomics      177</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> Proteomics       <span style="text-decoration: underline;">3</span>126</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="pivot-experiment">Pivot Experiment<a class="anchor" aria-label="anchor" href="#pivot-experiment"></a>
</h3>
<div class="sourceCode" id="cb75"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Pivot experiment data to a tibble</span></span>
<span><span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/pivot_exp.html">pivot_exp</a></span><span class="op">(</span>omics_name <span class="op">=</span> <span class="st">"Proteomics"</span>,</span>
<span>            features <span class="op">=</span> <span class="st">"TAF7"</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 6 × 273</span></span>m</span>
<span><span class="co">##   exp_name   .feature .sample counts  h_PM hs_greenyn300_h h_greenyn300_preg</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>            </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> Proteomics TAF7     s555      4.85  7.48 1               1                </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> Proteomics TAF7     s368      5.98 12.7  0               0                </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> Proteomics TAF7     s175     10.5   8.85 1               1                </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">4</span> Proteomics TAF7     s241      5.73 14.7  1               1                </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">5</span> Proteomics TAF7     s609      2.59 14.3  0               0                </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">6</span> Proteomics TAF7     s257      4.26 35.4  1               0                </span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 266 more variables: hs_participation_3cat &lt;fct&gt;, hs_smk_parents &lt;fct&gt;,</span></span>9m</span>
<span><span class="co">## <span style="color: #949494;">#   hs_asthma &lt;ord&gt;, hs_child_age &lt;dbl&gt;, e3_sex &lt;fct&gt;, PC1 &lt;dbl&gt;, PC2 &lt;dbl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   PC3 &lt;dbl&gt;, PC4 &lt;dbl&gt;, PC5 &lt;dbl&gt;, PC6 &lt;dbl&gt;, PC7 &lt;dbl&gt;, PC8 &lt;dbl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   PC9 &lt;dbl&gt;, PC10 &lt;dbl&gt;, PC11 &lt;dbl&gt;, PC12 &lt;dbl&gt;, PC13 &lt;dbl&gt;, PC14 &lt;dbl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   PC15 &lt;dbl&gt;, PC16 &lt;dbl&gt;, PC17 &lt;dbl&gt;, PC18 &lt;dbl&gt;, PC19 &lt;dbl&gt;, PC20 &lt;dbl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   PC21 &lt;dbl&gt;, PC22 &lt;dbl&gt;, PC23 &lt;dbl&gt;, PC24 &lt;dbl&gt;, PC25 &lt;dbl&gt;, PC26 &lt;dbl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   PC27 &lt;dbl&gt;, PC28 &lt;dbl&gt;, PC29 &lt;dbl&gt;, PC30 &lt;dbl&gt;, PC31 &lt;dbl&gt;, PC32 &lt;dbl&gt;, …</span></span>9m</span></code></pre>
<p>We can use this functionality to create custom plots or analyses
based on the exposure and feature data. For example, we can plot the
abundance of TAF7 by asthma status:</p>
<div class="sourceCode" id="cb77"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot the distribution of a specific feature across samples</span></span>
<span><span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/pivot_exp.html">pivot_exp</a></span><span class="op">(</span>omics_name <span class="op">=</span> <span class="st">"Proteomics"</span>,</span>
<span>            features <span class="op">=</span> <span class="st">"TAF7"</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">hs_asthma</span>, </span>
<span>             y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span>,</span>
<span>             color <span class="op">=</span> <span class="va">hs_asthma</span>,</span>
<span>             fill <span class="op">=</span> <span class="va">hs_asthma</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">geom_boxplot</a></span><span class="op">(</span>alpha<span class="op">=</span><span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_jitter.html" class="external-link">geom_jitter</a></span><span class="op">(</span>alpha<span class="op">=</span><span class="fl">0.1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggpubr</span><span class="fu">::</span><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/geom_pwc.html" class="external-link">geom_pwc</a></span><span class="op">(</span></span>
<span>     label <span class="op">=</span> <span class="st">"{p.adj.format}{p.adj.signif}"</span></span>
<span>  <span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_minimal</a></span><span class="op">(</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">ggpubr</span><span class="fu">::</span><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/rotate_axis_text.html" class="external-link">rotate_x_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_color_tidy_exp</span><span class="op">(</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_fill_tidy_exp</span><span class="op">(</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">""</span>,</span>
<span>       y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/expression.html" class="external-link">expression</a></span><span class="op">(</span><span class="va">Log</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">*</span><span class="st">"Abd."</span><span class="op">)</span>,</span>
<span>       fill <span class="op">=</span> <span class="st">"Asthma Status"</span>,</span>
<span>       color <span class="op">=</span> <span class="st">"Asthma Status"</span><span class="op">)</span></span></code></pre></div>
<p><img src="tidyexposomics_files/figure-html/unnamed-chunk-57-1.png" width="384" style="display: block; margin: auto;"></p>
</div>
</div>
<div class="section level2">
<h2 id="pipeline-overview">Pipeline Overview<a class="anchor" aria-label="anchor" href="#pipeline-overview"></a>
</h2>
<p>To summarize the steps we have taken in this analysis, we can use the
<code>run_pipeline_summary</code> function. This function will provide a
summary of the steps taken in the analysis. We can set
<code>console_print</code> to <code>TRUE</code> to print the summary to
the console. Setting <code>include_notes</code> to <code>TRUE</code>
will include notes on the steps taken in the analysis.</p>
<div class="sourceCode" id="cb78"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Run the pipeline summary</span></span>
<span><span class="va">expom</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/run_pipeline_summary.html">run_pipeline_summary</a></span><span class="op">(</span></span>
<span>    console_print <span class="op">=</span> <span class="cn">T</span>,</span>
<span>    include_notes <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 1. filter_missing — Filtered exposure variables and omics features with more than 1% missing values. QC plots generated for exposure and omics data.</span></span>
<span><span class="co">## 2. run_impute_missing — </span></span>
<span><span class="co">## 3. run_pca — Outliers:  s936, s764, s588, s857, s378, s411, s918</span></span>
<span><span class="co">## 4. filter_sample_outliers — Outliers:  s411, s378, s588, s764, s857, s918, s936</span></span>
<span><span class="co">## 5. run_normality_check — Assessed normality of 37 numeric exposure variables. 0 were normally distributed (p &gt; 0.05), 37 were not.</span></span>
<span><span class="co">## 6. transform_exposure — Applied 'boxcox_best' transformation to 35 exposure variables. 2 passed normality (Shapiro-Wilk p &gt; 0.05, 5.7%).</span></span>
<span><span class="co">## 7. run_cluster_samples — Optimal number of clusters for samples: 6</span></span>
<span><span class="co">## 8. run_correlation — Correlated exposures features with exposures.</span></span>
<span><span class="co">## 9. run_association — Performed association analysis using source: exposures</span></span>
<span><span class="co">## 10. run_association — Performed association analysis using source: exposures</span></span>
<span><span class="co">## 11. run_exposome_score_exposome_median_score — Exposome score computed using method: 'median'</span></span>
<span><span class="co">## 12. run_exposome_score_exposome_pca_score — Exposome score computed using method: 'pca'</span></span>
<span><span class="co">## 13. run_exposome_score_exposome_irt_score — Exposome score computed using method: 'irt'</span></span>
<span><span class="co">## 14. run_exposome_score_exposome_quantile_score — Exposome score computed using method: 'quantile'</span></span>
<span><span class="co">## 15. run_exposome_score_exposome_var_score — Exposome score computed using method: 'var'</span></span>
<span><span class="co">## 16. run_association — Performed association analysis using source: exposures</span></span>
<span><span class="co">## 17. run_differential_abundance — Performed differential abundance analysis across all assays.</span></span>
<span><span class="co">## 18. run_sensitivity_analysis_bootstrap — Performed bootstrapped sensitivity analysis ( 3  iterations) using 1 methods and 1 scaling strategies, with feature-level stability scores recorded per run. Covariates removed in model variations: hs_child_age, e3_sex</span></span>
<span><span class="co">## 19. run_multiomics_integration — Performed multi-omics integration using MCIA with 10 latent factors. Scaling was enabled.</span></span>
<span><span class="co">## 20. run_association — Performed association analysis using source: exposures</span></span>
<span><span class="co">## 21. extract_top_factor_features — Selected 9612 features contributing to specified factors.</span></span>
<span><span class="co">## 22. run_factor_overlap — Identified shared top features across integration factors and annotated based on 'stability_score' scores.</span></span>
<span><span class="co">## 23. run_correlation — Correlated exposures features with exposures.</span></span>
<span><span class="co">## 24. run_create_network — Created undirected network from correlation results for feature_type = 'degs'.</span></span>
<span><span class="co">## 25. run_exposure_impact — Computed exposure impact using degs correlation network.</span></span>
<span><span class="co">## 26. run_enrichment — Performed enrichment on deg_exp_cor features.</span></span></code></pre>
<div class="DiagrammeR html-widget html-fill-item" id="htmlwidget-ac96cb3ee4656e2e9ec3" style="width:480px;height:1440px;"></div>
<script type="application/json" data-for="htmlwidget-ac96cb3ee4656e2e9ec3">{"x":{"diagram":"graph TD\nstep1[\"1. filter_missing — Filtered exposure variables and omics features with more than 1% missing values. QC plots generated for exposure and omics data.\"] --> step2[\"2. run_impute_missing — \"]\nstep2[\"2. run_impute_missing — \"] --> step3[\"3. run_pca — Outliers:  s936, s764, s588, s857, s378, s411, s918\"]\nstep3[\"3. run_pca — Outliers:  s936, s764, s588, s857, s378, s411, s918\"] --> step4[\"4. filter_sample_outliers — Outliers:  s411, s378, s588, s764, s857, s918, s936\"]\nstep4[\"4. filter_sample_outliers — Outliers:  s411, s378, s588, s764, s857, s918, s936\"] --> step5[\"5. run_normality_check — Assessed normality of 37 numeric exposure variables. 0 were normally distributed (p > 0.05), 37 were not.\"]\nstep5[\"5. run_normality_check — Assessed normality of 37 numeric exposure variables. 0 were normally distributed (p > 0.05), 37 were not.\"] --> step6[\"6. transform_exposure — Applied 'boxcox_best' transformation to 35 exposure variables. 2 passed normality (Shapiro-Wilk p > 0.05, 5.7%).\"]\nstep6[\"6. transform_exposure — Applied 'boxcox_best' transformation to 35 exposure variables. 2 passed normality (Shapiro-Wilk p > 0.05, 5.7%).\"] --> step7[\"7. run_cluster_samples — Optimal number of clusters for samples: 6\"]\nstep7[\"7. run_cluster_samples — Optimal number of clusters for samples: 6\"] --> step8[\"8. run_correlation — Correlated exposures features with exposures.\"]\nstep8[\"8. run_correlation — Correlated exposures features with exposures.\"] --> step9[\"9. run_association — Performed association analysis using source: exposures\"]\nstep9[\"9. run_association — Performed association analysis using source: exposures\"] --> step10[\"10. run_association — Performed association analysis using source: exposures\"]\nstep10[\"10. run_association — Performed association analysis using source: exposures\"] --> step11[\"11. run_exposome_score_exposome_median_score — Exposome score computed using method: 'median'\"]\nstep11[\"11. run_exposome_score_exposome_median_score — Exposome score computed using method: 'median'\"] --> step12[\"12. run_exposome_score_exposome_pca_score — Exposome score computed using method: 'pca'\"]\nstep12[\"12. run_exposome_score_exposome_pca_score — Exposome score computed using method: 'pca'\"] --> step13[\"13. run_exposome_score_exposome_irt_score — Exposome score computed using method: 'irt'\"]\nstep13[\"13. run_exposome_score_exposome_irt_score — Exposome score computed using method: 'irt'\"] --> step14[\"14. run_exposome_score_exposome_quantile_score — Exposome score computed using method: 'quantile'\"]\nstep14[\"14. run_exposome_score_exposome_quantile_score — Exposome score computed using method: 'quantile'\"] --> step15[\"15. run_exposome_score_exposome_var_score — Exposome score computed using method: 'var'\"]\nstep15[\"15. run_exposome_score_exposome_var_score — Exposome score computed using method: 'var'\"] --> step16[\"16. run_association — Performed association analysis using source: exposures\"]\nstep16[\"16. run_association — Performed association analysis using source: exposures\"] --> step17[\"17. run_differential_abundance — Performed differential abundance analysis across all assays.\"]\nstep17[\"17. run_differential_abundance — Performed differential abundance analysis across all assays.\"] --> step18[\"18. run_sensitivity_analysis_bootstrap — Performed bootstrapped sensitivity analysis ( 3  iterations) using 1 methods and 1 scaling strategies, with feature-level stability scores recorded per run. Covariates removed in model variations: hs_child_age, e3_sex\"]\nstep18[\"18. run_sensitivity_analysis_bootstrap — Performed bootstrapped sensitivity analysis ( 3  iterations) using 1 methods and 1 scaling strategies, with feature-level stability scores recorded per run. Covariates removed in model variations: hs_child_age, e3_sex\"] --> step19[\"19. run_multiomics_integration — Performed multi-omics integration using MCIA with 10 latent factors. Scaling was enabled.\"]\nstep19[\"19. run_multiomics_integration — Performed multi-omics integration using MCIA with 10 latent factors. Scaling was enabled.\"] --> step20[\"20. run_association — Performed association analysis using source: exposures\"]\nstep20[\"20. run_association — Performed association analysis using source: exposures\"] --> step21[\"21. extract_top_factor_features — Selected 9612 features contributing to specified factors.\"]\nstep21[\"21. extract_top_factor_features — Selected 9612 features contributing to specified factors.\"] --> step22[\"22. run_factor_overlap — Identified shared top features across integration factors and annotated based on 'stability_score' scores.\"]\nstep22[\"22. run_factor_overlap — Identified shared top features across integration factors and annotated based on 'stability_score' scores.\"] --> step23[\"23. run_correlation — Correlated exposures features with exposures.\"]\nstep23[\"23. run_correlation — Correlated exposures features with exposures.\"] --> step24[\"24. run_create_network — Created undirected network from correlation results for feature_type = 'degs'.\"]\nstep24[\"24. run_create_network — Created undirected network from correlation results for feature_type = 'degs'.\"] --> step25[\"25. run_exposure_impact — Computed exposure impact using degs correlation network.\"]\nstep25[\"25. run_exposure_impact — Computed exposure impact using degs correlation network.\"] --> step26[\"26. run_enrichment — Performed enrichment on deg_exp_cor features.\"]"},"evals":[],"jsHooks":[]}</script><div class="section level3">
<h3 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h3>
<ol style="list-style-type: decimal">
<li><p>Wild CP. Complementing the genome with an ‘exposome’: the
outstanding challenge of environmental exposure measurement in molecular
epidemiology. <em>Cancer Epidemiol Biomarkers Prev.</em>
2005;14(8):1847–1850. <a href="https://pubmed.ncbi.nlm.nih.gov/16103423/" class="external-link">PubMed</a></p></li>
<li><p>Chung MK, Holmes L, Gao X, Ghosh D, Li Y, Lee YJ. Decoding the
exposome: data science methodologies and implications in exposome-wide
association studies (ExWASs). <em>Exposome.</em> 2024;4(1):osae001. <a href="https://academic.oup.com/exposome/article/4/1/osae001/7574628" class="external-link">Exposome</a></p></li>
<li><p>Maitre L, Guimbaud JB, Warembourg C, Güil-Oumrait N, Petrone PM,
Chadeau-Hyam M, Vrijheid M, Basagaña X, Gonzalez JR; Exposome Data
Challenge Participant Consortium. State-of-the-art methods for
exposure-health studies: Results from the exposome data challenge event.
<em>Environ Int.</em> 2022;168:107422. doi: <a href="https://doi.org/10.1016/j.envint.2022.107422" class="external-link">10.1016/j.envint.2022.107422</a></p></li>
<li><p>MultiAssay Special Interest Group. MultiAssayExperiment: Manage
and Analyze Multi-Omics Experiments. <em>Bioconductor Vignette</em>,
2024. <a href="https://www.bioconductor.org/packages/release/bioc/vignettes/MultiAssayExperiment/inst/doc/MultiAssayExperiment.html" class="external-link">Link</a></p></li>
</ol>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Jason Laird.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
